SGE analysis pipeline - VHL_global_20230403.Rmd
Version 23.04.12.VHL.
RStudio Version 1.4.1106
Written by Greg Findlay
The Francis Crick Institute

Updated global analysis scripts for VHL SGE project.

Requires having run "230412_VHL_all_no_plots.Rmd", as dataframes produced in this script for each sge region are combined into a merged dataframe to perform global analyses included in the study.

Purposes of the script include:
 - To apply filters to remove variants not well represented in the experiment.
 - To derive final function scores by normalizing across regions
 - To derive final RNA scores
 - To annotate variants with data from additional databases (e.g. ClinVar, cBioPortal, comp predictors, etc.)
 - To analyze SGE data in conjunction with other data types, for instance, to determine predictive power
 - To produce nearly all figures for publication
 - To produce the supplementary table containing all experimentally derived data


```{r}
#Load packages needed
library(plyr)
library(dplyr)
library(gam)
library(plotROC)
require(tidyverse)
require(ggplot2)
require(gridExtra)
require(mclust)
require(VennDiagram)
library(mixtools)
library(plotROC)
library(PRROC)

#Defining colour schemes for plotting
named_conseq_colors_clinvar_path.lp_benign = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#083092","Conflicting interpretations of pathogenicity" = "#EBEAEA","Likely benign" = "#EBEAEA","Likely pathogenic" = "#A40000","Pathogenic" = "#A40000","Uncertain significance" = "#EBEAEA" )

named_conseq_colors_meg = c("STOP_GAINED" = "#CC6677","CANONICAL_SPLICE" = "#DDCC77","INTRONIC" = "#646CFF","NON_SYNONYMOUS" = "#326C37","SPLICE_SITE" = "#66B1A3","SYNONYMOUS" = "#88CCEE","STOP_LOST" = "#AA4499","REGULATORY" = "#929000","5PRIME_UTR" = "#929000","3PRIME_UTR" = "#332288","REF" = "#000000")

#final colors
named_conseq_colors = c("STOP_GAINED" = "#BD0016","CANONICAL_SPLICE" = "#FFAA38","INTRONIC" = "#88CCEE", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#15327C", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#CB57D7","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#EA93F3", "REF" = "#000000")

named_conseq_colors_ns_cs_syn = c("STOP_GAINED" = "#BD0016","CANONICAL_SPLICE" = "#FFAA38", "SYNONYMOUS" = "#456CCC","INTRONIC" = "#B0B2B3", "NON_SYNONYMOUS" = "#B0B2B3", "SPLICE_SITE" = "#B0B2B3","STOP_LOST" = "#B0B2B3","REGULATORY"="#B0B2B3","5PRIME_UTR"="#B0B2B3", "3PRIME_UTR"="#B0B2B3", "REF" = "#B0B2B3", "absent" = "#B0B2B3")

named_conseq_colors_full = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#0B3DB9", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#CF7700", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#013DD3","Conflicting interpretations of pathogenicity" = "#FFCB00","Likely benign" = "#2360F9","Likely pathogenic" = "#FA2039","Pathogenic" = "#A40000","Uncertain significance" = "#00A54D" )
named_conseq_colors_full_simple = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#013DD3","Conflicting interpretations of pathogenicity" = "#FFCB00","Likely benign" = "#2360F9","Likely pathogenic" = "#FA2039","Pathogenic" = "#A40000","Uncertain significance" = "#00A54D","Gray" = "#000000" )
named_conseq_colors_hide_missense = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#0B3DB9", "NON_SYNONYMOUS" = "#FFFFFF", "SPLICE_SITE" = "#CF7700", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#47A300", "REF" = "#000000")
named_conseq_colors_highlight_mis = c("STOP_GAINED" = "#B0B2B3","CANONICAL_SPLICE" = "#B0B2B3","INTRONIC" = "#B0B2B3", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#B0B2B3", "SYNONYMOUS" = "#B0B2B3","STOP_LOST" = "#B0B2B3","REGULATORY"="#B0B2B3","5PRIME_UTR"="#B0B2B3", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#00BA74","Conflicting interpretations of pathogenicity" = "#2C2C2C","Likely benign" = "#35CD93","Likely pathogenic" = "#FA2039","Pathogenic" = "#A40000","Uncertain significance" = "#C6C6C6" )
named_conseq_colors_highlight_syn_int = c("STOP_GAINED" = "#B0B2B3","CANONICAL_SPLICE" = "#B0B2B3","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#B0B2B3", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#B0B2B3","REGULATORY"="#B0B2B3","5PRIME_UTR"="#B0B2B3", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#00BA74","Conflicting interpretations of pathogenicity" = "#2C2C2C","Likely benign" = "#35CD93","Likely pathogenic" = "#FA2039","Pathogenic" = "#A40000","Uncertain significance" = "#C6C6C6" )
named_conseq_colors_clinvar_gray = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#EBEAEA","Conflicting interpretations of pathogenicity" = "#EBEAEA","Likely benign" = "#EBEAEA","Likely pathogenic" = "#EBEAEA","Pathogenic" = "#EBEAEA","Uncertain significance" = "#EBEAEA" )
named_conseq_colors_clinvar_path.lp = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#EBEAEA","Conflicting interpretations of pathogenicity" = "#EBEAEA","Likely benign" = "#EBEAEA","Likely pathogenic" = "#A40000","Pathogenic" = "#A40000","Uncertain significance" = "#EBEAEA" )
named_conseq_colors_clinvar_path.lp_benign.lb_confl.vus = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#083092","Conflicting interpretations of pathogenicity" = "#FFB400","Likely benign" = "#083092","Likely pathogenic" = "#A40000","Pathogenic" = "#A40000","Uncertain significance" = "#FFB400" )
named_conseq_colors_clinvar_path.lp_benign.lb = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#EBEAEA","Benign" = "#083092","Conflicting interpretations of pathogenicity" = "#EBEAEA","Likely benign" = "#083092","Likely pathogenic" = "#A40000","Pathogenic" = "#A40000","Uncertain significance" = "#EBEAEA" )
named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#979090","Benign" = "#0433FF","Conflicting interpretations of pathogenicity" = "#936104","Likely benign" = "#486BFC","Likely pathogenic" = "#C56B99","Pathogenic" = "#882255","Uncertain significance" = "#DCA237" )


#Defining colours -- here from Meg. Different colours for different sets being plotted.
named_conseq_colors_clinvar_path.lp_benign = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#88CCEE",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#88CCEE",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#EBEAEA",
  "Conflicting interpretations of pathogenicity" = "#EBEAEA",
  "Likely benign" = "#EBEAEA",
  "Likely pathogenic" = "#882255",
  "Pathogenic" = "#882255",
  "Uncertain significance" = "#EBEAEA")
named_conseq_colors_full = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#646CFF",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#DCA237",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#0433FF",
  "Conflicting interpretations of pathogenicity" = "#EBEAEA",
  "Likely benign" = "#EBEAEA",
  "Likely pathogenic" = "#882255",
  "Pathogenic" = "#882255",
  "Uncertain significance" = "#EBEAEA")
named_conseq_colors_full_simple = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#88CCEE",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#88CCEE",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#0433FF",
  "Conflicting interpretations of pathogenicity" = "#929000",
  "Likely benign" = "#88CCEE",
  "Likely pathogenic" = "#AA4499",
  "Pathogenic" = "#CC6677",
  "Uncertain significance" = "#326C37",
  "Gray" = "#000000")
named_conseq_colors_hide_missense = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#646CFF",
  "NON_SYNONYMOUS" = "#FFFFFF",
  "SPLICE_SITE" = "#DCA237",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#47A300",
  "3PRIME_UTR" = "#332288",  
  "REF" = "#000000")
named_conseq_colors_highlight_mis = c("STOP_GAINED" = "#EBEAEA",
  "CANONICAL_SPLICE" = "#EBEAEA",
  "INTRONIC" = "#EBEAEA",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#EBEAEA",
  "SYNONYMOUS" = "#EBEAEA",
  "STOP_LOST" = "#EBEAEA",
  "REGULATORY" = "#EBEAEA",
  "5PRIME_UTR" = "#EBEAEA",
  "3PRIME_UTR" = "#EBEAEA",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#326C37",
  "Conflicting interpretations of pathogenicity" = "#2C2C2C",
  "Likely benign" = "#66B1A3",
  "Likely pathogenic" = "#CC6677",
  "Pathogenic" = "#882255",
  "Uncertain significance" = "#C6C6C6")
named_conseq_colors_highlight_syn_int = c("STOP_GAINED" = "#EBEAEA",
  "CANONICAL_SPLICE" = "#EBEAEA",
  "INTRONIC" = "#646CFF",
  "NON_SYNONYMOUS" = "#EBEAEA",
  "SPLICE_SITE" = "#646CFF",
  "SYNONYMOUS" = "#646CFF",
  "STOP_LOST" = "#EBEAEA",
  "REGULATORY" = "#EBEAEA",
  "5PRIME_UTR" = "#EBEAEA",
  "3PRIME_UTR" = "#EBEAEA",  
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#326C37",
  "Conflicting interpretations of pathogenicity" = "#2C2C2C",
  "Likely benign" = "#66B1A3",
  "Likely pathogenic" = "#CC6677",
  "Pathogenic" = "#882255",
  "Uncertain significance" = "#C6C6C6")
named_conseq_colors_clinvar_gray = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#88CCEE",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#88CCEE",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#EBEAEA",
  "Conflicting interpretations of pathogenicity" = "#EBEAEA",
  "Likely benign" = "#EBEAEA",
  "Likely pathogenic" = "#EBEAEA",
  "Pathogenic" = "#EBEAEA",
  "Uncertain significance" = "#EBEAEA")
named_conseq_colors_clinvar_path.lp = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#88CCEE",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#88CCEE",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#EBEAEA",
  "Conflicting interpretations of pathogenicity" = "#EBEAEA",
  "Likely benign" = "#EBEAEA",
  "Likely pathogenic" = "#882255",
  "Pathogenic" = "#882255",
  "Uncertain significance" = "#EBEAEA")
named_conseq_colors_clinvar_path.lp_benign.lb_confl.vus = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#88CCEE",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#88CCEE",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#7C7B7B",
  "Benign" = "#0433FF",
  "Conflicting interpretations of pathogenicity" = "#DCA237",
  "Likely benign" = "#0433FF",
  "Likely pathogenic" = "#882255",
  "Pathogenic" = "#882255",
  "Uncertain significance" = "#DCA237")
named_conseq_colors_clinvar_path.lp_benign.lb = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#88CCEE",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#88CCEE",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#0433FF",
  "Conflicting interpretations of pathogenicity" = "#EBEAEA",
  "Likely benign" = "#0433FF",
  "Likely pathogenic" = "#882255",
  "Pathogenic" = "#882255",
  "Uncertain significance" = "#EBEAEA")

named_conseq_colors_clinvar_path_benign.lb = c("STOP_GAINED" = "#CC6677",
  "CANONICAL_SPLICE" = "#DDCC77",
  "INTRONIC" = "#88CCEE",
  "NON_SYNONYMOUS" = "#326C37",
  "SPLICE_SITE" = "#88CCEE",
  "SYNONYMOUS" = "#88CCEE",
  "STOP_LOST" = "#AA4499",
  "REGULATORY" = "#929000",
  "5PRIME_UTR" = "#929000",
  "3PRIME_UTR" = "#332288",
  "REF" = "#000000",
  "absent" = "#EBEAEA",
  "Benign" = "#0433FF",
  "Conflicting interpretations of pathogenicity" = "#EBEAEA",
  "Likely benign" = "#0433FF",
  "Likely pathogenic" = "#EBEAEA",
  "Pathogenic" = "#882255",
  "Uncertain significance" = "#EBEAEA")
```

```{r}
#requires having run all scripts for specific regions
VHLx2rLD_tHDR_pos_merge_df$hdr_cut <- VHLx2rLD_tHDR_pos_merge_df$hdr5
VHLx3arLD_tHDR_pos_merge_df$hdr_cut <- VHLx3arLD_tHDR_pos_merge_df$hdr5
VHLx3brLD_tHDR_pos_merge_df$hdr_cut <- VHLx3brLD_tHDR_pos_merge_df$hdr5
VHLx1brLD_tHDR_pos_merge_df$hdr_cut <- VHLx1brLD_tHDR_pos_merge_df$hdr5
VHLx1c2rLD_tHDR_pos_merge_df$hdr_cut <- VHLx1c2rLD_tHDR_pos_merge_df$hdr5
VHLx1a2rLD_tHDR_pos_merge_df$hdr_cut <- VHLx1a2rLD_tHDR_pos_merge_df$hdr5
VHLx1prLD_tHDR_pos_merge_df$hdr_cut <- VHLx1prLD_tHDR_pos_merge_df$hdr5

VHLx2rLD_tHDR_pos_merge_df$sge_region <- "exon 2"
VHLx3arLD_tHDR_pos_merge_df$sge_region <- "exon 3a"
VHLx3brLD_tHDR_pos_merge_df$sge_region <- "exon 3b"
VHLx1brLD_tHDR_pos_merge_df$sge_region <- "exon 1b"
VHLx1a2rLD_tHDR_pos_merge_df$sge_region <- "exon 1a"
VHLx1c2rLD_tHDR_pos_merge_df$sge_region <- "exon 1c"
VHLx1prLD_tHDR_pos_merge_df$sge_region <- "exon 1p"

#cut positions -- hg19
VHLx1a_cut_pos = 10183863+.5
VHLx1a2_cut_pos = 10183863+.5
VHLx1b_cut_pos = 10183661+.5
VHLx1c_cut_pos = 10183571+.5
VHLx1c2_cut_pos = 10183571+.5
VHLx1p_cut_pos = 10184471+.5
VHLx2_cut_pos = 10188239+.5
VHLx3a_cut_pos = 10191502+.5
VHLx3b_cut_pos = 10191631+.5

#binding all of the merge df's together...
VHLxrLD_all_master_df <- rbind(VHLx1c2rLD_tHDR_pos_merge_df,VHLx1brLD_tHDR_pos_merge_df,VHLx1a2rLD_tHDR_pos_merge_df,VHLx1prLD_tHDR_pos_merge_df,VHLx2rLD_tHDR_pos_merge_df,VHLx3arLD_tHDR_pos_merge_df,VHLx3brLD_tHDR_pos_merge_df)

VHLxrLD_all_master_df$pos_alt <- paste(VHLxrLD_all_master_df$pos, VHLxrLD_all_master_df$alt, sep='')
VHLxrLD_all_master_df$function_score <- log2(VHLxrLD_all_master_df$tHDR_post_pre_ratio_rLD1rLD2_mean_synnorm)

#calculate a post / post2 ratio, for rLD1, rLD2, and mean, normalise to synonymous
VHLxrLD_all_master_df$tHDR_post_post2_ratio <- VHLxrLD_all_master_df$tHDR_post_pseudo_freq/VHLxrLD_all_master_df$tHDR_post2_pseudo_freq
VHLxrLD_all_master_df$rLD2_tHDR_post_post2_ratio <- VHLxrLD_all_master_df$rLD2_tHDR_post_pseudo_freq/VHLxrLD_all_master_df$rLD2_tHDR_post2_pseudo_freq
VHLxrLD_all_master_df$tHDR_post_post2_ratio_rLD1rLD2_mean <- 2^(log2(VHLxrLD_all_master_df$tHDR_post_pseudo_freq/VHLxrLD_all_master_df$tHDR_post2_pseudo_freq)+log2(VHLxrLD_all_master_df$rLD2_tHDR_post_pseudo_freq/VHLxrLD_all_master_df$rLD2_tHDR_post2_pseudo_freq)/2)
VHLxrLD_all_tHDR_post_post2_ratio_rLD1rLD2_med_syn <- median(VHLxrLD_all_master_df[which(VHLxrLD_all_master_df$conseq == 'SYNONYMOUS'),]$tHDR_post_post2_ratio_rLD1rLD2_mean)

VHLxrLD_all_master_df$tHDR_post_post2_ratio_rLD1rLD2_mean_synnorm <- VHLxrLD_all_master_df$tHDR_post_post2_ratio_rLD1rLD2_mean/VHLxrLD_all_tHDR_post_post2_ratio_rLD1rLD2_med_syn


VHLxrLD_all_master_df <- VHLxrLD_all_master_df %>% 
  mutate(pHGVS = case_when(
    !is.na(Intron) ~ NA_character_,
    conseq == "5PRIME_UTR" ~ NA_character_,
    !is.na(Exon) ~ paste0("p.", oAA, protPos, nAA)
  ))

VHLxrLD_all_master_df <- VHLxrLD_all_master_df %>% 
  mutate(cHGVS = case_when(
    (is.na(CDSpos)&(conseq == "5PRIME_UTR")) ~ paste0("c.-",10183532-as.numeric(as.character(pos)), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 1a")) ~ paste0("c.340+", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 1p")) ~ paste0("c.340+", abs(as.numeric(as.character(pos))-10183871), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 2")) ~ paste0("c.341-", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "2/2")&(sge_region == "exon 2")) ~ paste0("c.463+", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "2/2")&(sge_region == "exon 3a")) ~ paste0("c.464-", abs(as.numeric(as.character(pos))-10191471), Ref,">",alt),
    (is.na(CDSpos)&(conseq == "3PRIME_UTR")) ~ paste0("c.*",as.numeric(as.character(pos))-10191649, Ref,">",alt),
    !is.na(CDSpos) ~ paste0("c.", CDSpos, Ref,">",alt)
  ))

VHLxrLD_all_master_df$hg38_pos <- as.numeric(as.character(VHLxrLD_all_master_df$pos))-41684
VHLxrLD_all_master_df$hg38_pos_alt <- paste0(VHLxrLD_all_master_df$hg38_pos,VHLxrLD_all_master_df$alt)

#For saving work here:
write.table(VHLxrLD_all_master_df, "/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLxrLD_all_master_20230412.txt", sep="\t")
#test_VHLxrLD_all_master_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLxrLD_all_master_20230412.txt", sep="\t", header=TRUE)

#threshold for replicate agreement #removes 5 points with scores >2 different
lib_freq_thresh <- which(VHLxrLD_all_master_df$lib_pseudo_freq >= 0.0001)
pre_freq_thresh <- which(VHLxrLD_all_master_df$tHDR_pre_pseudo_freq >= 0.00001 & VHLxrLD_all_master_df$rLD2_tHDR_pre_pseudo_freq >= 0.00001)

#Filter for discordancy between replicates set at difference of 2 or more in log2 ratios AND scoring above -1 in at least 1 replicate (i.e. won't filter our variants scoring at -2, -4, as there is more noise in low end of measurements)
repl_conc_thresh <- which((abs(log2(VHLxrLD_all_master_df$tHDR_post_pre_ratio_synnorm)-log2(VHLxrLD_all_master_df$rLD2_tHDR_post_pre_ratio_synnorm))<1.5)|(log2(VHLxrLD_all_master_df$tHDR_post_pre_ratio_synnorm) <= -1 & log2(VHLxrLD_all_master_df$rLD2_tHDR_post_pre_ratio_synnorm) <= -1) |(log2(VHLxrLD_all_master_df$tHDR_post_pre_ratio_synnorm) >= 0 & log2(VHLxrLD_all_master_df$rLD2_tHDR_post_pre_ratio_synnorm) >= 0))

#Filter removing variants with post2 scores widely out of line with post scores
post2_conc_thresh <- which((abs(log2(VHLxrLD_all_master_df$tHDR_post_post2_ratio_rLD1rLD2_mean_synnorm)-log2(VHLxrLD_all_master_df$tHDR_post2_pre_ratio_rLD1rLD2_mean_synnorm))<2)|(log2(VHLxrLD_all_master_df$tHDR_post_post2_ratio_rLD1rLD2_mean_synnorm) <= -0.5 & log2(VHLxrLD_all_master_df$tHDR_post2_pre_ratio_rLD1rLD2_mean_synnorm) <= -0.5))

#Filter removing variants with high predicted rate of reads caused by sequencing error -- note, this only acts on cHDR reads, and doesn't include positions for which no error rate could be calculated due to overlapping PAM edit (these get left in)
seq_error_thresh <- which((VHLxrLD_all_master_df$pre-VHLxrLD_all_master_df$corr_pre)/VHLxrLD_all_master_df$pre < 0.5 | (VHLxrLD_all_master_df$rLD2_pre-VHLxrLD_all_master_df$rLD2_corr_pre)/VHLxrLD_all_master_df$rLD2_pre <0.5 | is.na((VHLxrLD_all_master_df$pre-VHLxrLD_all_master_df$corr_pre)/VHLxrLD_all_master_df$pre) | is.na((VHLxrLD_all_master_df$rLD2_pre-VHLxrLD_all_master_df$rLD2_corr_pre)/VHLxrLD_all_master_df$rLD2_pre))

not_1c_thresh <- which((VHLxrLD_all_master_df$sge_region != "exon 1c"))

filter_combo_1 <- intersect(lib_freq_thresh,pre_freq_thresh)
filter_combo_2 <- intersect(filter_combo_1,repl_conc_thresh)
filter_combo_3a <- intersect(filter_combo_2,post2_conc_thresh)
filter_combo_3b <- intersect(filter_combo_2,seq_error_thresh)
filter_combo_4 <- intersect(filter_combo_3a,seq_error_thresh)
filter_combo_5 <- intersect(filter_combo_4,not_1c_thresh)

#no filter
ggplot(VHLxrLD_all_master_df[,], aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rLD2_tHDR_post_pre_ratio_synnorm),color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('tHDR post/pre repl. 2, log-2')+xlab('tHDR post/pre repl. 1, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#filter set 4
ggplot(VHLxrLD_all_master_df[filter_combo_4,], aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rLD2_tHDR_post_pre_ratio_synnorm),color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('tHDR post/pre repl. 2, log-2')+xlab('tHDR post/pre repl. 1, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#cHDR discordancy?
ggplot(VHLxrLD_all_master_df[filter_combo_4,], aes(color=conseq))+geom_point(alpha=1,aes(x=log2(post_pre_ratio_synnorm), y=log2(rLD2_post_pre_ratio_synnorm)))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('cHDR post/pre repl. 2, log-2')+xlab('cHDR post/pre repl. 1, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#does cHDR not match tHDR?
ggplot(VHLxrLD_all_master_df[filter_combo_4,], aes(x=log2(post_pre_ratio_rLD1rLD2_mean_synnorm), y=log2(tHDR_post_pre_ratio_rLD1rLD2_mean_synnorm),color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('tHDR post/pre mean, log-2')+xlab('cHDR post/pre mean, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=cHGVS,size=0.1))

#new dataframe using filter combo 4 (including exon 1c region variants)
final_filtered_df <- VHLxrLD_all_master_df[filter_combo_4,]
final_filtered_df <- final_filtered_df[which(is.na(final_filtered_df$cHGVS)==FALSE),]

#optional progress save
write.table(final_filtered_df, "/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLrLD_final_filtered_df_20230412.txt", sep="\t")
#final_filtered_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLrLD_final_filtered_df_20230412.txt", sep="\t", header=TRUE)

```
Section on global scaling of filtered data by SYN and NS variants
general formula:  b/(a-c)*(A-C) where b is snv's function score, a is sge_region med syn, c is sge_region med nonsense, A is global med syn, C is global med nonsense
Note:  using only nonsense variants in region of known functional impact p55-p194.

```{r}
#do this first for the repl. mean fits and scores :
VHLxrLD_function_score_median_syn <- median(final_filtered_df[which(final_filtered_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLxrLD_function_score_median_ns <- median(final_filtered_df[which(final_filtered_df$conseq == 'STOP_GAINED' & as.numeric(as.character(final_filtered_df$protPos)) >= 54 & as.numeric(as.character(final_filtered_df$protPos)) < 195),c('function_score')])

VHLx2rLD_final_df <- final_filtered_df[which(final_filtered_df$sge_region == "exon 2"),]
VHLx2rLD_function_score_median_syn <- median(VHLx2rLD_final_df[which(VHLx2rLD_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx2rLD_function_score_median_ns <- median(VHLx2rLD_final_df[which(VHLx2rLD_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx2rLD_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx2rLD_final_df$protPos)) < 195),c('function_score')])
VHLx2rLD_final_df$function_score_sns <- (VHLx2rLD_final_df$function_score/(VHLx2rLD_function_score_median_syn-VHLx2rLD_function_score_median_ns))*(VHLxrLD_function_score_median_syn-VHLxrLD_function_score_median_ns)

VHLx3arLD_final_df <- final_filtered_df[which(final_filtered_df$sge_region == "exon 3a"),]
VHLx3arLD_function_score_median_syn <- median(VHLx3arLD_final_df[which(VHLx3arLD_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx3arLD_function_score_median_ns <- median(VHLx3arLD_final_df[which(VHLx3arLD_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx3arLD_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx3arLD_final_df$protPos)) < 195),c('function_score')])
VHLx3arLD_final_df$function_score_sns <- (VHLx3arLD_final_df$function_score/(VHLx3arLD_function_score_median_syn-VHLx3arLD_function_score_median_ns))*(VHLxrLD_function_score_median_syn-VHLxrLD_function_score_median_ns)

VHLx3brLD_final_df <- final_filtered_df[which(final_filtered_df$sge_region == "exon 3b"),]
VHLx3brLD_function_score_median_syn <- median(VHLx3brLD_final_df[which(VHLx3brLD_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx3brLD_function_score_median_ns <- median(VHLx3brLD_final_df[which(VHLx3brLD_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx3brLD_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx3brLD_final_df$protPos)) < 195),c('function_score')])
VHLx3brLD_final_df$function_score_sns <- (VHLx3brLD_final_df$function_score/(VHLx3brLD_function_score_median_syn-VHLx3brLD_function_score_median_ns))*(VHLxrLD_function_score_median_syn-VHLxrLD_function_score_median_ns)

VHLx1brLD_final_df <- final_filtered_df[which(final_filtered_df$sge_region == "exon 1b"),]
VHLx1brLD_function_score_median_syn <- median(VHLx1brLD_final_df[which(VHLx1brLD_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx1brLD_function_score_median_ns <- median(VHLx1brLD_final_df[which(VHLx1brLD_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx1brLD_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx1brLD_final_df$protPos)) < 195),c('function_score')])
VHLx1brLD_final_df$function_score_sns <- (VHLx1brLD_final_df$function_score/(VHLx1brLD_function_score_median_syn-VHLx1brLD_function_score_median_ns))*(VHLxrLD_function_score_median_syn-VHLxrLD_function_score_median_ns)

VHLx1arLD_final_df <- final_filtered_df[which(final_filtered_df$sge_region == "exon 1a"),]
VHLx1arLD_function_score_median_syn <- median(VHLx1arLD_final_df[which(VHLx1arLD_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx1arLD_function_score_median_ns <- median(VHLx1arLD_final_df[which(VHLx1arLD_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx1arLD_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx1arLD_final_df$protPos)) < 195),c('function_score')])
VHLx1arLD_final_df$function_score_sns <- (VHLx1arLD_final_df$function_score/(VHLx1arLD_function_score_median_syn-VHLx1arLD_function_score_median_ns))*(VHLxrLD_function_score_median_syn-VHLxrLD_function_score_median_ns)

VHLx1crLD_final_df <- final_filtered_df[which(final_filtered_df$sge_region == "exon 1c"),]
VHLx1crLD_final_df$function_score_sns <- VHLx1crLD_final_df$function_score

VHLx1prLD_final_df <- final_filtered_df[which(final_filtered_df$sge_region == "exon 1p"),]
VHLx1prLD_final_df$function_score_sns <- VHLx1prLD_final_df$function_score
```
Section on calculating significance of score on a per region basis, and on merging scores for variants represented in multiple libraries to yield a single score:

Scores to derive in all df's: final_function_score, rna_score, and post_rna_score, will be the same values for any regions not represented twice. Every "final df" needs to have the 3 scores added to merge back together correctly.

When only 1 score is kept decision made based on noise of each data set.

```{r}
#will keep same number of columns as region's "final" df, to facilitate score merging
VHLx1crLD_replace_df <- VHLx1crLD_final_df[,c("pos_alt","function_score_sns","tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm","tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm")]
VHLx1crLD_replace_df$function_score_final <- VHLx1crLD_replace_df$function_score_sns
VHLx1crLD_replace_df$rna_score <- log2(VHLx1crLD_replace_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm)
VHLx1crLD_replace_df$post_rna_score <- log2(VHLx1crLD_replace_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm)

VHLx1brLD_replace_df <- VHLx1brLD_final_df[,c("pos_alt","function_score_sns","tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm","tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm")]
VHLx1brLD_replace_df$function_score_final <- VHLx1brLD_replace_df$function_score_sns
VHLx1brLD_replace_df$rna_score <- log2(VHLx1brLD_replace_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm)
VHLx1brLD_replace_df$post_rna_score <- log2(VHLx1brLD_replace_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm)

VHLx1crLD_replace_df <- merge(VHLx1crLD_replace_df, VHLx1brLD_replace_df, by="pos_alt",all.x=TRUE)
VHLx1crLD_replace_df <- VHLx1crLD_replace_df %>% 
  mutate(function_score_final.x = case_when(
    !is.na(function_score_final.y) ~ function_score_final.y,
    is.na(function_score_final.y) ~ function_score_final.x
    ))
VHLx1crLD_replace_df <- VHLx1crLD_replace_df %>% 
  mutate(rna_score.x = case_when(
    !is.na(rna_score.y) ~ rna_score.y,
    is.na(rna_score.y) ~ rna_score.x
    ))
VHLx1crLD_replace_df <- VHLx1crLD_replace_df %>% 
  mutate(post_rna_score.x = case_when(
    !is.na(post_rna_score.y) ~ post_rna_score.y,
    is.na(post_rna_score.y) ~ post_rna_score.x
    ))

VHLx1crLD_replace_df <- VHLx1crLD_replace_df %>% 
  rename(
    function_score_final = function_score_final.x,
    rna_score = rna_score.x,
    post_rna_score = post_rna_score.x
    )

VHLx1crLD_final_df <- merge(VHLx1crLD_final_df, VHLx1crLD_replace_df[,c("pos_alt","function_score_final","rna_score","post_rna_score")], by="pos_alt",all.x=TRUE)

#end exon 1c overlapping filtering
#being exon 1a overlapping filtering, goal to average scores with exon 1b

#will keep same number of columns as region's "final" df, to facilitate score merging
VHLx1arLD_replace_df <- VHLx1arLD_final_df[,c("pos_alt","protPos","conseq","function_score_sns","tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm","tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm")]
VHLx1arLD_replace_df$function_score_final <- VHLx1arLD_replace_df$function_score_sns
VHLx1arLD_replace_df$rna_score <- log2(VHLx1arLD_replace_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm)
VHLx1arLD_replace_df$post_rna_score <- log2(VHLx1arLD_replace_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm)
#VHLx1brLD_replace_df (same as above)
VHLx1arLD_replace_df <- merge(VHLx1arLD_replace_df, VHLx1brLD_replace_df, by="pos_alt",all.x=TRUE)


#correlation between points represented in exon 1a and exon 1b
ggplot(VHLx1arLD_replace_df)+geom_point(aes(x=function_score_sns.x,y=function_score_sns.y,color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SGE Region 1b Function Score')+xlab('SGE Region 1a Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)
#Will average all points if present in both, nice correlation between replicates covering the same variants. Looked specifically at scores for R79 (PAM edit in x1A) -- scores similar between replicates; no evidence re-coded allele changes results

VHLx1arLD_replace_df <- VHLx1arLD_replace_df %>% 
  mutate(function_score_final.x = case_when(
    !is.na(function_score_final.y) ~ (function_score_final.y+function_score_final.x)/2,
    is.na(function_score_final.y) ~ function_score_final.x
    ))
VHLx1arLD_replace_df <- VHLx1arLD_replace_df %>% 
  mutate(rna_score.x = case_when(
    !is.na(rna_score.y) ~ (rna_score.y+rna_score.x)/2,
    is.na(rna_score.y) ~ rna_score.x
    ))
VHLx1arLD_replace_df <- VHLx1arLD_replace_df %>% 
  mutate(post_rna_score.x = case_when(
    !is.na(post_rna_score.y) ~ (post_rna_score.y+post_rna_score.x)/2,
    is.na(post_rna_score.y) ~ post_rna_score.x
    ))

#These x1a scores being merged into x1b data frame have already been averaged where possible, so do not average again, just replace if the .x version is present after merging
VHLx1brLD_replace_df <- merge(VHLx1brLD_replace_df, VHLx1arLD_replace_df[,c("pos_alt","function_score_final.x","rna_score.x","post_rna_score.x")], by="pos_alt",all.x=TRUE)

VHLx1brLD_replace_df <- VHLx1brLD_replace_df %>% 
  mutate(function_score_final = case_when(
    !is.na(function_score_final.x) ~ function_score_final.x,
    is.na(function_score_final.x) ~ function_score_final
    ))
VHLx1brLD_replace_df <- VHLx1brLD_replace_df %>% 
  mutate(rna_score = case_when(
    !is.na(rna_score.x) ~ rna_score.x,
    is.na(rna_score.x) ~ rna_score
    ))
VHLx1brLD_replace_df <- VHLx1brLD_replace_df %>% 
  mutate(post_rna_score = case_when(
    !is.na(post_rna_score.x) ~ post_rna_score.x,
    is.na(post_rna_score.x) ~ post_rna_score
    ))

#replace df for 1a has all final values in as "function_score_final.x" or "post_rna_score.x"
#replace df for 1b has all final values in as "function_score_final" or "post_rna_score"

VHLx1arLD_replace_df <- VHLx1arLD_replace_df %>% 
  rename(
    function_score_final = function_score_final.x,
    rna_score = rna_score.x,
    post_rna_score = post_rna_score.x
    )

VHLx1arLD_final_df <- merge(VHLx1arLD_final_df, VHLx1arLD_replace_df[,c("pos_alt","function_score_final","rna_score","post_rna_score")], by="pos_alt",all.x=TRUE)

VHLx1brLD_final_df <- merge(VHLx1brLD_final_df, VHLx1brLD_replace_df[,c("pos_alt","function_score_final","rna_score","post_rna_score")], by="pos_alt",all.x=TRUE)

###end exon 1 final df merging

###start exon 1p final df merging
VHLx1prLD_final_df$function_score_final <- VHLx1prLD_final_df$function_score_sns
VHLx1prLD_final_df$rna_score <- log2(VHLx1prLD_final_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm)
VHLx1prLD_final_df$post_rna_score <- log2(VHLx1prLD_final_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm)

###start exon 2 final df merging
VHLx2rLD_final_df$function_score_final <- VHLx2rLD_final_df$function_score_sns
VHLx2rLD_final_df$rna_score <- log2(VHLx2rLD_final_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm)
VHLx2rLD_final_df$post_rna_score <- log2(VHLx2rLD_final_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm)

###start exon 3 final df merging
VHLx3arLD_replace_df <- VHLx3arLD_final_df[,c("pos_alt","protPos","conseq","function_score_sns","tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm","tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm")]
VHLx3arLD_replace_df$function_score_final <- VHLx3arLD_replace_df$function_score_sns
VHLx3arLD_replace_df$rna_score <- log2(VHLx3arLD_replace_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm)
VHLx3arLD_replace_df$post_rna_score <- log2(VHLx3arLD_replace_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm)

VHLx3brLD_replace_df <- VHLx3brLD_final_df[,c("pos_alt","function_score_sns","tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm","tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm")]
VHLx3brLD_replace_df$function_score_final <- VHLx3brLD_replace_df$function_score_sns
VHLx3brLD_replace_df$rna_score <- log2(VHLx3brLD_replace_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm)
VHLx3brLD_replace_df$post_rna_score <- log2(VHLx3brLD_replace_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm)

VHLx3arLD_replace_df <- merge(VHLx3arLD_replace_df, VHLx3brLD_replace_df, by="pos_alt",all.x=TRUE)

#plan to average all shared points -- check for general concordance. Note, R182 has silent edit in x3a lib but only partially covered by x3b library
ggplot(VHLx3arLD_replace_df)+geom_point(aes(x=function_score_sns.x,y=function_score_sns.y,color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SGE Region 3b Function Score')+xlab('SGE Region 3a Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#replace all 3a points that overlap to be average scores
VHLx3arLD_replace_df <- VHLx3arLD_replace_df %>% 
  mutate(function_score_final.x = case_when(
    !is.na(function_score_final.y) ~ (function_score_final.y+function_score_final.x)/2,
    is.na(function_score_final.y) ~ function_score_final.x
    ))
VHLx3arLD_replace_df <- VHLx3arLD_replace_df %>% 
  mutate(rna_score.x = case_when(
    !is.na(rna_score.y) ~ (rna_score.y+rna_score.x)/2,
    is.na(rna_score.y) ~ rna_score.x
    ))
VHLx3arLD_replace_df <- VHLx3arLD_replace_df %>% 
  mutate(post_rna_score.x = case_when(
    !is.na(post_rna_score.y) ~ (post_rna_score.y+post_rna_score.x)/2,
    is.na(post_rna_score.y) ~ post_rna_score.x
    ))

#These x3a scores being merged into x3b data frame have already been averaged where possible, so do not average again, just replace if the .x version is present after merging
VHLx3brLD_replace_df <- merge(VHLx3brLD_replace_df, VHLx3arLD_replace_df[,c("pos_alt","function_score_final.x","rna_score.x","post_rna_score.x")], by="pos_alt",all.x=TRUE)

VHLx3brLD_replace_df <- VHLx3brLD_replace_df %>% 
  mutate(function_score_final = case_when(
    !is.na(function_score_final.x)  ~ function_score_final.x,
    is.na(function_score_final.x) ~ function_score_final
    ))
VHLx3brLD_replace_df <- VHLx3brLD_replace_df %>% 
  mutate(rna_score = case_when(
    !is.na(rna_score.x) ~ rna_score.x,
    is.na(rna_score.x) ~ rna_score
    ))
VHLx3brLD_replace_df <- VHLx3brLD_replace_df %>% 
  mutate(post_rna_score = case_when(
    !is.na(post_rna_score.x) ~ post_rna_score.x,
    is.na(post_rna_score.x) ~ post_rna_score
    ))

#VHLx3arLD_replace_df has all final values in as "function_score_final.x" or "post_rna_score.x"
#VHLx3brLD_replace_df has all final values in as "function_score_final" or "post_rna_score"

VHLx3arLD_replace_df <- VHLx3arLD_replace_df %>% 
  rename(
    function_score_final = function_score_final.x,
    rna_score = rna_score.x,
    post_rna_score = post_rna_score.x
    )

VHLx3arLD_final_df <- merge(VHLx3arLD_final_df, VHLx3arLD_replace_df[,c("pos_alt","function_score_final","rna_score","post_rna_score")], by="pos_alt",all.x=TRUE)

VHLx3brLD_final_df <- merge(VHLx3brLD_final_df, VHLx3brLD_replace_df[,c("pos_alt","function_score_final","rna_score","post_rna_score")], by="pos_alt",all.x=TRUE)

###Calculate per exon significance using null distribution of syn variants with pnorm and BH adjustment
VHLx1crLD_null_fss <- VHLx1crLD_final_df[which(VHLx1crLD_final_df$conseq == "SYNONYMOUS"),]$function_score_sns 
VHLx1crLD_null_mean <- mean(VHLx1crLD_null_fss)
VHLx1crLD_null_sd <- sd(VHLx1crLD_null_fss)
VHLx1crLD_final_df$pvalues <- pnorm(VHLx1crLD_final_df$function_score_sns, VHLx1crLD_null_mean, sd=VHLx1crLD_null_sd) 
VHLx1crLD_final_df$fdr <- p.adjust(VHLx1crLD_final_df$pvalues, method = "BH")
VHLx1crLD_final_df$fs_sig <- "FDR NA"
VHLx1crLD_final_df[which(VHLx1crLD_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx1crLD_final_df[which(VHLx1crLD_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx1crLD_final_df[which(VHLx1crLD_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx1crLD_final_df[which(VHLx1crLD_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx1crLD_final_df[which(VHLx1crLD_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx1crLD_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())
max(VHLx1crLD_final_df[which(VHLx1crLD_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx1crLD_final_df[which(VHLx1crLD_final_df$fdr > 0.01),]$function_score_sns)

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx1brLD_null_fss <- VHLx1brLD_final_df[which(VHLx1brLD_final_df$conseq == "SYNONYMOUS" & VHLx1brLD_final_df$rna_score >-1),]$function_score_sns 
VHLx1brLD_null_mean <- mean(VHLx1brLD_null_fss)
VHLx1brLD_null_sd <- sd(VHLx1brLD_null_fss)
VHLx1brLD_final_df$pvalues <- pnorm(VHLx1brLD_final_df$function_score_sns, VHLx1brLD_null_mean, sd=VHLx1brLD_null_sd) 
VHLx1brLD_final_df$fdr <- p.adjust(VHLx1brLD_final_df$pvalues, method = "BH")
VHLx1brLD_final_df$fs_sig <- "FDR NA"
VHLx1brLD_final_df[which(VHLx1brLD_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx1brLD_final_df[which(VHLx1brLD_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx1brLD_final_df[which(VHLx1brLD_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx1brLD_final_df[which(VHLx1brLD_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx1brLD_final_df[which(VHLx1brLD_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx1brLD_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())
max(VHLx1brLD_final_df[which(VHLx1brLD_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx1brLD_final_df[which(VHLx1brLD_final_df$fdr > 0.01),]$function_score_sns)

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx1arLD_null_fss <- VHLx1arLD_final_df[which(VHLx1arLD_final_df$conseq == "SYNONYMOUS" & VHLx1arLD_final_df$rna_score >-1),]$function_score_sns 
VHLx1arLD_null_mean <- mean(VHLx1arLD_null_fss)
VHLx1arLD_null_sd <- sd(VHLx1arLD_null_fss)
VHLx1arLD_final_df$pvalues <- pnorm(VHLx1arLD_final_df$function_score_sns, VHLx1arLD_null_mean, sd=VHLx1arLD_null_sd) 
VHLx1arLD_final_df$fdr <- p.adjust(VHLx1arLD_final_df$pvalues, method = "BH")
VHLx1arLD_final_df$fs_sig <- "FDR NA"
VHLx1arLD_final_df[which(VHLx1arLD_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx1arLD_final_df[which(VHLx1arLD_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx1arLD_final_df[which(VHLx1arLD_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx1arLD_final_df[which(VHLx1arLD_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx1arLD_final_df[which(VHLx1arLD_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx1arLD_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())
max(VHLx1arLD_final_df[which(VHLx1arLD_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx1arLD_final_df[which(VHLx1arLD_final_df$fdr > 0.01),]$function_score_sns)

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx1prLD_null_fss <- VHLx1prLD_final_df[which(VHLx1prLD_final_df$conseq != "SYNONYMOUS"),]$function_score_sns 
VHLx1prLD_null_mean <- mean(VHLx1prLD_null_fss)
VHLx1prLD_null_sd <- sd(VHLx1prLD_null_fss)
VHLx1prLD_final_df$pvalues <- pnorm(VHLx1prLD_final_df$function_score_sns, VHLx1prLD_null_mean, sd=VHLx1prLD_null_sd) 
VHLx1prLD_final_df$fdr <- p.adjust(VHLx1prLD_final_df$pvalues, method = "BH")
VHLx1prLD_final_df$fs_sig <- "FDR NA"
VHLx1prLD_final_df[which(VHLx1prLD_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx1prLD_final_df[which(VHLx1prLD_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx1prLD_final_df[which(VHLx1prLD_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx1prLD_final_df[which(VHLx1prLD_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx1prLD_final_df[which(VHLx1prLD_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx1prLD_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())
max(VHLx1prLD_final_df[which(VHLx1prLD_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx1prLD_final_df[which(VHLx1prLD_final_df$fdr > 0.01),]$function_score_sns)

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx2rLD_null_fss <- VHLx2rLD_final_df[which(VHLx2rLD_final_df$conseq == "SYNONYMOUS" & VHLx2rLD_final_df$rna_score >-1),]$function_score_sns 
VHLx2rLD_null_mean <- mean(VHLx2rLD_null_fss)
VHLx2rLD_null_sd <- sd(VHLx2rLD_null_fss)
VHLx2rLD_final_df$pvalues <- pnorm(VHLx2rLD_final_df$function_score_sns, VHLx2rLD_null_mean, sd=VHLx2rLD_null_sd) 
VHLx2rLD_final_df$fdr <- p.adjust(VHLx2rLD_final_df$pvalues, method = "BH")
VHLx2rLD_final_df$fs_sig <- "FDR NA"
VHLx2rLD_final_df[which(VHLx2rLD_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx2rLD_final_df[which(VHLx2rLD_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx2rLD_final_df[which(VHLx2rLD_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx2rLD_final_df[which(VHLx2rLD_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx2rLD_final_df[which(VHLx2rLD_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx2rLD_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())
max(VHLx2rLD_final_df[which(VHLx2rLD_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx2rLD_final_df[which(VHLx2rLD_final_df$fdr > 0.01),]$function_score_sns)

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx3arLD_null_fss <- VHLx3arLD_final_df[which(VHLx3arLD_final_df$conseq == "SYNONYMOUS" & VHLx3arLD_final_df$rna_score >-1),]$function_score_sns 
VHLx3arLD_null_mean <- mean(VHLx3arLD_null_fss)
VHLx3arLD_null_sd <- sd(VHLx3arLD_null_fss)
VHLx3arLD_final_df$pvalues <- pnorm(VHLx3arLD_final_df$function_score_sns, VHLx3arLD_null_mean, sd=VHLx3arLD_null_sd) 
VHLx3arLD_final_df$fdr <- p.adjust(VHLx3arLD_final_df$pvalues, method = "BH")
VHLx3arLD_final_df$fs_sig <- "FDR NA"
VHLx3arLD_final_df[which(VHLx3arLD_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx3arLD_final_df[which(VHLx3arLD_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx3arLD_final_df[which(VHLx3arLD_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx3arLD_final_df[which(VHLx3arLD_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx3arLD_final_df[which(VHLx3arLD_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"


ggplot(VHLx3arLD_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())
max(VHLx3arLD_final_df[which(VHLx3arLD_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx3arLD_final_df[which(VHLx3arLD_final_df$fdr > 0.01),]$function_score_sns)

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx3brLD_null_fss <- VHLx3brLD_final_df[which(VHLx3brLD_final_df$conseq == "SYNONYMOUS" & VHLx3brLD_final_df$rna_score >-1),]$function_score_sns 
VHLx3brLD_null_mean <- mean(VHLx3brLD_null_fss)
VHLx3brLD_null_sd <- sd(VHLx3brLD_null_fss)
VHLx3brLD_final_df$pvalues <- pnorm(VHLx3brLD_final_df$function_score_sns, VHLx3brLD_null_mean, sd=VHLx3brLD_null_sd) 
VHLx3brLD_final_df$fdr <- p.adjust(VHLx3brLD_final_df$pvalues, method = "BH")
VHLx3brLD_final_df$fs_sig <- "FDR NA"
VHLx3brLD_final_df[which(VHLx3brLD_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx3brLD_final_df[which(VHLx3brLD_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx3brLD_final_df[which(VHLx3brLD_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx3brLD_final_df[which(VHLx3brLD_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx3brLD_final_df[which(VHLx3brLD_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx3brLD_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())
max(VHLx3brLD_final_df[which(VHLx3brLD_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx3brLD_final_df[which(VHLx3brLD_final_df$fdr > 0.01),]$function_score_sns)


# bind all the "final" dataframes back together
final_filtered_df <- rbind(VHLx1brLD_final_df,VHLx1crLD_final_df[which(VHLx1crLD_final_df$function_score_final == VHLx1crLD_final_df$function_score_sns),],VHLx1arLD_final_df[which(VHLx1arLD_final_df$function_score_final == VHLx1arLD_final_df$function_score_sns),],VHLx1prLD_final_df,VHLx2rLD_final_df,VHLx3brLD_final_df,VHLx3arLD_final_df[which(VHLx3arLD_final_df$function_score_final == VHLx3arLD_final_df$function_score_sns),])
```

Applying same filtering, normalization steps to non-DAB replicates (rL4)
```{r}
VHLx2rL4_tHDR_pos_merge_df$hdr_cut <- VHLx2rL4_tHDR_pos_merge_df$hdr5
VHLx3arL4_tHDR_pos_merge_df$hdr_cut <- VHLx3arL4_tHDR_pos_merge_df$hdr5
VHLx3brL4_tHDR_pos_merge_df$hdr_cut <- VHLx3brL4_tHDR_pos_merge_df$hdr5
VHLx1brL4_tHDR_pos_merge_df$hdr_cut <- VHLx1brL4_tHDR_pos_merge_df$hdr5
VHLx1crL4_tHDR_pos_merge_df$hdr_cut <- VHLx1crL4_tHDR_pos_merge_df$hdr5
VHLx1arL4_tHDR_pos_merge_df$hdr_cut <- VHLx1arL4_tHDR_pos_merge_df$hdr5
VHLx1prL4_tHDR_pos_merge_df$hdr_cut <- VHLx1prL4_tHDR_pos_merge_df$hdr5

VHLx2rL4_tHDR_pos_merge_df$sge_region <- "exon 2"
VHLx3arL4_tHDR_pos_merge_df$sge_region <- "exon 3a"
VHLx3brL4_tHDR_pos_merge_df$sge_region <- "exon 3b"
VHLx1brL4_tHDR_pos_merge_df$sge_region <- "exon 1b"
VHLx1arL4_tHDR_pos_merge_df$sge_region <- "exon 1a"
VHLx1crL4_tHDR_pos_merge_df$sge_region <- "exon 1c"
VHLx1prL4_tHDR_pos_merge_df$sge_region <- "exon 1p"

#binding all of the merge df's together...
VHLxrL4_all_master_df <- rbind(VHLx1crL4_tHDR_pos_merge_df,VHLx1brL4_tHDR_pos_merge_df,VHLx1arL4_tHDR_pos_merge_df,VHLx1prL4_tHDR_pos_merge_df,VHLx2rL4_tHDR_pos_merge_df,VHLx3arL4_tHDR_pos_merge_df,VHLx3brL4_tHDR_pos_merge_df)

VHLxrL4_all_master_df$pos_alt <- paste(VHLxrL4_all_master_df$pos, VHLxrL4_all_master_df$alt, sep='')
VHLxrL4_all_master_df$function_score <- log2(VHLxrL4_all_master_df$tHDR_post_pre_ratio_rL41rL42_mean_synnorm)

#calculate a post / post2 ratio, for rL41, rL42, and mean, normalise to synonymous
VHLxrL4_all_master_df$tHDR_post_post2_ratio <- VHLxrL4_all_master_df$tHDR_post_pseudo_freq/VHLxrL4_all_master_df$tHDR_post2_pseudo_freq
VHLxrL4_all_master_df$rL42_tHDR_post_post2_ratio <- VHLxrL4_all_master_df$rL42_tHDR_post_pseudo_freq/VHLxrL4_all_master_df$rL42_tHDR_post2_pseudo_freq
VHLxrL4_all_master_df$tHDR_post_post2_ratio_rL41rL42_mean <- 2^(log2(VHLxrL4_all_master_df$tHDR_post_pseudo_freq/VHLxrL4_all_master_df$tHDR_post2_pseudo_freq)+log2(VHLxrL4_all_master_df$rL42_tHDR_post_pseudo_freq/VHLxrL4_all_master_df$rL42_tHDR_post2_pseudo_freq)/2)
VHLxrL4_all_tHDR_post_post2_ratio_rL41rL42_med_syn <- median(VHLxrL4_all_master_df[which(VHLxrL4_all_master_df$conseq == 'SYNONYMOUS'),]$tHDR_post_post2_ratio_rL41rL42_mean)

VHLxrL4_all_master_df$tHDR_post_post2_ratio_rL41rL42_mean_synnorm <- VHLxrL4_all_master_df$tHDR_post_post2_ratio_rL41rL42_mean/VHLxrL4_all_tHDR_post_post2_ratio_rL41rL42_med_syn


VHLxrL4_all_master_df <- VHLxrL4_all_master_df %>% 
  mutate(pHGVS = case_when(
    !is.na(Intron) ~ NA_character_,
    conseq == "5PRIME_UTR" ~ NA_character_,
    !is.na(Exon) ~ paste0("p.", oAA, protPos, nAA)
  ))

VHLxrL4_all_master_df <- VHLxrL4_all_master_df %>% 
  mutate(cHGVS = case_when(
    (is.na(CDSpos)&(conseq == "5PRIME_UTR")) ~ paste0("c.-",10183532-as.numeric(as.character(pos)), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 1a")) ~ paste0("c.340+", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 1p")) ~ paste0("c.340+", abs(as.numeric(as.character(pos))-10183871), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 2")) ~ paste0("c.341-", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "2/2")&(sge_region == "exon 2")) ~ paste0("c.463+", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "2/2")&(sge_region == "exon 3a")) ~ paste0("c.464-", abs(as.numeric(as.character(pos))-10191471), Ref,">",alt),
    (is.na(CDSpos)&(conseq == "3PRIME_UTR")) ~ paste0("c.*",as.numeric(as.character(pos))-10191649, Ref,">",alt),
    !is.na(CDSpos) ~ paste0("c.", CDSpos, Ref,">",alt)
  ))

VHLxrL4_all_master_df$hg38_pos <- as.numeric(as.character(VHLxrL4_all_master_df$pos))-41684
VHLxrL4_all_master_df$hg38_pos_alt <- paste0(VHLxrL4_all_master_df$hg38_pos,VHLxrL4_all_master_df$alt)

#For saving work in R here.
write.table(VHLxrL4_all_master_df, "/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLxrL4_all_master_20230412.txt", sep="\t")
#VHLxrL4_all_master_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLxrL4_all_master_20230412.txt", sep="\t", header=TRUE)

#threshold for replicate agreement #removes 5 points with scores >2 different
rL4_lib_freq_thresh <- which(VHLxrL4_all_master_df$lib_pseudo_freq >= 0.0001)
rL4_pre_freq_thresh <- which(VHLxrL4_all_master_df$tHDR_pre_pseudo_freq >= 0.00001 & VHLxrL4_all_master_df$rL42_tHDR_pre_pseudo_freq >= 0.00001)

#Filter for discordancy between replicates set at difference of 2 or more in log2 ratios AND scoring above -1 in at least 1 replicate (i.e. won't filter our variants scoring at -2, -4, as there is more noise in low end of measurements)
#rL4_repl_conc_thresh <- which((abs(log2(VHLxrL4_all_master_df$tHDR_post_pre_ratio_synnorm)-log2(VHLxrL4_all_master_df$rL42_tHDR_post_pre_ratio_synnorm))<1.5)|(log2(VHLxrL4_all_master_df$tHDR_post_pre_ratio_synnorm) <= -1 & log2(VHLxrL4_all_master_df$rL42_tHDR_post_pre_ratio_synnorm) <= -1) |(log2(VHLxrL4_all_master_df$tHDR_post_pre_ratio_synnorm) >= 0 & log2(VHLxrL4_all_master_df$rL42_tHDR_post_pre_ratio_synnorm) >= 0))

#rL4_post2_conc_thresh <- which((abs(log2(VHLxrL4_all_master_df$tHDR_post_post2_ratio_rL41rL42_mean_synnorm)-log2(VHLxrL4_all_master_df$tHDR_post2_pre_ratio_rL41rL42_mean_synnorm))<2)|(log2(VHLxrL4_all_master_df$tHDR_post_post2_ratio_rL41rL42_mean_synnorm) <= -0.5 & log2(VHLxrL4_all_master_df$tHDR_post2_pre_ratio_rL41rL42_mean_synnorm) <= -0.5))

#Filter removing variants with high predicted rate of reads caused by sequencing error -- note, this only acts on cHDR reads, and doesn't include positions for which no error rate could be calculated due to overlapping PAM edit (these get left in)

rL4_seq_error_thresh <- which((VHLxrL4_all_master_df$pre-VHLxrL4_all_master_df$corr_pre)/VHLxrL4_all_master_df$pre < 0.5 | (VHLxrL4_all_master_df$rL42_pre-VHLxrL4_all_master_df$rL42_corr_pre)/VHLxrL4_all_master_df$rL42_pre <0.5 | is.na((VHLxrL4_all_master_df$pre-VHLxrL4_all_master_df$corr_pre)/VHLxrL4_all_master_df$pre) | is.na((VHLxrL4_all_master_df$rL42_pre-VHLxrL4_all_master_df$rL42_corr_pre)/VHLxrL4_all_master_df$rL42_pre))

rL4_not_1c_thresh <- which((VHLxrL4_all_master_df$sge_region != "exon 1c"))
rL4_not_1a_thresh <- which((VHLxrL4_all_master_df$sge_region != "exon 1a"))

rL4_filter_combo_1 <- intersect(rL4_lib_freq_thresh,rL4_pre_freq_thresh)
rL4_filter_combo_5 <- intersect(rL4_filter_combo_1,rL4_not_1c_thresh)
rL4_filter_combo_6 <- intersect(rL4_filter_combo_5,rL4_not_1a_thresh)


#no filter
ggplot(VHLxrL4_all_master_df[,], aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rL42_tHDR_post_pre_ratio_synnorm),color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('tHDR post/pre repl. 2, log-2')+xlab('tHDR post/pre repl. 1, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

ggplot(VHLxrL4_all_master_df[rL4_filter_combo_6,], aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rL42_tHDR_post_pre_ratio_synnorm),color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('tHDR post/pre repl. 2, log-2')+xlab('tHDR post/pre repl. 1, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#new dataframe   Reduce(intersect, list(rL4_lib_freq_thresh,repl_conc_thresh,seq_error_thresh))
rL4_final_filtered_df <- VHLxrL4_all_master_df[rL4_filter_combo_6,]
rL4_final_filtered_df <- rL4_final_filtered_df[which(is.na(rL4_final_filtered_df$cHGVS)==FALSE),]

write.table(rL4_final_filtered_df, "/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/rL4_final_filtered_df_20230412.txt", sep="\t")
#rL4_final_filtered_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/rL4_final_filtered_df_20230412.txt", sep="\t", header=TRUE)


#### START SYN NS global scaling of filtered data
#for rL4 data, scaling done for x1b, x2, x3a, x3b, x1p
VHLxrL4_function_score_median_syn <- median(rL4_final_filtered_df[which(rL4_final_filtered_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLxrL4_function_score_median_ns <- median(rL4_final_filtered_df[which(rL4_final_filtered_df$conseq == 'STOP_GAINED' & as.numeric(as.character(rL4_final_filtered_df$protPos)) >= 54 & as.numeric(as.character(rL4_final_filtered_df$protPos)) < 195),c('function_score')])

VHLx2rL4_final_df <- rL4_final_filtered_df[which(rL4_final_filtered_df$sge_region == "exon 2"),]
VHLx2rL4_function_score_median_syn <- median(VHLx2rL4_final_df[which(VHLx2rL4_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx2rL4_function_score_median_ns <- median(VHLx2rL4_final_df[which(VHLx2rL4_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx2rL4_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx2rL4_final_df$protPos)) < 195),c('function_score')])
VHLx2rL4_final_df$function_score_sns <- (VHLx2rL4_final_df$function_score/(VHLx2rL4_function_score_median_syn-VHLx2rL4_function_score_median_ns))*(VHLxrL4_function_score_median_syn-VHLxrL4_function_score_median_ns)

VHLx3arL4_final_df <- rL4_final_filtered_df[which(rL4_final_filtered_df$sge_region == "exon 3a"),]
VHLx3arL4_function_score_median_syn <- median(VHLx3arL4_final_df[which(VHLx3arL4_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx3arL4_function_score_median_ns <- median(VHLx3arL4_final_df[which(VHLx3arL4_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx3arL4_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx3arL4_final_df$protPos)) < 195),c('function_score')])
VHLx3arL4_final_df$function_score_sns <- (VHLx3arL4_final_df$function_score/(VHLx3arL4_function_score_median_syn-VHLx3arL4_function_score_median_ns))*(VHLxrL4_function_score_median_syn-VHLxrL4_function_score_median_ns)

VHLx3brL4_final_df <- rL4_final_filtered_df[which(rL4_final_filtered_df$sge_region == "exon 3b"),]
VHLx3brL4_function_score_median_syn <- median(VHLx3brL4_final_df[which(VHLx3brL4_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx3brL4_function_score_median_ns <- median(VHLx3brL4_final_df[which(VHLx3brL4_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx3brL4_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx3brL4_final_df$protPos)) < 195),c('function_score')])
VHLx3brL4_final_df$function_score_sns <- (VHLx3brL4_final_df$function_score/(VHLx3brL4_function_score_median_syn-VHLx3brL4_function_score_median_ns))*(VHLxrL4_function_score_median_syn-VHLxrL4_function_score_median_ns)

VHLx1brL4_final_df <- rL4_final_filtered_df[which(rL4_final_filtered_df$sge_region == "exon 1b"),]
VHLx1brL4_function_score_median_syn <- median(VHLx1brL4_final_df[which(VHLx1brL4_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx1brL4_function_score_median_ns <- median(VHLx1brL4_final_df[which(VHLx1brL4_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx1brL4_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx1brL4_final_df$protPos)) < 195),c('function_score')])
VHLx1brL4_final_df$function_score_sns <- (VHLx1brL4_final_df$function_score/(VHLx1brL4_function_score_median_syn-VHLx1brL4_function_score_median_ns))*(VHLxrL4_function_score_median_syn-VHLxrL4_function_score_median_ns)

VHLx1prL4_final_df <- rL4_final_filtered_df[which(rL4_final_filtered_df$sge_region == "exon 1p"),]
VHLx1prL4_final_df$function_score_sns <- VHLx1prL4_final_df$function_score
#### END SYN NS global scaling of filtered data

###merge scores for variants represented in multiple libraries to yield a single score:
##scores to derive in all df's: final_function_score, rna_score, and post_rna_score, will be the same values for any regions not represented twice. Every "final df" needs to have the 3 scores added to merge back together correctly.

VHLx1brL4_final_df$function_score_final <- VHLx1brL4_final_df$function_score_sns
VHLx1brL4_final_df$rna_score <- log2(VHLx1brL4_final_df$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)
VHLx1brL4_final_df$post_rna_score <- log2(VHLx1brL4_final_df$tHDR_rna2_post_ratio_rL41rL42_mean_synnorm)

###end exon 1 final df merging

###start exon 1p final df merging
VHLx1prL4_final_df$function_score_final <- VHLx1prL4_final_df$function_score_sns
VHLx1prL4_final_df$rna_score <- log2(VHLx1prL4_final_df$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)
VHLx1prL4_final_df$post_rna_score <- log2(VHLx1prL4_final_df$tHDR_rna2_post_ratio_rL41rL42_mean_synnorm)

###start exon 2 final df merging
VHLx2rL4_final_df$function_score_final <- VHLx2rL4_final_df$function_score_sns
VHLx2rL4_final_df$rna_score <- log2(VHLx2rL4_final_df$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)
VHLx2rL4_final_df$post_rna_score <- log2(VHLx2rL4_final_df$tHDR_rna2_post_ratio_rL41rL42_mean_synnorm)

###start exon 3 final df merging
VHLx3arL4_replace_df <- VHLx3arL4_final_df[,c("pos_alt","protPos","conseq","function_score_sns","tHDR_rna_pre_ratio_rL41rL42_mean_synnorm","tHDR_rna2_post_ratio_rL41rL42_mean_synnorm")]
VHLx3arL4_replace_df$function_score_final <- VHLx3arL4_replace_df$function_score_sns
VHLx3arL4_replace_df$rna_score <- log2(VHLx3arL4_replace_df$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)
VHLx3arL4_replace_df$post_rna_score <- log2(VHLx3arL4_replace_df$tHDR_rna2_post_ratio_rL41rL42_mean_synnorm)

VHLx3brL4_replace_df <- VHLx3brL4_final_df[,c("pos_alt","function_score_sns","tHDR_rna_pre_ratio_rL41rL42_mean_synnorm","tHDR_rna2_post_ratio_rL41rL42_mean_synnorm")]
VHLx3brL4_replace_df$function_score_final <- VHLx3brL4_replace_df$function_score_sns
VHLx3brL4_replace_df$rna_score <- log2(VHLx3brL4_replace_df$tHDR_rna_pre_ratio_rL41rL42_mean_synnorm)
VHLx3brL4_replace_df$post_rna_score <- log2(VHLx3brL4_replace_df$tHDR_rna2_post_ratio_rL41rL42_mean_synnorm)

VHLx3arL4_replace_df <- merge(VHLx3arL4_replace_df, VHLx3brL4_replace_df, by="pos_alt",all.x=TRUE)

#plan to average all shared points -- check for general concordance. Note, R182 has silent edit in x3a lib but only partially covered by x3b library
ggplot(VHLx3arL4_replace_df)+geom_point(aes(x=function_score_sns.x,y=function_score_sns.y,color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('SGE Region 3b Function Score')+xlab('SGE Region 3a Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#replace all 3a points that overlap to be average scores
VHLx3arL4_replace_df <- VHLx3arL4_replace_df %>% 
  mutate(function_score_final.x = case_when(
    !is.na(function_score_final.y) ~ (function_score_final.y+function_score_final.x)/2,
    is.na(function_score_final.y) ~ function_score_final.x
    ))
VHLx3arL4_replace_df <- VHLx3arL4_replace_df %>% 
  mutate(rna_score.x = case_when(
    !is.na(rna_score.y) ~ (rna_score.y+rna_score.x)/2,
    is.na(rna_score.y) ~ rna_score.x
    ))
VHLx3arL4_replace_df <- VHLx3arL4_replace_df %>% 
  mutate(post_rna_score.x = case_when(
    !is.na(post_rna_score.y) ~ (post_rna_score.y+post_rna_score.x)/2,
    is.na(post_rna_score.y) ~ post_rna_score.x
    ))

#These x3a scores being merged into x3b data frame have already been averaged where possible, so do not average again, just replace if the .x version is present after merging
VHLx3brL4_replace_df <- merge(VHLx3brL4_replace_df, VHLx3arL4_replace_df[,c("pos_alt","function_score_final.x","rna_score.x","post_rna_score.x")], by="pos_alt",all.x=TRUE)

VHLx3brL4_replace_df <- VHLx3brL4_replace_df %>% 
  mutate(function_score_final = case_when(
    !is.na(function_score_final.x)  ~ function_score_final.x,
    is.na(function_score_final.x) ~ function_score_final
    ))
VHLx3brL4_replace_df <- VHLx3brL4_replace_df %>% 
  mutate(rna_score = case_when(
    !is.na(rna_score.x) ~ rna_score.x,
    is.na(rna_score.x) ~ rna_score
    ))
VHLx3brL4_replace_df <- VHLx3brL4_replace_df %>% 
  mutate(post_rna_score = case_when(
    !is.na(post_rna_score.x) ~ post_rna_score.x,
    is.na(post_rna_score.x) ~ post_rna_score
    ))

#VHLx3arL4_replace_df has all final values in as "function_score_final.x" or "post_rna_score.x"
#VHLx3brL4_replace_df has all final values in as "function_score_final" or "post_rna_score"

VHLx3arL4_replace_df <- VHLx3arL4_replace_df %>% 
  rename(
    function_score_final = function_score_final.x,
    rna_score = rna_score.x,
    post_rna_score = post_rna_score.x
    )

VHLx3arL4_final_df <- merge(VHLx3arL4_final_df, VHLx3arL4_replace_df[,c("pos_alt","function_score_final","rna_score","post_rna_score")], by="pos_alt",all.x=TRUE)

VHLx3brL4_final_df <- merge(VHLx3brL4_final_df, VHLx3brL4_replace_df[,c("pos_alt","function_score_final","rna_score","post_rna_score")], by="pos_alt",all.x=TRUE)

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx1brL4_null_fss <- VHLx1brL4_final_df[which(VHLx1brL4_final_df$conseq == "SYNONYMOUS"),]$function_score_sns 
VHLx1brL4_null_mean <- mean(VHLx1brL4_null_fss)
VHLx1brL4_null_sd <- sd(VHLx1brL4_null_fss)
VHLx1brL4_final_df$pvalues <- pnorm(VHLx1brL4_final_df$function_score_sns, VHLx1brL4_null_mean, sd=VHLx1brL4_null_sd) 
VHLx1brL4_final_df$fdr <- p.adjust(VHLx1brL4_final_df$pvalues, method = "BH")
VHLx1brL4_final_df$fs_sig <- "FDR NA"
VHLx1brL4_final_df[which(VHLx1brL4_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx1brL4_final_df[which(VHLx1brL4_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx1brL4_final_df[which(VHLx1brL4_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx1brL4_final_df[which(VHLx1brL4_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx1brL4_final_df[which(VHLx1brL4_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx1brL4_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx1prL4_null_fss <- VHLx1prL4_final_df[which(VHLx1prL4_final_df$conseq != "SYNONYMOUS"),]$function_score_sns 
VHLx1prL4_null_mean <- mean(VHLx1prL4_null_fss)
VHLx1prL4_null_sd <- sd(VHLx1prL4_null_fss)
VHLx1prL4_final_df$pvalues <- pnorm(VHLx1prL4_final_df$function_score_sns, VHLx1prL4_null_mean, sd=VHLx1prL4_null_sd) 
VHLx1prL4_final_df$fdr <- p.adjust(VHLx1prL4_final_df$pvalues, method = "BH")
VHLx1prL4_final_df$fs_sig <- "FDR NA"
VHLx1prL4_final_df[which(VHLx1prL4_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx1prL4_final_df[which(VHLx1prL4_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx1prL4_final_df[which(VHLx1prL4_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx1prL4_final_df[which(VHLx1prL4_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx1prL4_final_df[which(VHLx1prL4_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx1prL4_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx2rL4_null_fss <- VHLx2rL4_final_df[which(VHLx2rL4_final_df$conseq == "SYNONYMOUS"),]$function_score_sns 
VHLx2rL4_null_mean <- mean(VHLx2rL4_null_fss)
VHLx2rL4_null_sd <- sd(VHLx2rL4_null_fss)
VHLx2rL4_final_df$pvalues <- pnorm(VHLx2rL4_final_df$function_score_sns, VHLx2rL4_null_mean, sd=VHLx2rL4_null_sd) 
VHLx2rL4_final_df$fdr <- p.adjust(VHLx2rL4_final_df$pvalues, method = "BH")
VHLx2rL4_final_df$fs_sig <- "FDR NA"
VHLx2rL4_final_df[which(VHLx2rL4_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx2rL4_final_df[which(VHLx2rL4_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx2rL4_final_df[which(VHLx2rL4_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx2rL4_final_df[which(VHLx2rL4_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx2rL4_final_df[which(VHLx2rL4_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx2rL4_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx3arL4_null_fss <- VHLx3arL4_final_df[which(VHLx3arL4_final_df$conseq == "SYNONYMOUS"),]$function_score_sns 
VHLx3arL4_null_mean <- mean(VHLx3arL4_null_fss)
VHLx3arL4_null_sd <- sd(VHLx3arL4_null_fss)
VHLx3arL4_final_df$pvalues <- pnorm(VHLx3arL4_final_df$function_score_sns, VHLx3arL4_null_mean, sd=VHLx3arL4_null_sd) 
VHLx3arL4_final_df$fdr <- p.adjust(VHLx3arL4_final_df$pvalues, method = "BH")
VHLx3arL4_final_df$fs_sig <- "FDR NA"
VHLx3arL4_final_df[which(VHLx3arL4_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx3arL4_final_df[which(VHLx3arL4_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx3arL4_final_df[which(VHLx3arL4_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx3arL4_final_df[which(VHLx3arL4_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx3arL4_final_df[which(VHLx3arL4_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx3arL4_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx3brL4_null_fss <- VHLx3brL4_final_df[which(VHLx3brL4_final_df$conseq == "SYNONYMOUS"),]$function_score_sns 
VHLx3brL4_null_mean <- mean(VHLx3brL4_null_fss)
VHLx3brL4_null_sd <- sd(VHLx3brL4_null_fss)
VHLx3brL4_final_df$pvalues <- pnorm(VHLx3brL4_final_df$function_score_sns, VHLx3brL4_null_mean, sd=VHLx3brL4_null_sd) 
VHLx3brL4_final_df$fdr <- p.adjust(VHLx3brL4_final_df$pvalues, method = "BH")
VHLx3brL4_final_df$fs_sig <- "FDR NA"
VHLx3brL4_final_df[which(VHLx3brL4_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx3brL4_final_df[which(VHLx3brL4_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx3brL4_final_df[which(VHLx3brL4_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx3brL4_final_df[which(VHLx3brL4_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx3brL4_final_df[which(VHLx3brL4_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx3brL4_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())

#save this if needed
rL4_final_filtered_df <- rbind(VHLx1brL4_final_df,VHLx1prL4_final_df,VHLx2rL4_final_df,VHLx3arL4_final_df,VHLx3brL4_final_df)

#Improvement using DAB? How many more variants were significantly depleted?
length(which(rL4_final_filtered_df$fdr < 0.01))
length(which(final_filtered_df$fdr < 0.01))
#and another version with only the final scores collapse
rL4_collpased_final_filtered_df <- rbind(VHLx1brL4_final_df,VHLx1prL4_final_df,VHLx2rL4_final_df,VHLx3brL4_final_df,VHLx3arL4_final_df[which(VHLx3arL4_final_df$function_score_final == VHLx3arL4_final_df$function_score_sns),])
```

Again performing filtering, normalizing, and significance testing, this time for HIF1A-KO experiments (sge regions: exon 2 and exon 3a )

```{r}
VHLx2rLH_tHDR_pos_merge_df$hdr_cut <- VHLx2rLH_tHDR_pos_merge_df$hdr5
VHLx3arLH_tHDR_pos_merge_df$hdr_cut <- VHLx3arLH_tHDR_pos_merge_df$hdr5

VHLx2rLH_tHDR_pos_merge_df$sge_region <- "exon 2"
VHLx3arLH_tHDR_pos_merge_df$sge_region <- "exon 3a"

#binding all of the merge df's together...
VHLxrLH_all_master_df <- rbind(VHLx2rLH_tHDR_pos_merge_df,VHLx3arLH_tHDR_pos_merge_df)
VHLxrLH_all_master_df$pos_alt <- paste(VHLxrLH_all_master_df$pos, VHLxrLH_all_master_df$alt, sep='')
VHLxrLH_all_master_df$function_score <- log2(VHLxrLH_all_master_df$tHDR_post_pre_ratio_rLH1rLH2_mean_synnorm)

#calculate a post / post2 ratio, for rLH1, rLH2, and mean, normalise to synonymous
VHLxrLH_all_master_df$tHDR_post_post2_ratio <- VHLxrLH_all_master_df$tHDR_post_pseudo_freq/VHLxrLH_all_master_df$tHDR_post2_pseudo_freq
VHLxrLH_all_master_df$rLH2_tHDR_post_post2_ratio <- VHLxrLH_all_master_df$rLH2_tHDR_post_pseudo_freq/VHLxrLH_all_master_df$rLH2_tHDR_post2_pseudo_freq
VHLxrLH_all_master_df$tHDR_post_post2_ratio_rLH1rLH2_mean <- 2^(log2(VHLxrLH_all_master_df$tHDR_post_pseudo_freq/VHLxrLH_all_master_df$tHDR_post2_pseudo_freq)+log2(VHLxrLH_all_master_df$rLH2_tHDR_post_pseudo_freq/VHLxrLH_all_master_df$rLH2_tHDR_post2_pseudo_freq)/2)
VHLxrLH_all_tHDR_post_post2_ratio_rLH1rLH2_med_syn <- median(VHLxrLH_all_master_df[which(VHLxrLH_all_master_df$conseq == 'SYNONYMOUS'),]$tHDR_post_post2_ratio_rLH1rLH2_mean)

VHLxrLH_all_master_df$tHDR_post_post2_ratio_rLH1rLH2_mean_synnorm <- VHLxrLH_all_master_df$tHDR_post_post2_ratio_rLH1rLH2_mean/VHLxrLH_all_tHDR_post_post2_ratio_rLH1rLH2_med_syn


VHLxrLH_all_master_df <- VHLxrLH_all_master_df %>% 
  mutate(pHGVS = case_when(
    !is.na(Intron) ~ NA_character_,
    conseq == "5PRIME_UTR" ~ NA_character_,
    !is.na(Exon) ~ paste0("p.", oAA, protPos, nAA)
  ))

VHLxrLH_all_master_df <- VHLxrLH_all_master_df %>% 
  mutate(cHGVS = case_when(
    (is.na(CDSpos)&(conseq == "5PRIME_UTR")) ~ paste0("c.-",10183532-as.numeric(as.character(pos)), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 1a")) ~ paste0("c.340+", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 1p")) ~ paste0("c.340+", abs(as.numeric(as.character(pos))-10183871), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "1/2")&(sge_region == "exon 2")) ~ paste0("c.341-", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "2/2")&(sge_region == "exon 2")) ~ paste0("c.463+", abs(as.numeric(as.character(Dst2Splice))), Ref,">",alt),
    (is.na(CDSpos)&(Intron == "2/2")&(sge_region == "exon 3a")) ~ paste0("c.464-", abs(as.numeric(as.character(pos))-10191471), Ref,">",alt),
    (is.na(CDSpos)&(conseq == "3PRIME_UTR")) ~ paste0("c.*",as.numeric(as.character(pos))-10191649, Ref,">",alt),
    !is.na(CDSpos) ~ paste0("c.", CDSpos, Ref,">",alt)
  ))

VHLxrLH_all_master_df$hg38_pos <- as.numeric(as.character(VHLxrLH_all_master_df$pos))-41684
VHLxrLH_all_master_df$hg38_pos_alt <- paste0(VHLxrLH_all_master_df$hg38_pos,VHLxrLH_all_master_df$alt)

#For saving work here
write.table(VHLxrLH_all_master_df, "/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLxrLH_all_master_20230412.txt", sep="\t")
#VHLxrLH_all_master_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLxrLH_all_master_20230412.txt", sep="\t", header=TRUE)


#threshold for replicate agreement 
rLH_lib_freq_thresh <- which(VHLxrLH_all_master_df$lib_pseudo_freq >= 0.0001)
rLH_pre_freq_thresh <- which(VHLxrLH_all_master_df$tHDR_pre_pseudo_freq >= 0.00001 & VHLxrLH_all_master_df$rLH2_tHDR_pre_pseudo_freq >= 0.00001)

#Filter for discordancy between replicates set at difference of 2 or more in log2 ratios AND scoring above -1 in at least 1 replicate (i.e. won't filter our variants scoring at -2, -4, as there is more noise in low end of measurements)
rLH_repl_conc_thresh <- which((abs(log2(VHLxrLH_all_master_df$tHDR_post_pre_ratio_synnorm)-log2(VHLxrLH_all_master_df$rLH2_tHDR_post_pre_ratio_synnorm))<1.5)|(log2(VHLxrLH_all_master_df$tHDR_post_pre_ratio_synnorm) <= -1 & log2(VHLxrLH_all_master_df$rLH2_tHDR_post_pre_ratio_synnorm) <= -1) |(log2(VHLxrLH_all_master_df$tHDR_post_pre_ratio_synnorm) >= 0 & log2(VHLxrLH_all_master_df$rLH2_tHDR_post_pre_ratio_synnorm) >= 0))

rLH_post2_conc_thresh <- which((abs(log2(VHLxrLH_all_master_df$tHDR_post_post2_ratio_rLH1rLH2_mean_synnorm)-log2(VHLxrLH_all_master_df$tHDR_post2_pre_ratio_rLH1rLH2_mean_synnorm))<2)|(log2(VHLxrLH_all_master_df$tHDR_post_post2_ratio_rLH1rLH2_mean_synnorm) <= -0.5 & log2(VHLxrLH_all_master_df$tHDR_post2_pre_ratio_rLH1rLH2_mean_synnorm) <= -0.5))

#Filter removing variants with high predicted rate of reads caused by sequencing error -- note, this only acts on cHDR reads, and doesn't include positions for which no error rate could be calculated due to overlapping PAM edit (these get left in)
rLH_seq_error_thresh <- which((VHLxrLH_all_master_df$pre-VHLxrLH_all_master_df$corr_pre)/VHLxrLH_all_master_df$pre < 0.5 | (VHLxrLH_all_master_df$rLH2_pre-VHLxrLH_all_master_df$rLH2_corr_pre)/VHLxrLH_all_master_df$rLH2_pre <0.5 | is.na((VHLxrLH_all_master_df$pre-VHLxrLH_all_master_df$corr_pre)/VHLxrLH_all_master_df$pre) | is.na((VHLxrLH_all_master_df$rLH2_pre-VHLxrLH_all_master_df$rLH2_corr_pre)/VHLxrLH_all_master_df$rLH2_pre))

rLH_not_1c_thresh <- which((VHLxrLH_all_master_df$sge_region != "exon 1c"))

rLH_filter_combo_1 <- intersect(rLH_lib_freq_thresh,rLH_pre_freq_thresh)
rLH_filter_combo_2 <- intersect(rLH_filter_combo_1,rLH_repl_conc_thresh)
rLH_filter_combo_3a <- intersect(rLH_filter_combo_2,rLH_post2_conc_thresh)
rLH_filter_combo_3b <- intersect(rLH_filter_combo_2,rLH_seq_error_thresh)
rLH_filter_combo_4 <- intersect(rLH_filter_combo_3a,rLH_seq_error_thresh)
rLH_filter_combo_5 <- intersect(rLH_filter_combo_4,rLH_not_1c_thresh)

#no filter
ggplot(VHLxrLH_all_master_df[,], aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rLH2_tHDR_post_pre_ratio_synnorm),color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('tHDR post/pre repl. 2, log-2')+xlab('tHDR post/pre repl. 1, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#filter set 1
ggplot(VHLxrLH_all_master_df[rLH_filter_combo_1,], aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rLH2_tHDR_post_pre_ratio_synnorm),color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('tHDR post/pre repl. 2, log-2')+xlab('tHDR post/pre repl. 1, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#filter set 4
ggplot(VHLxrLH_all_master_df[rLH_filter_combo_4,], aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rLH2_tHDR_post_pre_ratio_synnorm),color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('tHDR post/pre repl. 2, log-2')+xlab('tHDR post/pre repl. 1, log-2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)

#new dataframe   Reduce(intersect, list(lib_freq_thresh,repl_conc_thresh,seq_error_thresh))
rLH_final_filtered_df <- VHLxrLH_all_master_df[rLH_filter_combo_4,]
rLH_final_filtered_df <- rLH_final_filtered_df[which(is.na(final_filtered_df$cHGVS)==FALSE),]

write.table(rLH_final_filtered_df, "/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLrLH_final_filtered_df_20230412.txt", sep="\t")
#rLH_final_filtered_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHLrLH_final_filtered_df_20230412.txt", sep="\t", header=TRUE)


#### SYN to NS global scaling of filtered data changed to SYN only

#do this first for the repl. mean fits and scores :
VHLxrLH_function_score_median_syn <- median(rLH_final_filtered_df[which(rLH_final_filtered_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLxrLH_function_score_median_ns <- median(rLH_final_filtered_df[which(rLH_final_filtered_df$conseq == 'STOP_GAINED' & as.numeric(as.character(rLH_final_filtered_df$protPos)) >= 54 & as.numeric(as.character(rLH_final_filtered_df$protPos)) < 195),c('function_score')])

VHLx2rLH_final_df <- rLH_final_filtered_df[which(rLH_final_filtered_df$sge_region == "exon 2"),]
VHLx2rLH_function_score_median_syn <- median(VHLx2rLH_final_df[which(VHLx2rLH_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx2rLH_function_score_median_ns <- median(VHLx2rLH_final_df[which(VHLx2rLH_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx2rLH_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx2rLH_final_df$protPos)) < 195),c('function_score')])
min(VHLx2rLH_final_df$function_score)
max(VHLx2rLH_final_df$function_score)
VHLx2rLH_final_df$function_score_sns <- log2(2^(VHLx2rLH_final_df$function_score-(VHLx2rLH_function_score_median_syn-VHLxrLH_function_score_median_syn))/2^(VHLxrLH_function_score_median_syn))
min(VHLx2rLH_final_df$function_score_sns)
max(VHLx2rLH_final_df$function_score_sns)
sd(VHLx2rLH_final_df$function_score_sns)

VHLx3arLH_final_df <- rLH_final_filtered_df[which(rLH_final_filtered_df$sge_region == "exon 3a"),]
VHLx3arLH_function_score_median_syn <- median(VHLx3arLH_final_df[which(VHLx3arLH_final_df$conseq == 'SYNONYMOUS'),c('function_score')])
VHLx3arLH_function_score_median_ns <- median(VHLx3arLH_final_df[which(VHLx3arLH_final_df$conseq == 'STOP_GAINED' & as.numeric(as.character(VHLx3arLH_final_df$protPos)) >= 54 & as.numeric(as.character(VHLx3arLH_final_df$protPos)) < 195),c('function_score')])
min(VHLx3arLH_final_df$function_score)
max(VHLx3arLH_final_df$function_score)
VHLx3arLH_final_df$function_score_sns <- log2(2^(VHLx3arLH_final_df$function_score-(VHLx3arLH_function_score_median_syn-VHLxrLH_function_score_median_syn))/2^(VHLxrLH_function_score_median_syn))
min(VHLx3arLH_final_df$function_score_sns)
max(VHLx3arLH_final_df$function_score_sns)
sd(VHLx3arLH_final_df$function_score_sns)

#### END global scaling of filtered data
#will keep same number of columns as region's "final" df, to facilitate score merging
###start exon 2 final df merging
VHLx2rLH_final_df$function_score_final <- VHLx2rLH_final_df$function_score_sns
VHLx2rLH_final_df$rna_score <- log2(VHLx2rLH_final_df$tHDR_rna_pre_ratio_rLH1rLH2_mean_synnorm)
VHLx2rLH_final_df$post_rna_score <- log2(VHLx2rLH_final_df$tHDR_rna2_post_ratio_rLH1rLH2_mean_synnorm)

###start exon 3 final df merging
VHLx3arLH_final_df$function_score_final <- VHLx3arLH_final_df$function_score_sns
VHLx3arLH_final_df$rna_score <- log2(VHLx3arLH_final_df$tHDR_rna_pre_ratio_rLH1rLH2_mean_synnorm)
VHLx3arLH_final_df$post_rna_score <- log2(VHLx3arLH_final_df$tHDR_rna2_post_ratio_rLH1rLH2_mean_synnorm)


###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx2rLH_null_fss <- VHLx2rLH_final_df[which(VHLx2rLH_final_df$conseq == "SYNONYMOUS"),]$function_score_sns 
VHLx2rLH_null_mean <- mean(VHLx2rLH_null_fss)
VHLx2rLH_null_sd <- sd(VHLx2rLH_null_fss)
VHLx2rLH_final_df$pvalues <- pnorm(VHLx2rLH_final_df$function_score_sns, VHLx2rLH_null_mean, sd=VHLx2rLH_null_sd) 
VHLx2rLH_final_df$fdr <- p.adjust(VHLx2rLH_final_df$pvalues, method = "BH")
VHLx2rLH_final_df$fs_sig <- "FDR NA"
VHLx2rLH_final_df[which(VHLx2rLH_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx2rLH_final_df[which(VHLx2rLH_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx2rLH_final_df[which(VHLx2rLH_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx2rLH_final_df[which(VHLx2rLH_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx2rLH_final_df[which(VHLx2rLH_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx2rLH_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+geom_text(data=VHLx2rLH_final_df[which(VHLx2rLH_final_df$fdr < 0.05),],aes(label=paste(cHGVS,pHGVS)),hjust=0.5, vjust=0.5, size = 1.4,colour='Gray')

max(VHLx2rLH_final_df[which(VHLx2rLH_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx2rLH_final_df[which(VHLx2rLH_final_df$fdr > 0.01),]$function_score_sns)

###per exon significance using null distribution of syn variants, pnorm, and BH adjustment
VHLx3arLH_null_fss <- VHLx3arLH_final_df[which(VHLx3arLH_final_df$conseq == "SYNONYMOUS"),]$function_score_sns 
VHLx3arLH_null_mean <- mean(VHLx3arLH_null_fss)
VHLx3arLH_null_sd <- sd(VHLx3arLH_null_fss)
VHLx3arLH_final_df$pvalues <- pnorm(VHLx3arLH_final_df$function_score_sns, VHLx3arLH_null_mean, sd=VHLx3arLH_null_sd) 
VHLx3arLH_final_df$fdr <- p.adjust(VHLx3arLH_final_df$pvalues, method = "BH")
VHLx3arLH_final_df$fs_sig <- "FDR NA"
VHLx3arLH_final_df[which(VHLx3arLH_final_df$fdr > 0.5),]$fs_sig <- "FDR > 0.5"
VHLx3arLH_final_df[which(VHLx3arLH_final_df$fdr < 0.5),]$fs_sig <- "FDR < 0.5"
VHLx3arLH_final_df[which(VHLx3arLH_final_df$fdr < 0.2),]$fs_sig <- "FDR < 0.2"
VHLx3arLH_final_df[which(VHLx3arLH_final_df$fdr < 0.05),]$fs_sig <- "FDR < 0.05"
VHLx3arLH_final_df[which(VHLx3arLH_final_df$fdr < 0.01),]$fs_sig <- "FDR < 0.01"

ggplot(VHLx3arLH_final_df, aes(x=as.numeric(as.character(pos)), y=function_score_sns,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+geom_text(data=VHLx3arLH_final_df[which(VHLx3arLH_final_df$fdr < 0.05),],aes(label=paste(cHGVS,pHGVS)),hjust=0.5, vjust=0.5, size = 1.4,colour='Gray')


max(VHLx3arLH_final_df[which(VHLx3arLH_final_df$fdr < 0.01),]$function_score_sns)
min(VHLx3arLH_final_df[which(VHLx3arLH_final_df$fdr > 0.01),]$function_score_sns)

# bind all the "final" dataframes back together
rLH_final_filtered_df <- rbind(VHLx2rLH_final_df,VHLx3arLH_final_df)

```

Section to IMPORT OUTSIDE DATA ON VHL VARIANTS

List of resources:
Human variation:
Disease:
- ClinVar
- VHLDB
Cancer:
- cBioPortal
Population:
- UKB
- gnomad
- Bravo/Topmed)

Predictors:
- CADD
- boostDM
- EVE
- VARITY
- REVEL

```{r}
#write.table(final_filtered_df, "/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL_final_filtered_pre_annotated_20230504.txt", sep="\t",row.names=F)
#final_filtered_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL_final_filtered_pre_annotated_20230504.txt", sep="\t", header=TRUE)

#clinvar push to most recent data:
clinvar_04052023_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/clinvar_VHL_SNVs_1star_04052023.txt", sep="\t", header=TRUE)

#remove column containing older clinvar annotations - will split clinvar_simple into .x and .y
final_filtered_df <- merge(final_filtered_df, clinvar_04052023_df[,c("pos_alt","clinvar_simple")], by="pos_alt",all.x=TRUE)

final_filtered_df <- final_filtered_df %>% 
  rename(
    clinvar_simple = clinvar_simple.y
    )

#add a factor level to include new "absent"
levels(final_filtered_df$clinvar_simple) = c("Benign","Conflicting interpretations of pathogenicity", "Likely benign", "Likely pathogenic", "Pathogenic", "Uncertain significance", "absent")
#change NA values to "absent" from ClinVar
final_filtered_df$clinvar_simple[is.na(final_filtered_df$clinvar_simple)] <- "absent"
#remove column containing older clinvar annotations
final_filtered_df <- final_filtered_df[,!names(final_filtered_df) %in% c("clinvar_simple.x")]

#VHLdb analysis
VHLdb_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/20220805_vhldb_processed.txt", sep="\t", header=TRUE)
VHLdb_df <- VHLdb_df %>% 
  rename(
    cHGVS = Variant,
    vhldb_codon = Codon,
    vhldb_effect = Effect,
    vhldb_type = Type,
    vhldb_surface = Surface,
    vhldb_dssp = Dssp,
    vhldb_transmission_simple = VHLdb_transmission_simple
    )
final_filtered_df <- merge(final_filtered_df, VHLdb_df, by="cHGVS",all.x=TRUE)

final_filtered_df$vhldb_germline_count[is.na(final_filtered_df$vhldb_germline_count)] <- 0
final_filtered_df$vhldb_somatic_count[is.na(final_filtered_df$vhldb_somatic_count)] <- 0
final_filtered_df$vhldb_syndrome_count[is.na(final_filtered_df$vhldb_syndrome_count)] <- 0
final_filtered_df$vhldb_rcc_count[is.na(final_filtered_df$vhldb_rcc_count)] <- 0
final_filtered_df$vhldb_pheo_count[is.na(final_filtered_df$vhldb_pheo_count)] <- 0
final_filtered_df$vhldb_hb_count[is.na(final_filtered_df$vhldb_hb_count)] <- 0
final_filtered_df$vhldb_t1_count[is.na(final_filtered_df$vhldb_t1_count)] <- 0
final_filtered_df$vhldb_t2_count[is.na(final_filtered_df$vhldb_t2_count)] <- 0
final_filtered_df$vhldb_t2a_count[is.na(final_filtered_df$vhldb_t2a_count)] <- 0
final_filtered_df$vhldb_t2b_count[is.na(final_filtered_df$vhldb_t2b_count)] <- 0
final_filtered_df$vhldb_t2c_count[is.na(final_filtered_df$vhldb_t2c_count)] <- 0

#import UKBB data from Genebass
UKB_VHL_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/genebass_2022_08_11_13_09_24.csv", sep=",", header=TRUE)
UKB_VHL_df <- UKB_VHL_df %>% 
  rename(
    ukb_conseq = consequence,
    ukb_hgvsc = hgvsc,
    ukb_hgvsp = hgvsp,
    ukb_pval_M03.9 = pval,
    ukb_beta_M03.9 = beta,
    ukb_ac_cases = ac_cases,
    ukb_an_cases =  an_cases,
    ukb_ac_controls = ac_controls,
    ukb_an_controls = an_controls,
    ukb_association_ac = association_ac,
    ukb_association_an = association_an,
    ukb_allele_count = allele_count,
    ukb_homozygote_count = homozygote_count,
    ukb_allele_number = allele_number,
    ukb_af_cases = af_cases,
    ukb_af_controls = af_controls,
    ukb_association_af = association_af,
    ukb_allele_frequency = allele_frequency
    )

final_filtered_df <- merge(final_filtered_df, UKB_VHL_df, by="cHGVS",all.x=TRUE)

final_filtered_df$ukb_association_af[is.na(final_filtered_df$ukb_association_af)] <- 0
final_filtered_df$ukb_allele_count[is.na(final_filtered_df$ukb_allele_count)] <- 0
final_filtered_df$ukb_association_ac[is.na(final_filtered_df$ukb_association_ac)] <- 0
final_filtered_df$ukb_association_an[is.na(final_filtered_df$ukb_association_an)] <- 0
final_filtered_df$ukb_af_cases[is.na(final_filtered_df$ukb_af_cases)] <- 0
final_filtered_df$ukb_allele_frequency[is.na(final_filtered_df$ukb_allele_frequency)] <- 0
final_filtered_df$ukb_af_controls[is.na(final_filtered_df$ukb_af_controls)] <- 0

#which points are in BRAVO?
#note -- used BRAVO Freeze 8, Quality = Pass
VHL_bravo_df <- read.csv('/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL_TOPMEDv8_220818_ENSG00000134086.csv',header = TRUE)
VHL_bravo_df <- VHL_bravo_df %>% 
  rename(
    bravo_het_count = Het,
    bravo_hom_count = HomAlt,
    bravo_freq = Frequency
    )
VHL_bravo_df <- VHL_bravo_df[,c("bravo_het_count","bravo_hom_count","bravo_freq","hg38_pos_alt")]
final_filtered_df <- merge(final_filtered_df, VHL_bravo_df, by="hg38_pos_alt",all.x=TRUE)
final_filtered_df$bravo_het_count[is.na(final_filtered_df$bravo_het_count)] <- 0
final_filtered_df$bravo_hom_count[is.na(final_filtered_df$bravo_hom_count)] <- 0
final_filtered_df$bravo_freq[is.na(final_filtered_df$bravo_freq)] <- 0

#Imported gnomAD v2 and v3 data sets separately.  NOTES:  used the non-TOPMED data points to prevent redundancy (did not use non-cancer)

VHL_gnomADv2_df <- read.csv('/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/gnomAD_v2.1.1_nonTOPMed_ENSG00000134086_2022_08_18.csv',header = TRUE)
VHL_gnomADv2_df <- VHL_gnomADv2_df %>% 
  rename(
    cHGVS = Transcript_Consequence,
    gv2_allele_count = Allele_Count,
    gv2_allele_number = Allele_Number,
    gv2_allele_freq = Allele_Frequency
    )
VHL_gnomADv2_df <- VHL_gnomADv2_df[,c("gv2_allele_count","gv2_allele_number","gv2_allele_freq","pos_alt")]
final_filtered_df <- merge(final_filtered_df, VHL_gnomADv2_df, by="pos_alt",all.x=TRUE)
final_filtered_df$gv2_allele_count[is.na(final_filtered_df$gv2_allele_count)] <- 0
final_filtered_df$gv2_allele_number[is.na(final_filtered_df$gv2_allele_number)] <- 0
final_filtered_df$gv2_allele_freq[is.na(final_filtered_df$gv2_allele_freq)] <- 0

VHL_gnomADv3_df <- read.csv('/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/gnomAD_v3.1.2_non-TOPMed_ENSG00000134086_2022_08_18.csv',header = TRUE)
VHL_gnomADv3_df <- VHL_gnomADv3_df %>% 
  rename(
    cHGVS = Transcript_Consequence,
    gv3_allele_count = Allele_Count,
    gv3_allele_number = Allele_Number,
    gv3_allele_freq = Allele_Frequency
    )
VHL_gnomADv3_df <- VHL_gnomADv3_df[,c("gv3_allele_count","gv3_allele_number","gv3_allele_freq","hg38_pos_alt")]
final_filtered_df <- merge(final_filtered_df, VHL_gnomADv3_df, by="hg38_pos_alt",all.x=TRUE)
final_filtered_df$gv3_allele_count[is.na(final_filtered_df$gv3_allele_count)] <- 0
final_filtered_df$gv3_allele_number[is.na(final_filtered_df$gv3_allele_number)] <- 0
final_filtered_df$gv3_allele_freq[is.na(final_filtered_df$gv3_allele_freq)] <- 0


#import all boostDM data:
VHL_RCCC_boostDM_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL.RCCC.prediction_boostDM.txt", sep="\t", header=TRUE)

VHL_RCCC_boostDM_df <- VHL_RCCC_boostDM_df %>% 
  rename(
    boostDM_frequency = frequency,
    boostDM_selected_model_ttype = selected_model_ttype,
    boostDM_ENSEMBL_TRANSCRIPT = ENSEMBL_TRANSCRIPT
    )

VHL_RCCC_boostDM_df$hg38_pos_alt <- paste0(VHL_RCCC_boostDM_df$pos,VHL_RCCC_boostDM_df$alt)
VHL_RCCC_boostDM_to_merge_df <- VHL_RCCC_boostDM_df[,c("hg38_pos_alt","boostDM_score","boostDM_class","boostDM_frequency","boostDM_selected_model_ttype","boostDM_ENSEMBL_TRANSCRIPT")]
final_filtered_df <- merge(final_filtered_df, VHL_RCCC_boostDM_to_merge_df, by="hg38_pos_alt",all.x=TRUE)

VHL_EVE_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL_HUMAN_EVE.csv", sep=",", header=TRUE)
EVE_drops <- c("wt_aa","position","mt_aa")
VHL_EVE_to_merge_df <- VHL_EVE_df[ , !(names(VHL_EVE_df) %in% EVE_drops)]
final_filtered_df <- merge(final_filtered_df, VHL_EVE_to_merge_df, by="pHGVS",all.x=TRUE)
final_filtered_df$frequency_gv2[is.na(final_filtered_df$frequency_gv2)] <- 0
final_filtered_df$frequency_gv3[is.na(final_filtered_df$frequency_gv3)] <- 0

final_filtered_df$combined_pop_count <- final_filtered_df$gv2_allele_count+final_filtered_df$gv3_allele_count+final_filtered_df$bravo_het_count+final_filtered_df$ukb_allele_count
length(which(final_filtered_df$combined_pop_count == 0))
length(which(final_filtered_df$combined_pop_count > 0))

VHL_VARITY_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL_P40337_1-213_VARITY_R_20220816.csv", sep=",", header=TRUE)
VHL_VARITY_to_merge_df <- VHL_VARITY_df[,c("VARITY_R","VARITY_R_LOO","VARITY_ER","VARITY_ER_LOO","pHGVS")]
final_filtered_df <- merge(final_filtered_df, VHL_VARITY_to_merge_df, by="pHGVS",all.x=TRUE)

VHL_revel_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/revel_grch38_chrom_03_000361460_025647602.csv", sep=",", header=TRUE)
VHL_revel_to_merge_df <- VHL_revel_df[,c("pos_alt","REVEL")]
final_filtered_df <- merge(final_filtered_df, VHL_revel_to_merge_df, by="pos_alt",all.x=TRUE)

#write.table(final_filtered_df, "/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL_final_filtered_annotated_20230504.txt", sep="\t",row.names=F)

#final_filtered_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL_final_filtered_annotated_20230504.txt", sep="\t", header=TRUE)
final_filtered_no1c_df <- final_filtered_df[which(final_filtered_df$sge_region != "exon 1c"),]

```

Section to define positions, add order to data frames

```{r}

VHLx1c_start <- min(as.numeric(as.character((final_filtered_df$pos))))
VHLx1a_start <-min(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 1a"),]$pos))))
VHLx1b_start <-min(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 1b"),]$pos))))
VHLx1p_start <-min(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 1p"),]$pos))))
VHLx2_start <-min(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 2"),]$pos))))
VHLx3a_start <-min(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 3a"),]$pos))))
VHLx3b_start <-min(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 3b"),]$pos))))

VHLx1c_end <- max(as.numeric(as.character((final_filtered_df$pos))))
VHLx1a_end <-max(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 1a"),]$pos))))
VHLx1b_end <-max(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 1b"),]$pos))))
VHLx1p_end <-max(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 1p"),]$pos))))
VHLx2_end <-max(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 2"),]$pos))))
VHLx3a_end <-max(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 3a"),]$pos))))
VHLx3b_end <-max(as.numeric(as.character((final_filtered_df[which(final_filtered_df$sge_region == "exon 3b"),]$pos))))

#conseq ordered 
conseq_ord_final_filtered_df <- final_filtered_df
conseq_ord_final_filtered_df$conseq <- factor(conseq_ord_final_filtered_df$conseq, levels = c('5PRIME_UTR','NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))
conseq_ord_final_filtered_no1c_df <- final_filtered_no1c_df
conseq_ord_final_filtered_no1c_df$conseq <- factor(conseq_ord_final_filtered_no1c_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))

#clinvar ordered
clinvar_ord_final_filtered_df <- final_filtered_df
clinvar_ord_final_filtered_df$clinvar_simple <- factor(clinvar_ord_final_filtered_df$clinvar_simple, levels = c("REF","absent","Conflicting interpretations of pathogenicity","Uncertain significance","Likely pathogenic","Pathogenic","Likely benign","Benign"))
clinvar_ord_final_filtered_no1c_df <- clinvar_ord_final_filtered_df[which(clinvar_ord_final_filtered_df$sge_region != "exon 1c"),]
clinvar_dots_final_filtered_df <- final_filtered_df
clinvar_dots_final_filtered_df$clinvar_simple <- factor(clinvar_dots_final_filtered_df$clinvar_simple, levels = c("Benign","Likely benign","Pathogenic","Likely pathogenic","Uncertain significance","Conflicting interpretations of pathogenicity","REF","absent"))
clinvar_dots_final_filtered_no1c_df <- clinvar_dots_final_filtered_df[which(clinvar_dots_final_filtered_df$sge_region != "exon 1c"),]

conseq_labels <- c("3PRIME_UTR" = "3' UTR","5PRIME_UTR" = "5' UTR", "CANONICAL_SPLICE" = "Canonical splice", "INTRONIC" = "Intronic", "NON_SYNONYMOUS" = "Missense", "SPLICE_SITE" = "Splice region", "STOP_GAINED" = "Nonsense", "STOP_LOST" = "Stop lost", "SYNONYMOUS" = "Synonymous")

clinvar_labels <- c("absent" = "Absent","Likely pathogenic" = "Likely path.", "Benign" = "Benign", "Likely benign" = "Likely ben.", "Pathogenic" = "Pathogenic", "Uncertain significance" = "Uncertain", "Conflicting interpretations of pathogenicity" = "Conflict")

#plot limits
VHL_lim_lo <- min(final_filtered_df$function_score_final)-0.5
VHL_lim_hi <- max(final_filtered_df$function_score_final)+0.5
VHL_no1c_lim_lo <- min(final_filtered_no1c_df$function_score_final)-0.5
VHL_no1c_lim_hi <- max(final_filtered_no1c_df$function_score_final)+0.5

```

```{r}
#ANALYSIS COMPARING EFFECT OF DAB VS NO DAB

#FS how many LoF variants detected with FDR 0.01?
length(which(final_filtered_df$fdr < 0.01))
length(which(final_filtered_df$fdr < 0.01 & (final_filtered_df$sge_region != "exon 1c" & final_filtered_df$sge_region != "exon 1a" )))
length(which(rL4_final_filtered_df$fdr < 0.01))

#280 variants with fdr < 0.01 in DAB experiments vs. 201 in no-DAB experiments (same regions for comparison)

#all histogram coloured by conseq
VHL_dnd_lim_lo <- min(c(min(final_filtered_no1c_df$function_score_sns),min(rL4_final_filtered_df$function_score_sns)))-0.5
VHL_dnd_lim_hi <- max(c(max(final_filtered_no1c_df$function_score_sns),max(rL4_final_filtered_df$function_score_sns)))+0.5

#FF SFig. 2  - Library distributions before and after recoding regions of exon 1
ggplot(VHLxrL4_all_master_df[which(VHLxrL4_all_master_df$sge_region == "exon 1c"),], aes(x=as.numeric(as.character(pos)),y=lib_pseudo_freq))+geom_point(aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variant freq.')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels=conseq_labels)+scale_y_continuous(trans='log10',limits=c(0.00001,0.01),breaks=c(0.00001,0.00001,0.0001,0.001,0.01))

#FF SFig. 2 CANFIG - VHLrL4_x1c_lib_histo
ggplot(VHLxrL4_all_master_df[which(VHLxrL4_all_master_df$sge_region == "exon 1c"),], aes(x=lib_pseudo_freq))+geom_histogram(aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Variant freq.')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)

#FF SFig. 2  - Library distributions before and after recoding regions of exon 1
ggplot(VHLxrL4_all_master_df[which(VHLxrL4_all_master_df$sge_region == "exon 1a"),], aes(x=as.numeric(as.character(pos)),y=lib_pseudo_freq))+geom_point(aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variant freq.')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels=conseq_labels)+scale_y_continuous(trans='log10',limits=c(0.00001,0.1),breaks=c(0.00001,0.00001,0.0001,0.001,0.01,0.1))

#FF SFig. 2 CANFIG - VHLrL4_x1a_lib_histo
ggplot(VHLxrL4_all_master_df[which(VHLxrL4_all_master_df$sge_region == "exon 1a"),], aes(x=lib_pseudo_freq))+geom_histogram(aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Variant freq.')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)


VHLxrLD_all_master_df$sge_region <- factor(VHLxrLD_all_master_df$sge_region, levels = c('exon 1c','exon 1b','exon 1a','exon 1p','exon 2','exon 3a','exon 3b'))

#FF SFig. 2 CANFIG
ggplot(VHLxrLD_all_master_df, aes(x=as.numeric(as.character(lib_pseudo_freq))))+geom_histogram(aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Variants')+xlab(' Library Freq.')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=8),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+facet_wrap(~ sge_region,nrow=2,scales="free")+scale_x_continuous(limits=c(0,0.004))


VHLx2rL4_final_df$conseq <- factor(VHLx2rL4_final_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))
VHLx2rLD_final_df$conseq <- factor(VHLx2rLD_final_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))


VHLx2rL4_fs_histo <- ggplot(VHLx2rL4_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq),binwidth = 0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))
VHLx2rL4_fs_histo

#FF Final Fig. 1f
ggplot(VHLx2rL4_final_df[which(VHLx2rL4_final_df$conseq == "CANONICAL_SPLICE" |VHLx2rL4_final_df$conseq == "STOP_GAINED" | VHLx2rL4_final_df$conseq == "SYNONYMOUS"),])+geom_histogram(aes(x=function_score_sns,fill=conseq),binwidth = 0.2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_ns_cs_syn)+coord_cartesian(xlim=c(-4,1))

VHLx2rLD_fs_histo <- ggplot(VHLx2rLD_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq),binwidth = 0.2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))
VHLx2rLD_fs_histo

#FS main text
median(VHLx2rL4_final_df[which(VHLx2rL4_final_df$conseq == "CANONICAL_SPLICE" |VHLx2rL4_final_df$conseq == "STOP_GAINED"),c("function_score_sns")])
median(VHLx2rLD_final_df[which(VHLx2rLD_final_df$conseq == "CANONICAL_SPLICE" |VHLx2rL4_final_df$conseq == "STOP_GAINED"),c("function_score_sns")])

VHLx2rLD_fs_ns_cs_syn_histo <- ggplot(VHLx2rLD_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq),binwidth = 0.2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_ns_cs_syn)+coord_cartesian(xlim=c(-4,1))

#only cs, ns, syn:
#FF Final Fig. 1g 
ggplot(VHLx2rLD_final_df[which(VHLx2rLD_final_df$conseq == "CANONICAL_SPLICE" |VHLx2rLD_final_df$conseq == "STOP_GAINED" | VHLx2rLD_final_df$conseq == "SYNONYMOUS"),])+geom_histogram(aes(x=function_score_sns,fill=conseq),binwidth = 0.2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_ns_cs_syn)+coord_cartesian(xlim=c(-4,1))

#for legend
ggplot(VHLx2rLD_final_df[which(VHLx2rLD_final_df$conseq == "CANONICAL_SPLICE" |VHLx2rLD_final_df$conseq == "STOP_GAINED" | VHLx2rLD_final_df$conseq == "SYNONYMOUS"),])+geom_histogram(aes(x=function_score_sns,fill=conseq),binwidth = 0.2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_ns_cs_syn,labels=conseq_labels)+coord_cartesian(xlim=c(-4,1))

#SFig3 section
VHLx1brL4_final_df$conseq <- factor(VHLx1brL4_final_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))
VHLx1brLD_final_df$conseq <- factor(VHLx1brLD_final_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))

VHLx1brL4_fs_histo <- ggplot(VHLx1brL4_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))  
VHLx1brL4_fs_histo

VHLx1brLD_fs_histo <- ggplot(VHLx1brLD_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1)) 
VHLx1brLD_fs_histo

VHLx1prL4_fs_histo <- ggplot(VHLx1prL4_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))   
VHLx1prL4_fs_histo

VHLx1prLD_fs_histo <- ggplot(VHLx1prLD_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))  
VHLx1prLD_fs_histo

VHLx2rL4_fs_histo <- ggplot(VHLx2rL4_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))  
VHLx2rL4_fs_histo

VHLx2rLD_fs_histo <- ggplot(VHLx2rLD_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))  
VHLx2rLD_fs_histo

VHLx3arL4_final_df$conseq <- factor(VHLx3arL4_final_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))
VHLx3arLD_final_df$conseq <- factor(VHLx3arLD_final_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))

VHLx3arL4_fs_histo <- ggplot(VHLx3arL4_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))   
VHLx3arL4_fs_histo

VHLx3arLD_fs_histo <- ggplot(VHLx3arLD_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))  
VHLx3arLD_fs_histo

VHLx3brL4_final_df$conseq <- factor(VHLx3brL4_final_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))
VHLx3brLD_final_df$conseq <- factor(VHLx3brLD_final_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))

VHLx3brL4_fs_histo <- ggplot(VHLx3brL4_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors) +coord_cartesian(xlim=c(-4,1))  
VHLx3brL4_fs_histo

VHLx3brLD_fs_histo <- ggplot(VHLx3brLD_final_df)+geom_histogram(aes(x=function_score_sns,fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))  
VHLx3brLD_fs_histo
#Final Sfig. 3

##FF Final Sfig. 3 SFig3 - final - export 3 x 10
grid.arrange(VHLx1brL4_fs_histo,VHLx1prL4_fs_histo,VHLx2rL4_fs_histo,VHLx3arL4_fs_histo,VHLx3brL4_fs_histo,VHLx1brLD_fs_histo,VHLx1prLD_fs_histo,VHLx2rLD_fs_histo,VHLx3arLD_fs_histo,VHLx3brLD_fs_histo,nrow=2)

#all colors with legend:
final_filtered_df$conseq <- factor(final_filtered_df$conseq, levels = c('5PRIME_UTR','3PRIME_UTR','STOP_LOST','NON_SYNONYMOUS','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))
ggplot(final_filtered_df)+geom_histogram(aes(x=function_score_sns,fill=conseq),binwidth = 0.2)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+coord_cartesian(xlim=c(-4,1))

```


```{r}
#Analysis comparing effects in HIF1KO vs normal cells
rLH_final_filtered_df$rLH_function_score_final <- rLH_final_filtered_df$function_score_final
rLH_final_filtered_df$rLH_fdr <- rLH_final_filtered_df$fdr
rLH_rLD_intersect_df <- merge(final_filtered_df, rLH_final_filtered_df[,c("rLH_function_score_final","rLH_fdr","cHGVS")], by="cHGVS",all= FALSE)

#conseq ordered 
rLH_rLD_intersect_df$conseq <- factor(rLH_rLD_intersect_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))
#clinvar ordered
rLH_rLD_intersect_df$clinvar_simple <- factor(rLH_rLD_intersect_df$clinvar_simple, levels = c("REF","absent","Conflicting interpretations of pathogenicity","Uncertain significance","Likely pathogenic","Pathogenic","Likely benign","Benign"))

significance_colors <-
  c("FDR < 0.01" = "#023C38",
  "FDR < 0.05" = "#65B55D",
  "FDR < 0.2" = "#DDA072",
  "FDR < 0.5" = "#DDA072",
  "FDR > 0.5" = "#DDA072")

significance_colors_simple <-
  c("FDR < 0.01" = "#276E69",
  "FDR < 0.05" = "#DDA072",
  "FDR < 0.2" = "#DDA072",
  "FDR < 0.5" = "#DDA072",
  "FDR > 0.5" = "#DDA072")

rLH_rLD_intersect_df$rLH_fs_sig <- "FDR > 0.5"
rLH_rLD_intersect_df[which(rLH_rLD_intersect_df$rLH_fdr < 0.5),]$rLH_fs_sig <- "FDR < 0.5"
rLH_rLD_intersect_df[which(rLH_rLD_intersect_df$rLH_fdr < 0.2),]$rLH_fs_sig <- "FDR < 0.2"
rLH_rLD_intersect_df[which(rLH_rLD_intersect_df$rLH_fdr < 0.05),]$rLH_fs_sig <- "FDR < 0.05"
rLH_rLD_intersect_df[which(rLH_rLD_intersect_df$rLH_fdr < 0.01),]$rLH_fs_sig <- "FDR < 0.01"

rLH_rLD_intersect_df$clinvar_simple <- factor(rLH_rLD_intersect_df$clinvar_simple, levels = c("Benign","Likely benign","Likely pathogenic","Pathogenic","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

rLH_rLD_intersect_df$clinvar_simple <- factor(rLH_rLD_intersect_df$clinvar_simple, levels = c("REF","absent","Conflicting interpretations of pathogenicity","Uncertain significance","Pathogenic","Likely pathogenic","Likely benign","Benign"))

rLH_rLD_intersect_df$conseq <- factor(rLH_rLD_intersect_df$conseq, levels = c('NON_SYNONYMOUS','3PRIME_UTR','STOP_LOST','SPLICE_SITE','INTRONIC','SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))

#Final Figure Fig. 5a 
VHLx23arLD_fs_histo <- ggplot(rLH_rLD_intersect_df)+geom_histogram(aes(x=function_score_final,fill=conseq),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))
VHLx23arLH_fs_histo <- ggplot(rLH_rLD_intersect_df)+geom_histogram(aes(x=rLH_function_score_final,fill=conseq),binwidth = 0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-4,1))
#FF Final Figure Fig. 5a 
grid.arrange(VHLx23arLD_fs_histo,VHLx23arLH_fs_histo,nrow=2)

rLH_rLD_intersect_df$clinvar_simple <- factor(rLH_rLD_intersect_df$clinvar_simple, levels = c("Benign","Likely benign","Likely pathogenic","Pathogenic","Uncertain significance","Conflicting interpretations of pathogenicity","absent","REF"))

#rLH scores by clinvar, alone
ggplot(rLH_rLD_intersect_df)+geom_histogram(aes(x=rLH_function_score_final,fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Variants')+xlab('Function Score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)

cHGVS_sign_in_HIF1A_KO <- rLH_rLD_intersect_df[rLH_rLD_intersect_df$rLH_fs_sig %in% c("FDR < 0.01"),]$cHGVS

#rLH scores vs rLD scores by clinvar
ggplot(rLH_rLD_intersect_df[,], aes(x=rLH_function_score_final,y=function_score_final))+geom_point(aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('HIF1A-KO function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(ylim=c(-4,1),xlim=c(-0.5,0.5))+geom_text(data=rLH_rLD_intersect_df[rLH_rLD_intersect_df$cHGVS %in% cHGVS_sign_in_HIF1A_KO,],aes(label=cHGVS),hjust=0.5, vjust=1.5, size = 2)

#FF CANFIG Final Figure 5 Fig. 5 b 5b
ggplot(rLH_rLD_intersect_df[,], aes(y=rLH_function_score_final,x=function_score_final))+geom_point(aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('Function score')+ylab('HIF1A-KO function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels = conseq_labels)+coord_cartesian(xlim=c(-4,1),ylim=c(-0.5,0.5))
#for legend
ggplot(rLH_rLD_intersect_df[,], aes(y=rLH_function_score_final,x=function_score_final))+geom_point(aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Function score')+ylab('HIF1A-KO function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels = conseq_labels)+coord_cartesian(xlim=c(-4,1),ylim=c(-0.5,0.5))+geom_text(data=rLH_rLD_intersect_df[rLH_rLD_intersect_df$cHGVS %in% cHGVS_sign_in_HIF1A_KO,],aes(label=cHGVS),hjust=0.5, vjust=1.5, size = 2)


#no significant correlation between HIF1-KO score and rLD score
VHLxrLD_rLH_fs_cor_s <- cor(rLH_rLD_intersect_df$rLH_function_score_final,rLH_rLD_intersect_df$function_score_final,method='spearman')
VHLxrLD_rLH_fs_cor_p <- cor(rLH_rLD_intersect_df$rLH_function_score_final,rLH_rLD_intersect_df$function_score_final,method='pearson')
cor.test(rLH_rLD_intersect_df$rLH_function_score_final,rLH_rLD_intersect_df$function_score_final,method='spearman')
cor.test(rLH_rLD_intersect_df$rLH_function_score_final,rLH_rLD_intersect_df$function_score_final,method='pearson')

```


```{r}
VHLrLD_full_gene_function_score_sns_by_pos <- ggplot(final_filtered_df, aes(x=pos, y=function_score_sns,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('funciton score norm. by exp')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=8),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+scale_x_discrete(breaks = c(VHLx1b_start,VHLx1a_start,VHLx1c_start,VHLx1p_start,VHLx2_start,VHLx3a_start,VHLx3b_start))

VHLrLD_full_gene_function_score_final_by_pos <- ggplot(final_filtered_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('funciton score final')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=8),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+scale_x_discrete(breaks = c(VHLx1b_start,VHLx1a_start,VHLx1c_start,VHLx1p_start,VHLx2_start,VHLx3a_start,VHLx3b_start))

VHLrLD_no1c_function_score_final_pos <- ggplot(final_filtered_no1c_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('funciton score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=8),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+scale_x_discrete(breaks = c(VHLx1b_start,VHLx1a_start,VHLx1p_start,VHLx2_start,VHLx3a_start,VHLx3b_start))

grid.arrange(VHLrLD_full_gene_function_score_sns_by_pos,VHLrLD_full_gene_function_score_final_by_pos,VHLrLD_no1c_function_score_final_pos,nrow=3)

#CANFIG function score by position:
final_filtered_no1c_df$pos <- as.numeric(as.character(final_filtered_no1c_df$pos))

VHLrLD_exon1_df <- final_filtered_no1c_df[which(final_filtered_no1c_df$pos < VHLx1p_start),]
VHLrLD_exon1_map <- ggplot(VHLrLD_exon1_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))
VHLrLD_exon1_map

VHLrLD_intron1_df <- final_filtered_no1c_df[which(final_filtered_no1c_df$pos > VHLx1p_start-50 & final_filtered_no1c_df$pos < VHLx2_start-50),]
VHLrLD_intron1_map <- ggplot(VHLrLD_intron1_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))
VHLrLD_intron1_map

VHLrLD_exon2_df <- final_filtered_no1c_df[which(final_filtered_no1c_df$sge_region == "exon 2"),]
VHLrLD_exon2_map <- ggplot(VHLrLD_exon2_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))
VHLrLD_exon2_map

VHLrLD_exon3_df <- final_filtered_no1c_df[which(final_filtered_no1c_df$pos > VHLx3a_start-50),]
VHLrLD_exon3_map <- ggplot(VHLrLD_exon3_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))
VHLrLD_exon3_map

#FF CANFIG Fig. 2c 
grid.arrange(VHLrLD_exon1_map,VHLrLD_intron1_map,VHLrLD_exon2_map,VHLrLD_exon3_map,nrow=2)

significance_colors <-
  c("FDR < 0.01" = "#023C38",
  "FDR < 0.05" = "#65B55D",
  "FDR < 0.2" = "#276E69",
  "FDR < 0.5" = "#8D4D1C",
  "FDR > 0.5" = "#DDA072")

significance_colors_simple <-
  c("FDR < 0.01" = "#276E69",
  "FDR < 0.05" = "#DDA072",
  "FDR < 0.2" = "#DDA072",
  "FDR < 0.5" = "#DDA072",
  "FDR > 0.5" = "#DDA072")

final_filtered_df$pos <- as.numeric(as.character(final_filtered_df$pos))

ggplot(final_filtered_df[,], aes(x=pos, y=function_score_final,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('funciton score final')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=8),strip.background = element_blank())+scale_color_manual(values=significance_colors)+scale_x_discrete(breaks = c(VHLx1b_start,VHLx1a_start,VHLx1c_start,VHLx1p_start,VHLx2_start,VHLx3a_start,VHLx3b_start))

#FF CANFIG - Final Sfig. 4a - exon 1
ggplot(final_filtered_df[which(final_filtered_df$pos < VHLx1p_start),], aes(x=pos, y=function_score_final,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+ylab('Funciton score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=significance_colors_simple)+scale_x_continuous(breaks = c(VHLx1b_start,VHLx1a_start,VHLx1c_start,VHLx1a_end,VHLx1p_start,VHLx2_start,VHLx3a_start,VHLx3b_start))+coord_cartesian(ylim=c(-4,2))

#CANFIG (not used) - intron 1 - VHLrLD_intron1_score_sig_by_pos, 3x6
ggplot(final_filtered_df[which(final_filtered_df$pos > VHLx1p_start-50 & final_filtered_df$pos < VHLx2_start-50),], aes(x=pos, y=function_score_final,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+ylab('Funciton score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=significance_colors_simple)+scale_x_continuous(breaks = c(VHLx1b_start,VHLx1a_start,VHLx1c_start,VHLx1p_start,VHLx1p_end,VHLx2_start,VHLx3a_start,VHLx3b_start))+coord_cartesian(ylim=c(-4,2))

#CANFIG (not used) - Final Sfig. X - exon 2 - VHLrLD_exon2_score_sig_by_pos, 3x6
ggplot(final_filtered_df[which(final_filtered_df$sge_region == "exon 2"),], aes(x=pos, y=function_score_final,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+ylab('Funciton score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=significance_colors_simple)+scale_x_continuous(breaks = c(VHLx1b_start,VHLx1a_start,VHLx1c_start,VHLx1p_start,VHLx2_start,VHLx2_end,VHLx3a_start,VHLx3b_start))+coord_cartesian(ylim=c(-4,2))

#CANFIG (not used) - Final Sfig. X - exon 3 - VHLrLD_exon3_score_sig_by_pos, 3x6
ggplot(final_filtered_df[which(final_filtered_df$pos > VHLx3a_start-50),], aes(x=pos, y=function_score_final,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+ylab('Funciton score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=significance_colors_simple)+scale_x_continuous(breaks = c(VHLx1b_start,VHLx1a_start,VHLx1c_start,VHLx1p_start,VHLx2_start,VHLx3a_start,VHLx3b_start,VHLx3b_end))+coord_cartesian(ylim=c(-4,2))

#for legend
ggplot(final_filtered_df[which(final_filtered_df$pos < VHLx1p_start),], aes(x=pos, y=function_score_final,color=fs_sig))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Funciton score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=significance_colors_simple)

##Variant filters in place (rLD) for ensuing analyses:  account for library frequency, day 5 frequency,replicate discrepancy, day 13 concordance.
#variants assayed twice now represented only 1 time in df (average to make final score)
#if variant was not assigned a cHGVS position, it's because it led to a different AA change due to syn edit, those are now removed, too, but scores (sns) calculated if relevant.

#all histogram function_score final coloured by conseq #CANFIG
ggplot(final_filtered_no1c_df, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#synonymous
ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "SYNONYMOUS"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#synonymous with no expression defect
ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "SYNONYMOUS" & final_filtered_no1c_df$rna_score >= -1),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "STOP_GAINED"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "CANONICAL_SPLICE"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "NON_SYNONYMOUS"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "SPLICE_SITE"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "INTRONIC"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "3PRIME_UTR"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "STOP_LOST"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#SPLICE_SITE all
ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "SPLICE_SITE"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("post/pre, log2 mean")+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#FF replicate plot #CANFIG Final Fig. 2a
ggplot(final_filtered_no1c_df, aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rLD2_tHDR_post_pre_ratio_synnorm),color=conseq))+geom_point(alpha=1,size=0.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score, Replicate 2')+xlab('Function score, Replicate 1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)

cor(log2(final_filtered_no1c_df$tHDR_post_pre_ratio_synnorm),log2(final_filtered_no1c_df$rLD2_tHDR_post_pre_ratio_synnorm),method='spearman')
#FS number used in manuscript:  correlation between FS's across replicates
cor(log2(final_filtered_no1c_df$tHDR_post_pre_ratio_synnorm),log2(final_filtered_no1c_df$rLD2_tHDR_post_pre_ratio_synnorm),method='pearson')


#FF Final Fig. 2b - CANFIG
ggplot(conseq_ord_final_filtered_no1c_df, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#legend only
ggplot(conseq_ord_final_filtered_no1c_df, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#FF CANFIG Final Fig. 2b inset
ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$conseq == "SYNONYMOUS"| conseq_ord_final_filtered_no1c_df$conseq == "STOP_GAINED"| conseq_ord_final_filtered_no1c_df$conseq == "CANONICAL_SPLICE"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position="None")+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

conseq_violin_final_filtered_no1c_df <- conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$conseq != "STOP_LOST"),]
conseq_violin_final_filtered_no1c_df$conseq <- factor(conseq_violin_final_filtered_no1c_df$conseq, levels = c('SYNONYMOUS','INTRONIC','SPLICE_SITE','STOP_LOST','3PRIME_UTR','NON_SYNONYMOUS','CANONICAL_SPLICE','STOP_GAINED'))


significance_colors_simple2 <- c(named_conseq_colors,c("FDR < 0.01" = "#000000",
  "FDR < 0.05" = "#B6B6B6",
  "FDR < 0.2" = "#B6B6B6",
  "FDR < 0.5" = "#B6B6B6",
  "FDR > 0.5" = "#B6B6B6"))

#CANFIG - violin of scores by conseq - 3x4 VHLrLD_
ggplot(conseq_violin_final_filtered_no1c_df, aes(fill=conseq,x=conseq, y=function_score_final))+geom_violin(aes())+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('SNV consequence')+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+geom_jitter(data=conseq_violin_final_filtered_no1c_df[which(conseq_violin_final_filtered_no1c_df$ fdr < 0.01),] ,size=0.6, alpha=0.3,width=0.2,aes(color=fs_sig))+scale_color_manual(values=significance_colors_simple2)+coord_cartesian(ylim=c(-4,1))

conseq_ord_final_filtered_df$sge_region <- factor(conseq_ord_final_filtered_df$sge_region, levels = c("exon 1c","exon 1b", "exon 1a","exon 1p","exon 2","exon 3a","exon 3b"))

#FF Final Sfig. 4 - histogram of function scores by SGE region
ggplot(conseq_ord_final_filtered_df, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq),bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=5),axis.text.y = element_text(size=5),axis.text.x = element_text(size=5),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+facet_wrap(~ sge_region,,nrow=2,scales="free")
#legend only
ggplot(conseq_ord_final_filtered_df, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq),bins=20)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+facet_wrap(~ sge_region,,nrow=2,scales="free")

#PLOTS SHOWING WHY 1C is excluded from final analysis:
ggplot(conseq_ord_final_filtered_df, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq),binwidth = 0.25)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+facet_wrap(~ sge_region,nrow=2)

exon1c_colors <-
  c("exon 1c" = "#DDA072",
  "exon 1b" = "#003737",
  "exon 1a" = "#0E5151",
  "exon 2" = "#3F8181",
  "exon 1p" = "#669B9B",
  "exon 3a" = "#29195D",
  "exon 3b" = "#413177")

sge_region_labels <-
  c("exon 1c" = "Exon 1, M1-E41",
  "exon 1b" = "Exon 1, S38-S80",
  "exon 1a" = "Exon 1, F76-R113",
  "exon 1p" = "Intron 1",
  "exon 2" = "Exon 2",
  "exon 3a" = "Exon 3, V155-L188",
  "exon 3b" = "Exon 3, R182-*214 ")

final_filtered_df$sge_region <- factor(final_filtered_df$sge_region, levels = c("exon 1c","exon 1b", "exon 1a","exon 1p","exon 2","exon 3a","exon 3b"))

#CANFIG, data on why region exon 1c not included
ggplot(final_filtered_df, aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rLD2_tHDR_post_pre_ratio_synnorm),color=sge_region))+geom_point(data=final_filtered_df[which(final_filtered_df$sge_region!="exon 1c"),],alpha=1,size=1)+geom_point(data=final_filtered_df[which(final_filtered_df$sge_region=="exon 1c"),],alpha=1,size=1.5)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+ylab('Function score, Replicate 2')+xlab('Function score, Replicate 1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=exon1c_colors,labels = sge_region_labels)
#legend only
ggplot(final_filtered_df, aes(x=log2(tHDR_post_pre_ratio_synnorm), y=log2(rLD2_tHDR_post_pre_ratio_synnorm),color=sge_region))+geom_point(alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score, Replicate 2')+xlab('Function score, Replicate 1')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=exon1c_colors,labels = sge_region_labels)

#plots for individual exons
#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 2"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 2"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Chr. 3 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#### RNA Plot
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')

#RNA d6
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))

#RNA d20
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=post_rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))

#RNAd6 vs RNAd20, plotted together
#grid.arrange(ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score (Day 6)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5)),ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=post_rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score (Day 20)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5)),nrow=2)

#RNA vs. DNA (stacked plots)
#grid.arrange(ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5)),ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi)),nrow=2)

#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 1b" ),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))


#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 1b" | conseq_ord_final_filtered_no1c_df$sge_region == "exon 1a"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))


#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 1b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Chr. 3 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))


#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 3a"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))


###
#RNA d6
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 1b")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))

#RNA d20
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 1b")),], aes(x=as.numeric(as.character(CDSpos)), y=post_rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))


#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 1b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Chr. 3 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#grid.arrange(ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 1b")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5)),ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 1b")),], aes(x=as.numeric(as.character(CDSpos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi)),nrow=2)
###

###
#RNA d6
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 1a")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))

#RNA d20
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 1a")),], aes(x=as.numeric(as.character(CDSpos)), y=post_rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))

#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 1a"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Chr. 3 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#grid.arrange(ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 1a")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5)),ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 1a")),], aes(x=as.numeric(as.character(CDSpos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi)),nrow=2)
###


#RNA d6
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 3a")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))

#RNA d20
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 3a")),], aes(x=as.numeric(as.character(CDSpos)), y=post_rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))


#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 3a"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Chr. 3 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#grid.arrange(ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 3a")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5)),ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 3a")),], aes(x=as.numeric(as.character(CDSpos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi)),nrow=2)


#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))


#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Chr. 3 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#ggplot(conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(data=conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b" & conseq_ord_final_filtered_no1c_df$conseq != "STOP_GAINED"),],alpha=1,size=3)+geom_point(data=conseq_ord_final_filtered_no1c_df[which(conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b" & conseq_ord_final_filtered_no1c_df$conseq == "STOP_GAINED"),],alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Chr. 3 position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-3,1))

#RNA d6
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b")),], aes(x=as.numeric(as.character(pos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))

#RNA d20
#ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b")),], aes(x=as.numeric(as.character(pos)), y=post_rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5))

#grid.arrange(ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b")),], aes(x=as.numeric(as.character(pos)), y=rna_score,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(-7.5,2.5)),ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region == "exon 3b")),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(alpha=1,size=3)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_full_simple)+scale_color_manual(values=named_conseq_colors_full_simple)+geom_text(aes(label=alt),hjust=0.5, vjust=0.5, size = 1.4,colour='White')+coord_cartesian(ylim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi)),nrow=2)


VHLrLD_min_syn_score <- min(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "SYNONYMOUS"),]$function_score_final)
VHLrLD_min_syn_norm_RNA_score <- min(final_filtered_no1c_df[which(final_filtered_no1c_df$conseq == "SYNONYMOUS" & final_filtered_no1c_df$rna_score >= -2.0),]$function_score_final)
VHLrLD_min_LB_score <- min(final_filtered_no1c_df[which(final_filtered_no1c_df$clinvar_simple == "Likely benign"),]$function_score_final)
VHLrLD_min_LB_norm_RNA_score <- min(final_filtered_no1c_df[which(final_filtered_no1c_df$clinvar_simple == "Likely benign" & final_filtered_no1c_df$rna_score >= -2.0),]$function_score_final)
VHLrLD_min_pop_count_norm_RNA_score <- min(final_filtered_no1c_df[which(final_filtered_no1c_df$combined_pop_count >= 10 & final_filtered_no1c_df$rna_score >= -2.0),]$function_score_final)

```

RNA scores section
```{r}

#FF Final Fig. 3 - CANFIG - VHLrLD_no1c_RNA_score_by_function_score - 3x4
ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region != "exon 1c")),], aes(x=rna_score, y=function_score_final,color=conseq))+geom_point(alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('RNA score (day 6)')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))
#legend only - Fig. 3a - legend function_by_rna_legend 3x4
ggplot(conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$sge_region != "exon 1c")),], aes(x=rna_score, y=function_score_final,color=conseq))+geom_point(alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('RNA score (day 6)')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels = conseq_labels)+geom_text(data=conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE & (conseq_ord_final_filtered_no1c_df$fs_sig == "FDR < 0.01") & (conseq_ord_final_filtered_no1c_df$rna_score < -2)),],aes(label="*"),hjust=0.5, vjust=0.43, size = 4,colour='Gray')+coord_cartesian(ylim=c(-4,1))

#### RNA Plots

VHLrLD_rna_score_df <- conseq_ord_final_filtered_no1c_df[which(is.na(conseq_ord_final_filtered_no1c_df$Exon) != TRUE),]

ggplot(VHLrLD_rna_score_df, aes(fill=conseq,x=conseq, y=rna_score))+geom_boxplot(aes())+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('SNV consequence')+ylab('RNA Score, day 6')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)

VHLrLD_rna_score_exon2_d6 <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab(NULL)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA D6')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-5.75,3))+scale_y_continuous(breaks=c(-4,-2,0,2))

VHLrLD_rna_score_exon2_d20 <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=post_rna_score,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab(NULL)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA D20')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-5.75,3))+scale_y_continuous(breaks=c(-4,-2,0,2))

VHLrLD_rna_score_exon2_fs <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 2")),], aes(x=as.numeric(as.character(CDSpos)), y=function_score_final,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))+scale_y_continuous(breaks=c(-4,-2,0))

#FF SFig. 5a-c (SFig. 5-2) - 3x6 - VHLrLD_RNA_scores_ex2 #CANFIG
grid.arrange(VHLrLD_rna_score_exon2_d6,VHLrLD_rna_score_exon2_d20,VHLrLD_rna_score_exon2_fs,nrow=3)

VHLrLD_rna_score_exon1_d6 <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 1b")|VHLrLD_rna_score_df$sge_region == "exon 1a"),], aes(x=as.numeric(as.character(CDSpos)), y=rna_score,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab(NULL)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA D6')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-5,2))+scale_y_continuous(breaks=c(-4,-2,0,2))
#RNA d20
VHLrLD_rna_score_exon1_d20 <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 1b")|VHLrLD_rna_score_df$sge_region == "exon 1a"),], aes(x=as.numeric(as.character(CDSpos)), y=post_rna_score,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab(NULL)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA D20')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-5,2))+scale_y_continuous(breaks=c(-4,-2,0,2))

VHLrLD_rna_score_exon1_fs <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 1b")|VHLrLD_rna_score_df$sge_region == "exon 1a"),], aes(x=as.numeric(as.character(CDSpos)), y=function_score_final,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))+scale_y_continuous(breaks=c(-4,-2,0))

#SFig. 5-1 - 3x6 - VHLrLD_RNA_scores_ex1
grid.arrange(VHLrLD_rna_score_exon1_d6,VHLrLD_rna_score_exon1_d20,VHLrLD_rna_score_exon1_fs,nrow=3)

VHLrLD_rna_score_exon3_d6 <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 3a")|VHLrLD_rna_score_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=rna_score,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab(NULL)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA D6')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-3,1))+scale_y_continuous(breaks=c(-4,-2,0,2))+scale_x_continuous(breaks = c(10191507,10191557,10191607,10191646),labels = c('500','550','600','639'))
#RNA d20
VHLrLD_rna_score_exon3_d20 <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 3a")|VHLrLD_rna_score_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=post_rna_score,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab(NULL)+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA D20')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-3,1))+scale_y_continuous(breaks=c(-4,-2,0,2))+scale_x_continuous(breaks = c(10191507,10191557,10191607,10191646),labels = c('500','550','600','639'))

VHLrLD_rna_score_exon3_fs <- ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 3a")|VHLrLD_rna_score_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))+scale_y_continuous(breaks=c(-4,-2,0))+scale_x_continuous(breaks = c(10191507,10191557,10191607,10191646),labels = c('500','550','600','639'))

#SFig. 5-3 - 3x6 - VHLrLD_RNA_scores_ex3
grid.arrange(VHLrLD_rna_score_exon3_d6,VHLrLD_rna_score_exon3_d20,VHLrLD_rna_score_exon3_fs,nrow=3)

#legend only
ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$sge_region == "exon 3a")|VHLrLD_rna_score_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=6),axis.text.y = element_text(size=6),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors,labels=conseq_labels)+coord_cartesian(ylim=c(-4,1))+scale_y_continuous(breaks=c(-4,-2,0))+scale_x_continuous(breaks = c(10191507,10191557,10191607,10191646),labels = c('500','550','600','639'))

#RNAd6 vs RNAd20, plotted together
VHLrLD_rna_score_df$delta_rna <- VHLrLD_rna_score_df$post_rna_score-VHLrLD_rna_score_df$rna_score

ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$conseq != "GAINED")),], aes(x=delta_rna,fill=conseq))+geom_histogram(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('Delta RNA')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)

#CANFIG
ggplot(VHLrLD_rna_score_df[which((VHLrLD_rna_score_df$conseq != "GAINED")),], aes(x=rna_score, y=function_score_final,color=conseq))+geom_point(alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('RNA score (Day 6)')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+geom_segment(data=VHLrLD_rna_score_df[which(VHLrLD_rna_score_df$delta_rna > 0.5),],aes(y=function_score_final,yend=function_score_final,x=rna_score,xend=post_rna_score,color="SPLICE_SITE",alpha=0.3),arrow = arrow(length = unit(0.2, "cm"),type = "closed"))+geom_segment(data=VHLrLD_rna_score_df[which(VHLrLD_rna_score_df$delta_rna < -0.5),],aes(y=function_score_final,yend=function_score_final,x=rna_score,xend=post_rna_score,color="STOP_GAINED",alpha=0.3),arrow = arrow(length = unit(0.2, "cm"),type = "closed"))

#FF Final Supplementary Fig. SFig. 5
grid.arrange(VHLrLD_rna_score_exon1_d6,VHLrLD_rna_score_exon2_d6,VHLrLD_rna_score_exon3_d6,nrow=3)

#FF Final Fig. Supplemenatary Figure 6 RNA score Fig #CANFIG - VHLrLD_RNA6_by_RNA20
ggplot(VHLrLD_rna_score_df[,], aes(y=rna_score, x=post_rna_score,color=conseq))+geom_point(alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('RNA score (Day 20)')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('RNA score (Day 6)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-8,3),ylim=c(-8,3))+geom_abline(slope=1,intercept=0,lty=3)

cor(x=VHLrLD_rna_score_df$rna_score,y=VHLrLD_rna_score_df$post_rna_score,method = "spearman")
cor(x=VHLrLD_rna_score_df$rna_score,y=VHLrLD_rna_score_df$post_rna_score,method = "pearson")

#FF Final Fig. Supplemenatary Figure 6 RNA score Fig #CANFIG - VHLrLD_fs_by_delta_rna
ggplot(VHLrLD_rna_score_df, aes(y=function_score_final, x=delta_rna,color=conseq))+geom_point(alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab('Delta RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(-2,4),ylim=c(-4,1))
#legend only expor 4x6 Final Fig. SFig. 5 Supp.
ggplot(VHLrLD_rna_score_df, aes(y=function_score_final, x=delta_rna,color=conseq))+geom_point(alpha=1,size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Delta RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors,labels=conseq_labels)+coord_cartesian(xlim=c(-2,4),ylim=c(-4,1))

#all clinvar colours #CANFIG
#ggplot(clinvar_ord_final_filtered_no1c_df, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi),ylim=c(0,550))

#pathogenic, lp vs benign, lb vs vus, simplified
#ggplot(clinvar_ord_final_filtered_no1c_df, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path.lp_benign.lb_confl.vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi),ylim=c(0,550))

#exclude absent, #CANFIG
#ggplot(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$clinvar_simple != "absent"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#exclude absent, uncertain, conflicting #CANFIG
#ggplot(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$clinvar_simple != "absent" & clinvar_ord_final_filtered_no1c_df$clinvar_simple != "Uncertain significance" & clinvar_ord_final_filtered_no1c_df$clinvar_simple != "Conflicting interpretations of pathogenicity"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#exclude none, all colours
#ggplot(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$clinvar_simple != "nt"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi),ylim=c(0,550))

#plots to highlight new data added
ggplot(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$clinvar_simple != "absent"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi),ylim=c(0,550))
#plots to highlight new data added
ggplot(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$clinvar_simple != "akaljskjlhent"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi),ylim=c(0,550))

#Lone likely benign variant scoring lowly has a low RNA score, exceedingly rare, syn.
ggplot(final_filtered_no1c_df[which(final_filtered_no1c_df$clinvar_simple == "Likely benign" & is.na(final_filtered_no1c_df$Exon)!=TRUE),], aes(x=function_score_final))+geom_point(alpha=1,aes(color=conseq,y=rna_score,size=combined_pop_count))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$clinvar_simple != "aakslhdnt"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_benign.lb)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLrLD_min_LB_norm_RNA_score,lty=2)


##VHLdb plots (in clinvar colours)
ggplot(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$vhldb_rcc_count >= 1),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

```

MERGE IN CBIOPORTAL DATA
```{r}

cBioPortal_colours = c("Absent" = "#D0D0D0", "Other" = "#D3A44C", "Clear Cell Renal Cell Carcinoma" = "#81CD42", "Renal Cell Carcinoma Other" = "#459ACA", "Chromophobe Renal Cell Carcinoma" = "#35718F", "Papillary Renal Cell Carcinoma" = "#324249", "Pheochromocytoma" = "#51822B")
VHLd_types = c("NA" = "#D0D0D0", "Type 1" = "#950944", "Type 2" = "#041F47", "Type 2A" = "#1A4687", "Type 2B" = "#3F6FB7", "Type 2C" = "#7FACF0")
cBioPortal_colours_present = c("Other" = "#D3A44C", "Clear Cell Renal Cell Carcinoma" = "#81CD42", "Renal Cell Carcinoma Other" = "#459ACA", "Chromophobe Renal Cell Carcinoma" = "#35718F", "Papillary Renal Cell Carcinoma" = "#324249", "Pheochromocytoma" = "#51822B")

cBioPortal_colours_gf_present = c("Other" = "#332288", "Clear Cell Renal Cell Carcinoma" = "#F3001C", "Renal Cell Carcinoma Other" = "#FD9100", "Chromophobe Renal Cell Carcinoma" = "#FD6000", "Papillary Renal Cell Carcinoma" = "#99006B", "Pheochromocytoma" = "#00BA74")

cBioPortal_colours_gf = c("Clear Cell Renal Cell Carcinoma" = "#E0004B", "Papillary Renal Cell Carcinoma" = "#99006B","Chromophobe Renal Cell Carcinoma" = "#FFBB33","Renal Cell Carcinoma Other" = "#F580A7","Pheochromocytoma" = "#00BA74","Other" = "#4F71E1","Absent" = "#D0D0D0","Renal Cell Carcinoma, NOS" = "#F580A7","Pancreatic Neuroendocrine" = "#CDF800")

cBioPortal_colours_simple = c("Absent" = "#D0D0D0", "Other" = "#D3A44C", "ccRCC" = "#81CD42", "RCC Other" = "#459ACA", "Pheo" = "#51822B")

cBioPortal_colours_gf_simple = c("Absent" = "#D0D0D0", "Other" = "#4F71E1", "ccRCC" = "#E0004B", "RCC Other" = "#FFBB33", "Pheo" = "#00BA74", "PNET" = "#CDF800")

germline_var_number_colours = c("0" = "#D0D0D0", "1" = "#D9C2A5", "2" = "#E8A95D", "3" = "#EF9C38", "4" = "#CF6E22", "5" = "#B24710", "6" = "9B2600")
hotspot_colours = c("no" = "#FB4750", "yes" = "#70DE65")
cBP_entries_colours = list("0" = "#D0D0D0", "1" = "#17285C", "2" = "#2C4D57", "3" = "#427353", "4" = "#58994E", "5" = "#76B053", "6" = "#9EB962", "7" = "#C7C371", "8" = "#ECCC80", "10" = "#BB835D", "13" = "#BF5051")
cBP_entries_colours_simple = list("0" = "#D0D0D0", "1" = "#17285C", "2-5" = "#58994E", "6-9" = "#ECCC80", "10+" = "#BF5051")
SIFT_colours = c("tolerated" = "#58994E", "deleterious" = "#BF5051")
PolyPhen_colours = c("benign" = "#005493", "possibly_damaging" = "#ECCC80", "probably_damaging" = "#BF5051")

VHL_cBioPortal_curated <- read.csv("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/20220728_cBioPortal_curated.csv")

# new cBioPortal df containing only SNVs and remove unnecessary columns
unique(VHL_cBioPortal_curated$Variant.Type)
#edited by GF to include more metrics
cBioPortal_SNVs <- VHL_cBioPortal_curated %>% 
  dplyr::filter(Variant.Type == "SNP") %>% 
  select(Study.of.Origin, Sample.ID, Cancer.Type, Cancer.Type.Detailed, Protein.Change, Annotation, Functional.Impact, Mutation.Type, Start.Pos, Ref, Var, HGVSg, HGVSc, Allele.Freq..T., X..Mut.in.Sample, Diagnosis.Age, Metastatic.Site, Oncotree.Code, Primary.Tumor.Site, Sex, COSMIC, Number.of.Samples.Per.Patient, Sample.Type, TMB..nonsynonymous., Patient.s.Vital.Status, Overall.Survival.Status, Birth.from.Initial.Pathologic.Diagnosis.Date, Months.of.disease.specific.survival, Neoplasm.Histologic.Grade, Other.Patient.ID, Progress.Free.Survival..Months., Race.Category)

#bring in SNVs processed with python script to aggergate data across duplicates
cBioPortal_unique_SNVs <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/2022102_VHL_cBioPortal_SNVs_dedup.txt", sep="\t", header=TRUE)


#Key point:  two data-frames,"cBPu" indicates cBP data has been collapsed such that 1 entry per SNV (entry has info across all cBP samples); "cBP" indicates 1 entry per cBP entry.
final_filtered_no1c_df_cBP <- left_join(x = final_filtered_no1c_df,
                                   y = cBioPortal_SNVs,
                                   by = c("pos" = "Start.Pos",
                                          "Ref" = "Ref",
                                          "alt" = "Var"))
                                          
final_filtered_no1c_df_cBPu <- left_join(x = final_filtered_no1c_df,
                                   y = cBioPortal_unique_SNVs,
                                   by = c("pos" = "Start.Pos",
                                          "Ref" = "Ref",
                                          "alt" = "Var"))


final_filtered_no1c_df_cBP_RCC <- final_filtered_no1c_df_cBP %>% 
  mutate(
    Cancer_type_filtered = factor(case_when(
      grepl(x = Cancer.Type.Detailed, pattern = "Renal Clear Cell Carcinoma") ~ "Clear Cell Renal Cell Carcinoma",
      grepl(x = Cancer.Type.Detailed, pattern = "Chromophobe Renal Cell Carcinoma") ~ "Chromophobe Renal Cell Carcinoma",
      grepl(x = Cancer.Type.Detailed, pattern = "Papillary Renal Cell Carcinoma") | grepl(x = Cancer.Type.Detailed, pattern = "Renal Non-Clear Cell Carcinoma") ~ "Papillary Renal Cell Carcinoma",
      grepl(x = Cancer.Type.Detailed, pattern = "Pheochromocytoma") ~ "Pheochromocytoma",
      grepl(x = Cancer.Type.Detailed, pattern = "Kidney Renal Cell Carcinoma Other") ~ "Renal Cell Carcinoma Other",
      grepl(x = Cancer.Type.Detailed, pattern = "Unclassified Renal Cell Carcinoma") ~ "Renal Cell Carcinoma Other",
      grepl(x = Cancer.Type.Detailed, pattern = "Unclassified Kidney Renal Cell Carcinoma") ~ "Renal Cell Carcinoma Other",
      grepl(x = Cancer.Type.Detailed, pattern = "Renal Cell Carcinoma") ~ "Renal Cell Carcinoma Other",
      grepl(x = Cancer.Type.Detailed, pattern = "Pancreatic Neuroendocrine Tumor") ~ "Pancreatic Neuroendocrine",
      grepl(x = Cancer.Type.Detailed, pattern = ".+") ~ "Other")))
      
final_filtered_no1c_df_cBPu_RCC <- final_filtered_no1c_df_cBPu %>% 
  mutate(
    Cancer_type_filtered = factor(case_when(
      grepl(x = Cancer.Type.Detailed, pattern = "Renal Clear Cell Carcinoma") ~ "Clear Cell Renal Cell Carcinoma",
      grepl(x = Cancer.Type.Detailed, pattern = "Chromophobe Renal Cell Carcinoma") ~ "Chromophobe Renal Cell Carcinoma",
      grepl(x = Cancer.Type.Detailed, pattern = "Papillary Renal Cell Carcinoma") | grepl(x = Cancer.Type.Detailed, pattern = "Renal Non-Clear Cell Carcinoma") ~ "Papillary Renal Cell Carcinoma",
      grepl(x = Cancer.Type.Detailed, pattern = "Pheochromocytoma") ~ "Pheochromocytoma",
      grepl(x = Cancer.Type.Detailed, pattern = "Kidney Renal Cell Carcinoma Other") ~ "Renal Cell Carcinoma Other",
      grepl(x = Cancer.Type.Detailed, pattern = "Unclassified Renal Cell Carcinoma") ~ "Renal Cell Carcinoma Other",
      grepl(x = Cancer.Type.Detailed, pattern = "Unclassified Kidney Renal Cell Carcinoma") ~ "Renal Cell Carcinoma Other",
      grepl(x = Cancer.Type.Detailed, pattern = "Renal Cell Carcinoma") ~ "Renal Cell Carcinoma Other",
      grepl(x = Cancer.Type.Detailed, pattern = "Pancreatic Neuroendocrine Tumor") ~ "Pancreatic Neuroendocrine",
      grepl(x = Cancer.Type.Detailed, pattern = ".+") ~ "Other")))

# change blank rows to 'Absent'
final_filtered_no1c_df_cBP_RCC <- final_filtered_no1c_df_cBP_RCC %>% 
  mutate(Cancer_type_filtered = fct_explicit_na(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered, na_level = "Absent"))
final_filtered_no1c_df_cBPu_RCC <- final_filtered_no1c_df_cBPu_RCC %>% 
  mutate(Cancer_type_filtered = fct_explicit_na(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered, na_level = "Absent"))


final_filtered_no1c_df_cBP_RCC <- final_filtered_no1c_df_cBP_RCC %>%
  mutate(Cancer_filtered_simple = case_when(
    grepl(x = Cancer_type_filtered, pattern = "Absent") ~ "Absent",
    grepl(x = Cancer_type_filtered, pattern = "Other") ~ "Other",
    grepl(x = Cancer_type_filtered, pattern = "Renal Cell Carcinoma Other") ~ "RCC Other",
    grepl(x = Cancer_type_filtered, pattern = "Papillary Renal Cell Carcinoma") ~ "RCC Other",
    grepl(x = Cancer_type_filtered, pattern = "Chromophobe Renal Cell Carcinoma") ~ "RCC Other",
    grepl(x = Cancer_type_filtered, pattern = "Pheochromocytoma") ~ "Pheo",
    grepl(x = Cancer_type_filtered, pattern = "Pancreatic Neuroendocrine") ~ "PNET",
    grepl(x = Cancer_type_filtered, pattern = "Clear Cell Renal Cell Carcinoma") ~ "ccRCC"))

final_filtered_no1c_df_cBPu_RCC <- final_filtered_no1c_df_cBPu_RCC %>%
  mutate(Cancer_filtered_simple = case_when(
    grepl(x = Cancer_type_filtered, pattern = "Absent") ~ "Absent",
    grepl(x = Cancer_type_filtered, pattern = "Other") ~ "Other",
    grepl(x = Cancer_type_filtered, pattern = "Renal Cell Carcinoma Other") ~ "RCC Other",
    grepl(x = Cancer_type_filtered, pattern = "Papillary Renal Cell Carcinoma") ~ "RCC Other",
    grepl(x = Cancer_type_filtered, pattern = "Chromophobe Renal Cell Carcinoma") ~ "RCC Other",
    grepl(x = Cancer_type_filtered, pattern = "Pheochromocytoma") ~ "Pheo",
    grepl(x = Cancer_type_filtered, pattern = "Pancreatic Neuroendocrine") ~ "PNET",
    grepl(x = Cancer_type_filtered, pattern = "Clear Cell Renal Cell Carcinoma") ~ "ccRCC"))

final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered = factor(
final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered,
  levels = c(
    "Absent",
    "Other",
    "Pheochromocytoma",
    "Pancreatic Neuroendocrine",
    "Renal Cell Carcinoma Other",
    "Papillary Renal Cell Carcinoma",
    "Chromophobe Renal Cell Carcinoma",
    "Clear Cell Renal Cell Carcinoma"
    ))

final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered <- factor(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered, levels = c(
    "Absent",
    "Other",
    "Pheochromocytoma",
    "Pancreatic Neuroendocrine",
    "Renal Cell Carcinoma Other",
    "Papillary Renal Cell Carcinoma",
    "Chromophobe Renal Cell Carcinoma",
    "Clear Cell Renal Cell Carcinoma"
    ))

#total cBP clear cell RCC will be counts of "Renal.Clear.Cell.Carcinoma" + "Renal.Clear.Cell.Carcinoma.with.Sarcomatoid.Features"
final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC <- final_filtered_no1c_df_cBPu_RCC$Renal.Clear.Cell.Carcinoma + final_filtered_no1c_df_cBPu_RCC$Renal.Clear.Cell.Carcinoma.with.Sarcomatoid.Features

#total_cBP_entries meant to exclude Pheo before on grounds of not necessarily being malignant -- here correcting to include all Pheo counts in total_cBP_entries count
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Pheochromocytoma >0),]$total_cBP_entries <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Pheochromocytoma >0),]$independent_samples

#In cBPu df, a single cancer type "Cancer_type_single" is assigned with the following priority order, based on relevance to VHL Disease:  ccRCC, pheo, pnet, Chromo, Pap, RCC-NOS, Other, Absent
final_filtered_no1c_df_cBPu_RCC$Cancer_type_single <- "Absent"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$total_cBP_entries > 0),]$Cancer_type_single <- "Other"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Kidney.Renal.Cell.Carcinoma.Other > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Kidney.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Kidney.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Renal.Non.Clear.Cell.Carcinoma > 0),]$Cancer_type_single <- "Renal Cell Carcinoma, NOS"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Papillary.Renal.Cell.Carcinoma > 0),]$Cancer_type_single <- "Papillary Renal Cell Carcinoma" 
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Chromophobe.Renal.Cell.Carcinoma.Ç.. > 0),]$Cancer_type_single <- "Chromophobe Renal Cell Carcinoma"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Pancreatic.Neuroendocrine.Tumor > 0),]$Cancer_type_single <- "Pancreatic Neuroendocrine"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Pheochromocytoma > 0),]$Cancer_type_single <- "Pheochromocytoma"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Renal.Clear.Cell.Carcinoma > 0| final_filtered_no1c_df_cBPu_RCC$Renal.Clear.Cell.Carcinoma.with.Sarcomatoid.Features > 0 ),]$Cancer_type_single <- "Clear Cell Renal Cell Carcinoma"

final_filtered_no1c_df_cBPu_RCC$Cancer_type_single <- factor(final_filtered_no1c_df_cBPu_RCC$Cancer_type_single, levels = c(
    "Absent",
    "Other",
    "Pheochromocytoma",
    "Pancreatic Neuroendocrine",
    "Renal Cell Carcinoma, NOS",
    "Papillary Renal Cell Carcinoma",
    "Chromophobe Renal Cell Carcinoma",
    "Clear Cell Renal Cell Carcinoma"
    ))


#FF CANFIG - Final Fig. 4b - cBioPortal export 3x5 as VHLrLD_fs_histo_cBP_single
ggplot(final_filtered_no1c_df_cBPu_RCC,
                                    aes(x = function_score_final)) +
  geom_histogram(alpha = 1,
                 bins = 30,
                 aes(fill = Cancer_type_single)) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),legend.position = 'None')+
  xlab("Function score") +
  ylab('SNVs')+
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_fill_manual(values = cBioPortal_colours_gf) +
  coord_cartesian(xlim = c(-4, 1))

#FF CANFIG - Final Fig. 4b inset - VHLrLD_fs_histo_by_cBP_single_present - single score per SNV
ggplot(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered != "Absent"),],
                                    aes(x = function_score_final)) +
  geom_histogram(alpha = 1,
                 aes(fill = Cancer_type_single)) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),legend.position = 'None') +
  xlab("Function score") +
  ylab('SNVs')+
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_fill_manual(values = cBioPortal_colours_gf) +
  coord_cartesian(xlim = c(-4, 1))

#alt CANFIG - Final Fig. 4b - cBioPortal (scores for SNVs across all samples, 1 entry per SAMPLE)
ggplot(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered != "Absent"),],
                                    aes(x = function_score_final)) +
  geom_histogram(bins = 20,
                 aes(fill = Cancer_type_filtered)) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),legend.position = 'None') +
        #) +
  xlab("Function score") +
  ylab('SNVs')+
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_fill_manual(values = cBioPortal_colours_gf) +
  coord_cartesian(xlim = c(-4, 1))

##depricated CANFIG -- , COSMIC counts x FS  (unique SNVs)
ggplot(final_filtered_no1c_df_cBPu_RCC,
                                    aes(x = function_score_final,y=COSMIC,colour = Cancer_type_single)) +
  geom_jitter(alpha = 1) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),legend.position = 'None') +
  xlab("Function score") +
  ylab('COSMIC') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values = cBioPortal_colours_gf) +
  coord_cartesian(xlim = c(-4, 1))

#FF CANFIG -- Final Fig. 4c , cBP total counts x FS  (unique SNVs)
ggplot(final_filtered_no1c_df_cBPu_RCC,
                                    aes(x = function_score_final,y=total_cBP_entries,colour = Cancer_type_single)) +
  geom_jitter(alpha = 1,height=0.15,width=0) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),legend.position = 'None') +
  xlab("Function score") +
  ylab('cBioPortal entries') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values = cBioPortal_colours_gf) +
  coord_cartesian(xlim = c(-4, 1),ylim = c(0,8))

#CANFIG -- , cBP rcc counts x FS  (unique SNVs)
ggplot(final_filtered_no1c_df_cBPu_RCC,
                                    aes(x = function_score_final,y=total_cBP_rcc,colour = Cancer_type_single)) +
  geom_jitter(alpha = 1,width=0,height=0.15) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),legend.position = 'None') +
  xlab("Function score") +
  ylab('cBioPortal RCC samples') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values = cBioPortal_colours_gf) +
  coord_cartesian(xlim = c(-4, 1))


#Diagnosis.Age x FS ccRCC (all cBioPortal entries for which it is provided)
ggplot(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma"),],
                                    aes(x = function_score_final,y=Diagnosis.Age,colour = clinvar_simple)) +
  geom_jitter(alpha = 1) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
        #legend.position = 'Bottom') +
  xlab("Function score") +
  ylab('Age Dx') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values = named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus) +
  coord_cartesian(xlim = c(-4, 1))

final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered <- factor(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered, levels = c("Absent","Clear Cell Renal Cell Carcinoma","Other","Pheochromocytoma","Pancreatic Neuroendocrine","Renal Cell Carcinoma Other","Papillary Renal Cell Carcinoma","Chromophobe Renal Cell Carcinoma"))

#FF CANFIG - Fig. 4 Figure 4, Allele Freq x FS (all cBioPortal entries for which it is provided)
ggplot(final_filtered_no1c_df_cBP_RCC[which( (final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered != "Absent") & !is.na(final_filtered_no1c_df_cBP_RCC$Allele.Freq..T.)),],
                                    aes(x = function_score_final,y=Allele.Freq..T.,colour = Cancer_type_filtered)) +
  geom_jitter(alpha = 1,width=0,height=0) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank(),legend.position = 'Bottom') +
  xlab("Function score") +
  ylab('Allele Freq.') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values = cBioPortal_colours_gf) +
  coord_cartesian(xlim = c(-4, 1))

#Metastasis? x FS (all cBioPortal entries for which it is provided)
final_filtered_no1c_df_cBP_RCC$Met_boolean <- TRUE
final_filtered_no1c_df_cBP_RCC[is.na(final_filtered_no1c_df_cBP_RCC$Metastatic.Site),]$Met_boolean <- NA
final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Metastatic.Site == ''),]$Met_boolean <- FALSE

length(which(final_filtered_no1c_df_cBP_RCC$Met_boolean == TRUE))
length(which(final_filtered_no1c_df_cBP_RCC$Met_boolean == FALSE))
length(which(is.na(final_filtered_no1c_df_cBP_RCC$Met_boolean) == TRUE))

ggplot(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma"),],
                                    aes(x = function_score_final,y=Met_boolean,colour = Cancer_type_filtered)) +
  geom_jitter(alpha = 1) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
        #legend.position = 'Bottom') +
  xlab("Function score") +
  ylab('Metastasis') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values = cBioPortal_colours_gf) +
  coord_cartesian(xlim = c(-4, 1))

median(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma"),]$function_score_final)
median(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Absent"),]$function_score_final)
median(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Other"),]$function_score_final)
median(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Pheochromocytoma"),]$function_score_final)
median(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Papillary Renal Cell Carcinoma"),]$function_score_final)
median(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Chromophobe Renal Cell Carcinoma"),]$function_score_final)
median(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Renal Cell Carcinoma Other"),]$function_score_final)

#FS Fig. 4 statistics CANFIG -- all renal cancer variants sign. lower than absent variants
wilcox.test(x = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma"),]$function_score_final, y = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered == "Absent"),]$function_score_final,paired = FALSE,alternative = "two.sided")

wilcox.test(x = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered == "Chromophobe Renal Cell Carcinoma"),]$function_score_final, y = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered == "Absent"),]$function_score_final,paired = FALSE,alternative = "two.sided")

wilcox.test(x = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered == "Papillary Renal Cell Carcinoma"),]$function_score_final, y = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered == "Absent"),]$function_score_final,paired = FALSE,alternative = "two.sided")

wilcox.test(x = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered == "Renal Cell Carcinoma Other"),]$function_score_final, y = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_filtered == "Absent"),]$function_score_final,paired = FALSE,alternative = "two.sided")


#spearman test on COSMIC Occurences (note, this is per AA position, not per SNV)
spearman.test.COSMIC.fs <- cor.test(-final_filtered_no1c_df_cBPu_RCC[!is.na(final_filtered_no1c_df_cBPu_RCC$COSMIC),]$function_score_final,final_filtered_no1c_df_cBPu_RCC[!is.na(final_filtered_no1c_df_cBPu_RCC$COSMIC),]$COSMIC,method='spearman')
spearman.test.COSMIC.fs

#FS Fig. 4 
spearman.test.cBP_total.fs <- cor.test(-final_filtered_no1c_df_cBPu_RCC[!is.na(final_filtered_no1c_df_cBPu_RCC$total_cBP_entries),]$function_score_final,final_filtered_no1c_df_cBPu_RCC[!is.na(final_filtered_no1c_df_cBPu_RCC$total_cBP_entries),]$total_cBP_entries,method='spearman')
spearman.test.cBP_total.fs

spearman.test.cBP_rcc.fs <- cor.test(-final_filtered_no1c_df_cBPu_RCC[!is.na(final_filtered_no1c_df_cBPu_RCC$total_cBP_rcc),]$function_score_final,final_filtered_no1c_df_cBPu_RCC[!is.na(final_filtered_no1c_df_cBPu_RCC$total_cBP_rcc),]$total_cBP_rcc,method='spearman')
spearman.test.cBP_rcc.fs

#No correlation between score and rcc entries among samples with at least 1 rcc entry.
spearman.test.cBP_rcc1.fs <- cor.test(-final_filtered_no1c_df_cBPu_RCC[-(!is.na(final_filtered_no1c_df_cBPu_RCC$total_cBP_rcc)) & final_filtered_no1c_df_cBPu_RCC$total_cBP_rcc > 0,]$function_score_final, final_filtered_no1c_df_cBPu_RCC[(!is.na(final_filtered_no1c_df_cBPu_RCC$total_cBP_rcc))& final_filtered_no1c_df_cBPu_RCC$total_cBP_rcc > 0,]$total_cBP_rcc,method='spearman')
spearman.test.cBP_rcc1.fs

#FS Fig. 4 - Allele frequency by indepdnent cBP entry (not uniquie df)
spearman.test.AF.fs <- cor.test(-final_filtered_no1c_df_cBP_RCC[!is.na(final_filtered_no1c_df_cBP_RCC$Allele.Freq..T.),]$function_score_final,final_filtered_no1c_df_cBP_RCC[!is.na(final_filtered_no1c_df_cBP_RCC$Allele.Freq..T.),]$Allele.Freq..T.,method='spearman')
spearman.test.AF.fs

#no significant met, survival, age of dx WRSTs
wilcox.test(x = final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma" & final_filtered_no1c_df_cBP_RCC$function_score_final < VHLrLD_min_LB_norm_RNA_score),]$Months.of.disease.specific.survival, y = final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma" & final_filtered_no1c_df_cBP_RCC$function_score_final >= VHLrLD_min_LB_norm_RNA_score),]$Months.of.disease.specific.survival,paired = FALSE,alternative = "two.sided")

wilcox.test(x = final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma" & final_filtered_no1c_df_cBP_RCC$function_score_final < VHLrLD_min_LB_norm_RNA_score),]$Diagnosis.Age, y = final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma" & final_filtered_no1c_df_cBP_RCC$function_score_final >= VHLrLD_min_LB_norm_RNA_score),]$Diagnosis.Age,paired = FALSE,alternative = "two.sided")

wilcox.test(x = final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Met_boolean == TRUE),]$function_score_final, y = final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Met_boolean == FALSE),]$function_score_final,paired = FALSE,alternative = "two.sided")

wilcox.test(x = final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma" & final_filtered_no1c_df_cBP_RCC$function_score_final < VHLrLD_min_LB_norm_RNA_score),]$TMB..nonsynonymous., y = final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma" & final_filtered_no1c_df_cBP_RCC$function_score_final >= VHLrLD_min_LB_norm_RNA_score),]$TMB..nonsynonymous.,paired = FALSE,alternative = "two.sided")

#Disease Specific Survival x FS ccRCC
ggplot(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma"),],
                                    aes(x = function_score_final,y=Months.of.disease.specific.survival,colour = clinvar_simple)) +
  geom_jitter(alpha = 1) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
        #legend.position = 'Bottom') +
  xlab("Function score") +
  ylab('Months of disease specific survival') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus) +
  coord_cartesian(xlim = c(-4, 1))

#Overall Survival Status vs. FS
ggplot(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma"),],
                                    aes(x = function_score_final,y=Overall.Survival.Status,colour = clinvar_simple)) +
  geom_jitter(alpha = 1) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
        #legend.position = 'Bottom') +
  xlab("Function score") +
  ylab('Survival') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus) +
  coord_cartesian(xlim = c(-4, 1))

ggplot(final_filtered_no1c_df_cBP_RCC[which(final_filtered_no1c_df_cBP_RCC$Cancer_type_filtered == "Clear Cell Renal Cell Carcinoma"),],
                                    aes(x = function_score_final,y=TMB..nonsynonymous.,color=conseq)) +
  geom_jitter(alpha = 1) +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(legend.title = element_blank(),
        legend.key = element_blank()) +
        #legend.position = 'Bottom') +
  xlab("Function score") +
  ylab('TMB, nonsyn.') +
  theme(panel.background = element_rect(fill = 'white', colour = 'white')) +
  theme(text = element_text(size = 12),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        strip.background = element_blank()
        ) +
  scale_colour_manual(values=named_conseq_colors) +
  coord_cartesian(xlim = c(-4, 1))

### analysis on overlap between LoF, rcc, ClinVar path/lp, vhlDB
VHLrLD_LoF_set <- which(final_filtered_no1c_df_cBPu_RCC$fdr < 0.01)

VHLrLD_LoF_clinvar_path_set <- which(final_filtered_no1c_df_cBPu_RCC$fdr < 0.01 & (final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic"))

VHLrLD_LoF_clinvar_path_vus_set <- which(final_filtered_no1c_df_cBPu_RCC$fdr < 0.01 & (final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Uncertain significance" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Conflicting interpretations of pathogenicity"))

VHLrLD_LoF_RCC_set <- which(final_filtered_no1c_df_cBPu_RCC$fdr < 0.01 & final_filtered_no1c_df_cBPu_RCC$total_cBP_rcc > 0)
VHLrLD_LoF_vhldb_set <- which(final_filtered_no1c_df_cBPu_RCC$fdr < 0.01 & final_filtered_no1c_df_cBPu_RCC$vhldb_syndrome_count > 0)

VHLrLD_LoF_cv_rcc <- union(VHLrLD_LoF_clinvar_path_vus_set,VHLrLD_LoF_RCC_set)
VHLrLD_LoF_cv_rcc_vhldb <- union(VHLrLD_LoF_cv_rcc,VHLrLD_LoF_vhldb_set)

median(final_filtered_no1c_df_cBPu_RCC[VHLrLD_LoF_set,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[VHLrLD_LoF_clinvar_path_set,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[VHLrLD_LoF_clinvar_path_vus_set,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[VHLrLD_LoF_RCC_set,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[VHLrLD_LoF_vhldb_set,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[VHLrLD_LoF_cv_rcc_vhldb,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[setdiff(VHLrLD_LoF_set, VHLrLD_LoF_cv_rcc_vhldb),]$function_score_final)

#among LoF (fdr < 0.01) variants, function scores are significantly lower for variants seen in disease db #86 SNVs not reported in any disease database, median -0.78
wilcox.test(x = final_filtered_no1c_df_cBPu_RCC[VHLrLD_LoF_cv_rcc_vhldb,]$function_score_final, y = final_filtered_no1c_df_cBPu_RCC[setdiff(VHLrLD_LoF_set, VHLrLD_LoF_cv_rcc_vhldb),]$function_score_final ,paired = FALSE,alternative = "two.sided")
median(final_filtered_no1c_df_cBPu_RCC[VHLrLD_LoF_cv_rcc_vhldb,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[setdiff(VHLrLD_LoF_set, VHLrLD_LoF_cv_rcc_vhldb),]$function_score_final)
  
#View(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$total_cBP_entries > 1 & final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC < 1),c("cHGVS","pHGVS","COSMIC","total_cBP_entries","total_cBP_rcc","total_cBP_ccRCC","total_cBP_other","function_score_final","combined_pop_count","clinvar_simple")])
        
#fixing Cancer_filtered_simple to follow same heirarchy, e.g. "other" will not include variants seen in any VHL Disease type:
final_filtered_no1c_df_cBPu_RCC$Cancer_filtered_simple <- "Absent"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$total_cBP_entries > 0),]$Cancer_filtered_simple <- "Other"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Kidney.Renal.Cell.Carcinoma.Other > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Kidney.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Kidney.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Renal.Non.Clear.Cell.Carcinoma > 0),]$Cancer_filtered_simple <- "RCC Other"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Papillary.Renal.Cell.Carcinoma > 0),]$Cancer_filtered_simple <- "RCC Other" 
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Chromophobe.Renal.Cell.Carcinoma.Ç.. > 0),]$Cancer_filtered_simple <- "RCC Other"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Pancreatic.Neuroendocrine.Tumor > 0),]$Cancer_filtered_simple <- "PNET"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Pheochromocytoma > 0),]$Cancer_filtered_simple <- "Pheo"
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Renal.Clear.Cell.Carcinoma > 0| final_filtered_no1c_df_cBPu_RCC$Renal.Clear.Cell.Carcinoma.with.Sarcomatoid.Features > 0 ),]$Cancer_filtered_simple <- "ccRCC"

#CANFIG - alternative - present only, simple classification by ccRCC, Other RCC, Pheo, Other -
#"Variants seen in multiple tumour types are coloured the type of the most common VHL-associated cancer."
ggplot(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_filtered_simple != "Absent"),],aes(x = function_score_final))+geom_histogram(alpha = 1,bins = 30,aes(fill = Cancer_filtered_simple)) + theme(panel.background = element_rect(fill = 'white', colour = 'white')) + theme(legend.title = element_blank(),legend.key = element_blank())+xlab("Function score") +ylab('SNVs') +theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(text = element_text(size = 12),axis.text.y = element_text(size = 12),axis.text.x = element_text(size = 12),strip.background = element_blank())+scale_fill_manual(values = cBioPortal_colours_gf_simple)+coord_cartesian(xlim = c(-4, 1))

#separate data frames for each type, rbind back together to make 1 plot showing scores by type:
VHLrLD_cancer_type_df_absent <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_single == "Absent"),]
VHLrLD_cancer_type_df_absent$cBP_cancer_type <- "Absent"
VHLrLD_cancer_type_df_other <- final_filtered_no1c_df_cBPu_RCC[which((final_filtered_no1c_df_cBPu_RCC$total_cBP_other > 0) &  (final_filtered_no1c_df_cBPu_RCC$total_cBP_other != final_filtered_no1c_df_cBPu_RCC$Pancreatic.Neuroendocrine.Tumor)),]
VHLrLD_cancer_type_df_other$cBP_cancer_type <- "Other"
VHLrLD_cancer_type_df_papillary <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Papillary.Renal.Cell.Carcinoma > 0),]
VHLrLD_cancer_type_df_papillary$cBP_cancer_type <- "Papillary Renal Cell Carcinoma"
VHLrLD_cancer_type_df_rccnos <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Kidney.Renal.Cell.Carcinoma.Other > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Kidney.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Unclassified.Kidney.Renal.Cell.Carcinoma > 0 | final_filtered_no1c_df_cBPu_RCC$Renal.Non.Clear.Cell.Carcinoma > 0),]
VHLrLD_cancer_type_df_rccnos$cBP_cancer_type <- "Renal Cell Carcinoma, NOS"
VHLrLD_cancer_type_df_chromophobe <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Chromophobe.Renal.Cell.Carcinoma.Ç.. > 0),]
VHLrLD_cancer_type_df_chromophobe$cBP_cancer_type <- "Chromophobe Renal Cell Carcinoma"
VHLrLD_cancer_type_df_pheo <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Pheochromocytoma > 0),]
VHLrLD_cancer_type_df_pheo$cBP_cancer_type <- "Pheochromocytoma"
VHLrLD_cancer_type_df_ccrcc <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Renal.Clear.Cell.Carcinoma > 0| final_filtered_no1c_df_cBPu_RCC$Renal.Clear.Cell.Carcinoma.with.Sarcomatoid.Features > 0 ),]
VHLrLD_cancer_type_df_ccrcc$cBP_cancer_type <- "Clear Cell Renal Cell Carcinoma"
VHLrLD_cancer_type_df_pnet <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Pancreatic.Neuroendocrine.Tumor > 0),]
VHLrLD_cancer_type_df_pnet$cBP_cancer_type <- "Pancreatic Neuroendocrine"

VHLrLD_cancer_type_df <- (rbind(VHLrLD_cancer_type_df_absent,VHLrLD_cancer_type_df_other,VHLrLD_cancer_type_df_papillary,VHLrLD_cancer_type_df_rccnos,VHLrLD_cancer_type_df_chromophobe,VHLrLD_cancer_type_df_pheo,VHLrLD_cancer_type_df_ccrcc,VHLrLD_cancer_type_df_pnet))

#plot score distribution by cBP_cancer_type (showing all unique variants seen in each cancer category with some variants seen in multiple categories)
VHLrLD_cancer_type_df$cBP_cancer_type <- factor(VHLrLD_cancer_type_df$cBP_cancer_type, levels = c("Clear Cell Renal Cell Carcinoma","Chromophobe Renal Cell Carcinoma","Papillary Renal Cell Carcinoma","Renal Cell Carcinoma, NOS","Pheochromocytoma","Pancreatic Neuroendocrine","Other","Absent"))

VHLrLD_cancer_type_labels <- c("Clear Cell Renal Cell Carcinoma" = "ccRCC","Chromophobe Renal Cell Carcinoma" = "chRCC","Papillary Renal Cell Carcinoma" = "pRCC","Renal Cell Carcinoma, NOS" = "RCC-NOS","Pheochromocytoma" = "Pheo","Pancreatic Neuroendocrine" = "PNET","Other" = "Other","Absent" = "Absent")

#CANFIG - Fig. 4 boxplot of function scores by cBP cancer type (all unique SNV-cancer pairs)
ggplot(VHLrLD_cancer_type_df, aes(x=cBP_cancer_type, y=function_score_final))+geom_boxplot(aes(),outlier.shape=NA)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+geom_jitter(width=0.2,height=0,aes(color=cBP_cancer_type))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('Cancer Type')+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values = cBioPortal_colours_gf,labels = VHLrLD_cancer_type_labels)+scale_x_discrete(labels=VHLrLD_cancer_type_labels)

#CANFIG - Fig. 4 (showing all unique variant-cancer type combinations)
ggplot(VHLrLD_cancer_type_df[which(VHLrLD_cancer_type_df$Cancer_filtered_simple != "show_all"),],aes(x = function_score_final))+geom_histogram(alpha = 1,bins = 30,aes(fill = factor(cBP_cancer_type,levels = rev(levels(factor(cBP_cancer_type)))))) + theme(panel.background = element_rect(fill = 'white', colour = 'white')) + theme(legend.title = element_blank(),legend.key = element_blank())+xlab("Function score") +ylab('SNVs') +theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(text = element_text(size = 12),axis.text.y = element_text(size = 12),axis.text.x = element_text(size = 12),strip.background = element_blank())+scale_fill_manual(values = cBioPortal_colours_gf,labels = VHLrLD_cancer_type_labels)+coord_cartesian(xlim = c(-4, 1))

#CANFIG - Fig. 4 (showing all unique variant-cancer type combinations, excluding absent
ggplot(VHLrLD_cancer_type_df[which(VHLrLD_cancer_type_df$Cancer_filtered_simple != "Absent"),],aes(x = function_score_final))+geom_histogram(alpha = 1,bins = 30,aes(fill = factor(cBP_cancer_type,levels = rev(levels(factor(cBP_cancer_type)))))) + theme(panel.background = element_rect(fill = 'white', colour = 'white')) + theme(legend.title = element_blank(),legend.key = element_blank(),legend.position = "None")+xlab("Function score") +ylab('SNVs') +theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(text = element_text(size = 12),axis.text.y = element_text(size = 12),axis.text.x = element_text(size = 12),strip.background = element_blank())+scale_fill_manual(values = cBioPortal_colours_gf,labels = VHLrLD_cancer_type_labels)+coord_cartesian(xlim = c(-4, 1))
```
Defining gold standard sets of ccRCC variants for assessing clinical performance

```{r}
gs_ccrcc_present <- which(final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC >= 1) 
gs_ccrcc_multiple <- which(final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC >= 2)
any_cv_path <- which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic")

#114 variants either in ClinVar as P/LP AND at least 1 ccRCC sample OR in multiple ccRCC samples)
gold_standard_ccrcc_set <- union(intersect(any_cv_path,gs_ccrcc_present),gs_ccrcc_multiple)

ggplot(final_filtered_no1c_df_cBPu_RCC[gold_standard_ccrcc_set,], aes(x = function_score_final,y=total_cBP_ccRCC,color=fs_sig)) +geom_jitter(alpha = 1,height=0.1,width=0.0)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title = element_blank(),legend.key = element_blank(),legend.position = 'Bottom')+xlab("Function score")+ylab('ccRCC cases')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size = 12),axis.text.y = element_text(size = 12),axis.text.x = element_text(size = 12),strip.background = element_blank()) +scale_colour_manual(values=significance_colors_simple)+coord_cartesian(xlim = c(-4, 1))

max(final_filtered_no1c_df_cBPu_RCC[gold_standard_ccrcc_set,]$function_score_final)
mean(final_filtered_no1c_df_cBPu_RCC[gold_standard_ccrcc_set,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[gold_standard_ccrcc_set,]$function_score_final)
quantile(final_filtered_no1c_df_cBPu_RCC[gold_standard_ccrcc_set,]$function_score_final)

#DEFINE A GOLD-STANDARD neutral set.

any_cv_benign <- which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Likely benign","Benign"))
gs_pop_count_1 <- which(final_filtered_no1c_df_cBPu_RCC$combined_pop_count > 0)

#108 variants LB/B and pop count > 0
gold_standard_benign_set <- intersect(any_cv_benign,gs_pop_count_1)

min(final_filtered_no1c_df_cBPu_RCC[gold_standard_benign_set,]$function_score_final)
mean(final_filtered_no1c_df_cBPu_RCC[gold_standard_benign_set,]$function_score_final)
median(final_filtered_no1c_df_cBPu_RCC[gold_standard_benign_set,]$function_score_final)
#dim(final_filtered_no1c_df_cBPu_RCC[gold_standard_benign_set,])
quantile(final_filtered_no1c_df_cBPu_RCC[gold_standard_benign_set,]$function_score_final)

###roc, precision-recall analysis
library(plotROC)
library(PRROC)


#gold_standard classifications geom_roc:
gs_benign_df <- final_filtered_no1c_df_cBPu_RCC[gold_standard_benign_set,]
gs_ccrcc_df <- final_filtered_no1c_df_cBPu_RCC[gold_standard_ccrcc_set,]
gs_benign_df$D_P.B <- 0
gs_benign_df$gs_rcc_class <- "Neutral"
gs_ccrcc_df$D_P.B <- 1
gs_ccrcc_df$gs_rcc_class <- "ccRCC"

VHLxrLD_gs_geom_roc <- rbind(gs_benign_df,gs_ccrcc_df)

VHLxrLD_gs_P.B_roc <- ggplot(VHLxrLD_gs_geom_roc, aes(d = D_P.B, m = -function_score_final)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_gs_P.B_roc.annotated <- VHLxrLD_gs_P.B_roc + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_gs_P.B_roc)$AUC, 3)))

pr_gs <- pr.curve(gs_benign_df$function_score_final,gs_ccrcc_df$function_score_final,curve=TRUE)
print(pr_gs)
plot(pr_gs)

VHLxrLD_gs_P.B_roc_cadd <- ggplot(VHLxrLD_gs_geom_roc, aes(d = D_P.B, m = CADD.phred)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_gs_P.B_roc_cadd.annotated <- VHLxrLD_gs_P.B_roc_cadd + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_gs_P.B_roc_cadd)$AUC, 3)))
VHLxrLD_gs_P.B_roc_cadd.annotated

#clinvar classifications only
cv_benign_df <- final_filtered_no1c_df_cBPu_RCC[any_cv_benign,]
cv_path_df <- final_filtered_no1c_df_cBPu_RCC[any_cv_path,]
cv_benign_df$D_P.B <- 0
cv_path_df$D_P.B <- 1

VHLxrLD_cv_geom_roc <- rbind(cv_benign_df,cv_path_df)
#FS 364 SNVs total, 190 b/lb + 174 path/lp
dim(cv_benign_df)[1]
dim(cv_path_df)[1]
dim(VHLxrLD_cv_geom_roc)[1]


VHLxrLD_cv_P.B_roc <- ggplot(VHLxrLD_cv_geom_roc, aes(d = D_P.B, m = -function_score_final))+geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_cv_P.B_roc.annotated <- VHLxrLD_cv_P.B_roc + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cv_P.B_roc)$AUC, 3)))
VHLxrLD_cv_P.B_roc.annotated

pr_cv <- pr.curve(-cv_path_df$function_score_final,-cv_benign_df$function_score_final,curve=TRUE)
plot(pr_cv)

VHLxrLD_cv_P.B_roc_cadd <- ggplot(VHLxrLD_cv_geom_roc, aes(d = D_P.B, m = CADD.phred)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_cv_P.B_roc_cadd.annotated <- VHLxrLD_cv_P.B_roc_cadd + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cv_P.B_roc_cadd)$AUC, 3)))
VHLxrLD_cv_P.B_roc_cadd.annotated

##using only "Pathogenic" as path set to exclude likely path
cv_firm_path_df <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic"),]
cv_firm_path_df$D_P.B <- 1
VHLxrLD_cvfp_geom_roc <- rbind(cv_benign_df,cv_firm_path_df)
#273 SNVs total, 114 b/lb + 100 path
VHLxrLD_cvfp_P.B_roc <- ggplot(VHLxrLD_cvfp_geom_roc, aes(d = D_P.B, m = -function_score_final)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_cvfp_P.B_roc.annotated <- VHLxrLD_cvfp_P.B_roc + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cvfp_P.B_roc)$AUC, 3)))

pr_cv_firm_path <- pr.curve(-cv_firm_path_df$function_score_final,-cv_benign_df$function_score_final,curve=TRUE)
print(pr_cv_firm_path)
plot(pr_cv_firm_path)

VHLxrLD_cvfp_P.B_roc_cadd <- ggplot(VHLxrLD_cvfp_geom_roc, aes(d = D_P.B, m = CADD.phred)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_cvfp_P.B_roc_cadd.annotated <- VHLxrLD_cvfp_P.B_roc_cadd + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cvfp_P.B_roc_cadd)$AUC, 3)))
VHLxrLD_cvfp_P.B_roc_cadd.annotated

#Define subset of variants associated with recessive conditions
VHL_clinvar_path_set <- which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic")
VHL_all_clinvar_path_df <- final_filtered_no1c_df_cBPu_RCC[VHL_clinvar_path_set,]
VHL_polycythemia_variants <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$vhldb_polycyt_count > 0 & grepl(final_filtered_no1c_df_cBPu_RCC$vhldb_transmission_simple, pattern = "Homozygous")),c("cHGVS")]
#Perrotta et al. NEJM 2020 
VHL_deficiency_variants <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$cHGVS ==  "c.222C>A"),c("cHGVS")]
#Lenglet et al. 2018 Blood --> This allele is likely pathogenic, but always occurs with a second nearby SNV
VHL_1p_plp <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$sge_region ==  "exon 1p" & (final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic" )),c("cHGVS")]
VHL_recessive_variants <- union(union(VHL_polycythemia_variants,VHL_deficiency_variants),VHL_1p_plp)

#re-calculate ROC and PR without recessive variants
#all clinvar classifications geom_roc:
cv_benign_df <- final_filtered_no1c_df_cBPu_RCC[any_cv_benign,]
cv_path_nr_df <- final_filtered_no1c_df_cBPu_RCC[any_cv_path,]
cv_path_nr_df <- cv_path_nr_df[!(cv_path_nr_df$cHGVS %in% VHL_recessive_variants),]

cv_benign_df$D_P.B <- 0
cv_path_nr_df$D_P.B <- 1

VHLxrLD_cv_nr_geom_roc <- rbind(cv_benign_df,cv_path_nr_df)
VHLxrLD_cv_nr_P.B_roc <- ggplot(VHLxrLD_cv_nr_geom_roc, aes(d = D_P.B, m = -function_score_final)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_cv_nr_P.B_roc.annotated <- VHLxrLD_cv_nr_P.B_roc + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cv_nr_P.B_roc)$AUC, 3)))

pr_cv_nr <- pr.curve(-cv_path_nr_df$function_score_final,-cv_benign_df$function_score_final,curve=TRUE)
print(pr_cv_nr)
plot(pr_cv_nr)

##using only "Pathogenic" as path set to exclude likely path and recessive variants
cv_firm_path_nr_df <- cv_firm_path_df[!(cv_firm_path_df$cHGVS %in% VHL_recessive_variants),]
cv_firm_path_nr_df$D_P.B <- 1
VHLxrLD_cvfp_nr_geom_roc <- rbind(cv_benign_df,cv_firm_path_nr_df)
#273 SNVs total, 114 b/lb + 100 path
VHLxrLD_cvfp_nr_P.B_roc <- ggplot(VHLxrLD_cvfp_nr_geom_roc, aes(d = D_P.B, m = -function_score_final)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_cvfp_nr_P.B_roc.annotated <- VHLxrLD_cvfp_nr_P.B_roc + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cvfp_nr_P.B_roc)$AUC, 3)))

pr_cv_firm_path_nr <- pr.curve(-cv_firm_path_nr_df$function_score_final,-cv_benign_df$function_score_final,curve=TRUE)
print(pr_cv_firm_path_nr)
plot(pr_cv_firm_path_nr)

VHLxrLD_cvfp_nr_P.B_roc_cadd <- ggplot(VHLxrLD_cvfp_nr_geom_roc, aes(d = D_P.B, m = CADD.phred)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
#with AUC:
VHLxrLD_cvfp_nr_P.B_roc_cadd.annotated <- VHLxrLD_cvfp_nr_P.B_roc_cadd + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cvfp_nr_P.B_roc_cadd)$AUC, 3)))

#Final Figure SFig. 7a Supplementary Figure 7 CANFIG - export 3 x 6 VHLrLD_fs_roc_curves_clinvar
#grid.arrange(VHLxrLD_cvfp_P.B_roc.annotated,VHLxrLD_cvfp_nr_P.B_roc.annotated,VHLxrLD_cv_P.B_roc.annotated,VHLxrLD_cv_nr_P.B_roc.annotated,nrow=1)
#excluding recessive has no big effect (1 variant scoring intermediately, therefore not showing)
grid.arrange(VHLxrLD_cvfp_P.B_roc.annotated,VHLxrLD_cv_P.B_roc.annotated,nrow=1)


pr_cv_nr <- pr.curve(-cv_path_nr_df$function_score_final,-cv_benign_df$function_score_final,curve=TRUE)
print(pr_cv_nr)
plot(pr_cv_nr)

#FS numbers of clinvar variants used in each ROC (SF 6)
dim(cv_path_df)[1]
dim(cv_path_nr_df)[1]
dim(cv_firm_path_df)[1]
dim(cv_firm_path_nr_df)[1]
dim(cv_benign_df)[1]

gs_rcc_colors = c("Neutral" = "#4F71E1","ccRCC" = "#E0004B")

#FF Final Fig. 4 candidate CANFIG export 3x6 VHLrLD_gs_rcc_score_histo
ggplot(VHLxrLD_gs_geom_roc, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=gs_rcc_class),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=gs_rcc_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#defining neutral missense variants to be those seen in at least one "control individual" but NOT ClinVar, making a new gs_missense dataframe with those and subset of neutral variants that are missense.

gs_missense_neutral_df <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "NON_SYNONYMOUS" & !(final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Pathogenic", "Likely pathogenic")) & final_filtered_no1c_df_cBPu_RCC$combined_pop_count > 1),]
gs_missense_ccrcc_df <- gs_ccrcc_df[which(gs_ccrcc_df$conseq == "NON_SYNONYMOUS"),]
gs_missense_neutral_df$D_P.B <- 0
gs_missense_neutral_df$gs_rcc_class <- "Neutral"
gs_missense_ccrcc_df$D_P.B <- 1
gs_missense_ccrcc_df$gs_rcc_class <- "ccRCC"

VHLxrLD_gs_missense_geom_roc <- rbind(gs_missense_neutral_df,gs_missense_ccrcc_df)

#CANFIG Final Fig. 4 candidate CANFIG export 3x6 VHLrLD_gs_rcc_missense_score_histo
VHLrLD_gs_rcc_score_histo <- ggplot(VHLxrLD_gs_geom_roc, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=gs_rcc_class),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=gs_rcc_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

VHLrLD_gs_rcc_missense_score_histo <- ggplot(VHLxrLD_gs_missense_geom_roc, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=gs_rcc_class),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=gs_rcc_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))
#FF CANFIG Final Fig. 4f/g export 3x6 VHLrLD_gs_rcc_missense_score_histo
grid.arrange(VHLrLD_gs_rcc_score_histo,VHLrLD_gs_rcc_missense_score_histo,nrow=2)

#FS numbers in each set for gold-standard (all, missense only)
dim(gs_benign_df)[1]
dim(gs_ccrcc_df)[1]
dim(gs_missense_neutral_df)[1]
dim(gs_missense_ccrcc_df)[1]
```
Comparison to computational predictors

```{r}
### determining set of missense SNVs scored by all
##using only "Pathogenic"
cv_firm_path_nr_ms_df <- cv_firm_path_nr_df[cv_firm_path_nr_df$conseq == "NON_SYNONYMOUS" & (!is.na(cv_firm_path_nr_df$EVE_scores_ASM)),]
cv_firm_path_nr_ms_df$D_P.B <- 1
cv_firm_path_nr_ms_df$gs_rcc_class <- NA_character_
#FS SFig. 8
length(cv_firm_path_nr_ms_df$D_P.B)

gs_ms55.207_neutral_df <- gs_missense_neutral_df[(!is.na(gs_missense_neutral_df$EVE_scores_ASM)),]
#FS SFig. 8
length(gs_ms55.207_neutral_df$D_P.B)
#using same set of missense present in population cohorts:
VHLxrLD_cvfp_nr_ms_geom_roc <- rbind(gs_ms55.207_neutral_df,cv_firm_path_nr_ms_df)

#FS Note:  Supplementary Figure 8 Fig.8 Fig. 8 SFig.8a - comparing prediction on clinvar missense "pathogenic" variants. EVE scores do not include protPos before M54 or after 207, therefore this is a subset of all missense SNVs tested in SGE.

#function_scores
VHLxrLD_cv_fpnr_missense_roc <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = -function_score_final)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc.annotated <- VHLxrLD_cv_fpnr_missense_roc + annotate("text", x = 0.5, y = 0.5, label = paste("Function scores\nAUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc)$AUC, 3)))

#EVE
VHLxrLD_cv_fpnr_missense_roc_eve <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = EVE_scores_ASM)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc_eve.annotated <- VHLxrLD_cv_fpnr_missense_roc_eve + annotate("text", x = 0.5, y = 0.5, label = paste("EVE scores\nAUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc_eve)$AUC, 3)))

#VARITY_R
VHLxrLD_cv_fpnr_missense_roc_v.r <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = VARITY_R)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc_v.r.annotated <- VHLxrLD_cv_fpnr_missense_roc_v.r + annotate("text", x = .5, y = .5, label = paste("VARITY R scores\nAUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc_v.r)$AUC, 3)))

#VARITY_ER
VHLxrLD_cv_fpnr_missense_roc_v.er <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = VARITY_ER)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc_v.er.annotated <- VHLxrLD_cv_fpnr_missense_roc_v.er + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc_v.er)$AUC, 3)))

#VARITY_R_LOO
VHLxrLD_cv_fpnr_missense_roc_v.rLOO <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = VARITY_R_LOO)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc_v.rLOO.annotated <- VHLxrLD_cv_fpnr_missense_roc_v.rLOO + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc_v.rLOO)$AUC, 3)))

#VARITY_ER_LOO
VHLxrLD_cv_fpnr_missense_roc_v.erLOO <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = VARITY_ER_LOO)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc_v.erLOO.annotated <- VHLxrLD_cv_fpnr_missense_roc_v.erLOO + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc_v.erLOO)$AUC, 3)))

#REVEL
VHLxrLD_cv_fpnr_missense_roc_REVEL <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = REVEL)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc_REVEL.annotated <- VHLxrLD_cv_fpnr_missense_roc_REVEL + annotate("text", x = .5, y = .5, label = paste("REVEL scores\nAUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc_REVEL)$AUC, 3)))

#boostDM_score
VHLxrLD_cv_fpnr_missense_roc_boostDM_score <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = boostDM_score)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc_boostDM_score.annotated <- VHLxrLD_cv_fpnr_missense_roc_boostDM_score + annotate("text", x = .5, y = .5, label = paste("boostDM scores\nAUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc_boostDM_score)$AUC, 3)))

#CADD.phred
VHLxrLD_cv_fpnr_missense_roc_cadd <- ggplot(VHLxrLD_cvfp_nr_ms_geom_roc[!is.na(VHLxrLD_cvfp_nr_ms_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = CADD.phred)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_cv_fpnr_missense_roc_cadd.annotated <- VHLxrLD_cv_fpnr_missense_roc_cadd + annotate("text", x = .5, y = .5, label = paste("CADD scores\nAUC =", round(calc_auc(VHLxrLD_cv_fpnr_missense_roc_cadd)$AUC, 3)))

#FF Supplementary Figure 8a CANFIG SFig.8 SFig. 8a export as 3 x 18 VHLrLD_cv_fpnr_missense_ROC_curves
grid.arrange(VHLxrLD_cv_fpnr_missense_roc.annotated,VHLxrLD_cv_fpnr_missense_roc_REVEL.annotated,VHLxrLD_cv_fpnr_missense_roc_boostDM_score.annotated,VHLxrLD_cv_fpnr_missense_roc_eve.annotated,VHLxrLD_cv_fpnr_missense_roc_v.r.annotated,VHLxrLD_cv_fpnr_missense_roc_cadd.annotated,nrow=1)

#repeat for gold standard.
#function_scores
VHLxrLD_gs_rcc_missense_roc <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = -function_score_final)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc.annotated <- VHLxrLD_gs_rcc_missense_roc + annotate("text", x = 0.5, y = 0.5, label = paste("Function scores\nAUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc)$AUC, 3)))

#EVE
VHLxrLD_gs_rcc_missense_roc_eve <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = EVE_scores_ASM)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc_eve.annotated <- VHLxrLD_gs_rcc_missense_roc_eve + annotate("text", x = 0.5, y = 0.5, label = paste("EVE scores\nAUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc_eve)$AUC, 3)))

#VARITY_R
VHLxrLD_gs_rcc_missense_roc_v.r <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = VARITY_R)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc_v.r.annotated <- VHLxrLD_gs_rcc_missense_roc_v.r + annotate("text", x = .5, y = .5, label = paste("VARITY R scores\nAUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc_v.r)$AUC, 3)))

#VARITY_ER
VHLxrLD_gs_rcc_missense_roc_v.er <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = VARITY_ER)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc_v.er.annotated <- VHLxrLD_gs_rcc_missense_roc_v.er + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc_v.er)$AUC, 3)))

#VARITY_R_LOO
VHLxrLD_gs_rcc_missense_roc_v.rLOO <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = VARITY_R_LOO)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc_v.rLOO.annotated <- VHLxrLD_gs_rcc_missense_roc_v.rLOO + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc_v.rLOO)$AUC, 3)))

#VARITY_ER_LOO
VHLxrLD_gs_rcc_missense_roc_v.erLOO <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = VARITY_ER_LOO)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc_v.erLOO.annotated <- VHLxrLD_gs_rcc_missense_roc_v.erLOO + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc_v.erLOO)$AUC, 3)))

#REVEL
VHLxrLD_gs_rcc_missense_roc_REVEL <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = REVEL)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc_REVEL.annotated <- VHLxrLD_gs_rcc_missense_roc_REVEL + annotate("text", x = .5, y = .5, label = paste("REVEL scores\nAUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc_REVEL)$AUC, 3)))

#boostDM_score
VHLxrLD_gs_rcc_missense_roc_boostDM_score <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = boostDM_score)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc_boostDM_score.annotated <- VHLxrLD_gs_rcc_missense_roc_boostDM_score + annotate("text", x = .5, y = .5, label = paste("boostDM scores\nAUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc_boostDM_score)$AUC, 3)))

#CADD.phred
VHLxrLD_gs_rcc_missense_roc_cadd <- ggplot(VHLxrLD_gs_missense_geom_roc[!is.na(VHLxrLD_gs_missense_geom_roc$EVE_scores_ASM),], aes(d = D_P.B, m = CADD.phred)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_gs_rcc_missense_roc_cadd.annotated <- VHLxrLD_gs_rcc_missense_roc_cadd + annotate("text", x = .5, y = .5, label = paste("CADD scores\nAUC =", round(calc_auc(VHLxrLD_gs_rcc_missense_roc_cadd)$AUC, 3)))

#FF Supplementary Figure 8b CANFIG SFig.8 SFig. 8b export as 3 x 18 VHLrLD_gs_missense_ROC_curves
grid.arrange(VHLxrLD_gs_rcc_missense_roc.annotated,VHLxrLD_gs_rcc_missense_roc_REVEL.annotated,VHLxrLD_gs_rcc_missense_roc_boostDM_score.annotated,VHLxrLD_gs_rcc_missense_roc_eve.annotated,VHLxrLD_gs_rcc_missense_roc_v.r.annotated,VHLxrLD_gs_rcc_missense_roc_cadd.annotated,nrow=1)

```
Testing computational predictors on unseen missense variants.

Defining cut-offs for LoF and neutral

```{r}
####assess same predictors on newly identified LoF variants vs. newly identified neutral variants####
#defining newly id'd missense variants as those completely absent from ClinVar 1-start set, VHLdb, population databases, and cBioPortal
#define window for LoF variants:
#cutoff for evidence of neutrality is greater than:
VHLxrLD_gs_neut_cutoff<- min(VHLxrLD_gs_geom_roc[which(VHLxrLD_gs_geom_roc$gs_rcc_class == "Neutral"),]$function_score_final)-.0001
min(VHLxrLD_gs_geom_roc[which(VHLxrLD_gs_geom_roc$gs_rcc_class == "Neutral"),]$fdr)
#cutoff for evidence of pathogenicity is less than:
VHLxrLD_gs_path_cutoff <- max(VHLxrLD_gs_geom_roc[which(VHLxrLD_gs_geom_roc$gs_rcc_class == "ccRCC"),]$function_score_final)+.0001
max(VHLxrLD_gs_geom_roc[which(VHLxrLD_gs_geom_roc$gs_rcc_class == "ccRCC"),]$fdr)

#general concordance between cut-offs and statistical test for LoF based on null model
#min(VHLxrLD_gs_geom_roc[which(VHLxrLD_gs_geom_roc$gs_rcc_class == "Neutral"),]$function_score_final)
#min(VHLxrLD_gs_geom_roc[which(VHLxrLD_gs_geom_roc$gs_rcc_class == "Neutral"),]$fdr)
#max(VHLxrLD_gs_geom_roc[which(VHLxrLD_gs_geom_roc$gs_rcc_class == "ccRCC"),]$function_score_final)
#max(VHLxrLD_gs_geom_roc[which(VHLxrLD_gs_geom_roc$gs_rcc_class == "ccRCC"),]$fdr)
#min(VHLxrLD_gs_missense_geom_roc[which(VHLxrLD_gs_missense_geom_roc$gs_rcc_class == "Neutral"),]$function_score_final)
#min(VHLxrLD_gs_missense_geom_roc[which(VHLxrLD_gs_missense_geom_roc$gs_rcc_class == "Neutral"),]$fdr)
#max(VHLxrLD_gs_missense_geom_roc[which(VHLxrLD_gs_missense_geom_roc$gs_rcc_class == "ccRCC"),]$function_score_final)
#max(VHLxrLD_gs_missense_geom_roc[which(VHLxrLD_gs_missense_geom_roc$gs_rcc_class == "ccRCC"),]$fdr)
#what are the scores on either side of the threshold in the complete data set?
#min(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$function_score_final > VHLxrLD_gs_path_cutoff),]$function_score_final)
#max(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$function_score_final)
#min(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$function_score_final > VHLxrLD_gs_neut_cutoff),]$function_score_final)
#max(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_neut_cutoff),]$function_score_final)
#max(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$fdr < 0.01),]$function_score_final)
#min(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$fdr > 0.01),]$function_score_final)


#assigning a ccRCC functional class based on function score for each variant in set of 2,200
final_filtered_no1c_df_cBPu_RCC$ccRCC_fs_class <- 'Intermediate' 
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$ccRCC_fs_class <- 'Pathogenic'
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$function_score_final > VHLxrLD_gs_neut_cutoff),]$ccRCC_fs_class <- 'Neutral'

View(final_filtered_no1c_df_cBPu_RCC[final_filtered_no1c_df_cBPu_RCC$ccRCC_fs_class == "Intermediate",c("function_score_final","fdr")])

VHLxrLD_cv_nr_geom_roc$clinvar_simple <- factor(VHLxrLD_cv_nr_geom_roc$clinvar_simple, levels =c("Likely pathogenic","Pathogenic","Likely benign","Benign"))

#FF Final Figure SFig. 7c Supplementary Figure 7c CANFIG - export 3 x 6 VHLrLD_clinvar_all_histo_with_cutoffs
ggplot(VHLxrLD_cv_geom_roc, aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(-4,1))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)

all_new_df <- final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "absent" & is.na(final_filtered_no1c_df_cBPu_RCC$vhldb_entries) &  final_filtered_no1c_df_cBPu_RCC$combined_pop_count == 0 & (is.na(final_filtered_no1c_df_cBPu_RCC$total_cBP_entries)| final_filtered_no1c_df_cBPu_RCC$total_cBP_entries == 0)),]
all_new_df$protPos <- as.numeric(as.character(all_new_df$protPos))

#threshold selected to pick the 100 SNVs scoring closest to 0.
new_missense_neutral_df <- all_new_df[which(all_new_df$conseq == "NON_SYNONYMOUS" & abs(all_new_df$function_score_final) < 0.0515 & (all_new_df$protPos > 54) & (all_new_df$protPos < 208)),]
dim(new_missense_neutral_df)

new_missense_LoF_df <- all_new_df[which(all_new_df$conseq == "NON_SYNONYMOUS" & all_new_df$function_score_final < VHLxrLD_gs_path_cutoff & all_new_df$fdr < 0.01),]
new_missense_neutral_df$D_P.B <- 0
new_missense_neutral_df$new_func_class <- "Neutral"
new_missense_LoF_df$D_P.B <- 1
new_missense_LoF_df$new_func_class <- "LoF"

VHLxrLD_new_missense_geom_roc <- rbind(new_missense_neutral_df,new_missense_LoF_df)
dim(VHLxrLD_new_missense_geom_roc)

#function_scores
VHLxrLD_new_LoF_missense_roc <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = -function_score_final)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc.annotated <- VHLxrLD_new_LoF_missense_roc + annotate("text", x = 0.5, y = 0.5, label = paste("Function scores\nAUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc)$AUC, 3)))

#EVE
VHLxrLD_new_LoF_missense_roc_eve <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = EVE_scores_ASM)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc_eve.annotated <- VHLxrLD_new_LoF_missense_roc_eve + annotate("text", x = 0.5, y = 0.5, label = paste("EVE scores\nAUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc_eve)$AUC, 3)))

#VARITY_R
VHLxrLD_new_LoF_missense_roc_v.r <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = VARITY_R)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc_v.r.annotated <- VHLxrLD_new_LoF_missense_roc_v.r + annotate("text", x = .5, y = .5, label = paste("VARITY R scores\nAUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc_v.r)$AUC, 3)))

#VARITY_ER
VHLxrLD_new_LoF_missense_roc_v.er <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = VARITY_ER)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc_v.er.annotated <- VHLxrLD_new_LoF_missense_roc_v.er + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc_v.er)$AUC, 3)))

#VARITY_R_LOO
VHLxrLD_new_LoF_missense_roc_v.rLOO <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = VARITY_R_LOO)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc_v.rLOO.annotated <- VHLxrLD_new_LoF_missense_roc_v.rLOO + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc_v.rLOO)$AUC, 3)))

#VARITY_ER_LOO
VHLxrLD_new_LoF_missense_roc_v.erLOO <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = VARITY_ER_LOO)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc_v.erLOO.annotated <- VHLxrLD_new_LoF_missense_roc_v.erLOO + annotate("text", x = .5, y = .5, label = paste("AUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc_v.erLOO)$AUC, 3)))

#REVEL
VHLxrLD_new_LoF_missense_roc_REVEL <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = REVEL)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc_REVEL.annotated <- VHLxrLD_new_LoF_missense_roc_REVEL + annotate("text", x = .5, y = .5, label = paste("REVEL scores\nAUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc_REVEL)$AUC, 3)))

#boostDM_score
VHLxrLD_new_LoF_missense_roc_boostDM_score <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = boostDM_score)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc_boostDM_score.annotated <- VHLxrLD_new_LoF_missense_roc_boostDM_score + annotate("text", x = .5, y = .5, label = paste("boostDM scores\nAUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc_boostDM_score)$AUC, 3)))

#CADD.phred
VHLxrLD_new_LoF_missense_roc_cadd <- ggplot(VHLxrLD_new_missense_geom_roc, aes(d = D_P.B, m = CADD.phred)) +geom_roc(n.cuts=0,cutoffs.at = NULL) + style_roc(xlab = "1 - Specificity")
VHLxrLD_new_LoF_missense_roc_cadd.annotated <- VHLxrLD_new_LoF_missense_roc_cadd + annotate("text", x = .5, y = .5, label = paste("CADD scores\nAUC =", round(calc_auc(VHLxrLD_new_LoF_missense_roc_cadd)$AUC, 3)))

#FF CANFIG SFig. 8 Supplementary Figure 8 SFig. 8d -- export VHLrLD_new_LoF_missense_prediction 3 x 15
grid.arrange(VHLxrLD_new_LoF_missense_roc_REVEL.annotated,VHLxrLD_new_LoF_missense_roc_boostDM_score.annotated,VHLxrLD_new_LoF_missense_roc_eve.annotated,VHLxrLD_new_LoF_missense_roc_v.r.annotated,VHLxrLD_new_LoF_missense_roc_cadd.annotated,nrow=1)

####missense from p.54 to 207 df 
final_filtered_missense55_df <- final_filtered_no1c_df_cBPu_RCC
final_filtered_missense55_df$protPos <- as.numeric(as.character(final_filtered_missense55_df$protPos))
final_filtered_missense55_df <- final_filtered_missense55_df[(final_filtered_missense55_df$protPos > 54) & (final_filtered_missense55_df$protPos < 208),]
final_filtered_missense55_df <- final_filtered_missense55_df[!(is.na(final_filtered_missense55_df$protPos)),]
final_filtered_missense55_df <- final_filtered_missense55_df[final_filtered_missense55_df$conseq == "NON_SYNONYMOUS",]

final_filtered_missense55_df$clinvar_simple <- factor(final_filtered_missense55_df$clinvar_simple,levels = c("Pathogenic","Likely pathogenic","Likely benign","Benign" ,"Conflicting interpretations of pathogenicity","Uncertain significance","absent"))

VHLrLD_function_score_vs_EVE_mis <- ggplot(final_filtered_missense55_df, aes(x=function_score_final,y=as.numeric(as.character(EVE_scores_ASM))))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple == "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple != "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('EVE score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+coord_cartesian(xlim=c(-4,1))+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)

VHLrLD_function_score_vs_VARITY.R <- ggplot(final_filtered_missense55_df, aes(x=function_score_final,y=as.numeric(as.character(VARITY_R))))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple == "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple != "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('VARITY R score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+coord_cartesian(xlim=c(-4,1))+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)

VHLrLD_function_score_vs_REVEL_mis <- ggplot(final_filtered_missense55_df, aes(x=function_score_final,y=as.numeric(as.character(REVEL))))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple == "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple != "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('REVEL score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+coord_cartesian(xlim=c(-4,1))+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)

VHLrLD_function_score_vs_boostDM_mis <- ggplot(final_filtered_missense55_df, aes(x=function_score_final,y=as.numeric(as.character(boostDM_score))))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple == "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple != "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('boostDM score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+coord_cartesian(xlim=c(-4,1))+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)

VHLrLD_function_score_vs_CADD_mis <- ggplot(final_filtered_missense55_df, aes(x=function_score_final,y=as.numeric(as.character(CADD.phred))))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple == "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple != "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('CADD score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+coord_cartesian(xlim=c(-4,1))+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)

#FF SFig. 8 Final Supplementary Figure 8 Fig. 8c - export as 3x15, VHLrLD_fs_vs_predictors_mis
grid.arrange(VHLrLD_function_score_vs_REVEL_mis,VHLrLD_function_score_vs_boostDM_mis,VHLrLD_function_score_vs_EVE_mis,VHLrLD_function_score_vs_VARITY.R,VHLrLD_function_score_vs_CADD_mis,nrow=1)

#SFig. 8c legend only
ggplot(final_filtered_missense55_df, aes(x=function_score_final,y=as.numeric(as.character(CADD.phred))))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple == "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+geom_jitter(data=final_filtered_missense55_df[final_filtered_missense55_df$clinvar_simple != "absent",],alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('CADD score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+coord_cartesian(xlim=c(-4,1))+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus,labels = clinvar_labels)
#FS Fig. 7C N
dim(final_filtered_missense55_df)

```

stats for classification by clinvar type
```{r}
VHLrLD_count_VUS <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Uncertain significance"),])[1]
VHLrLD_count_VUS_gs_path <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Uncertain significance"),]$function_score_final)<VHLxrLD_gs_path_cutoff))
VHLrLD_fraction_VUS_gs_path <- VHLrLD_count_VUS_gs_path/VHLrLD_count_VUS
VHLrLD_count_VUS
#FS VUS --> Path
VHLrLD_count_VUS_gs_path
VHLrLD_fraction_VUS_gs_path

VHLrLD_count_CI <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Conflicting interpretations of pathogenicity"),])[1]
VHLrLD_count_CI_gs_path <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Conflicting interpretations of pathogenicity"),]$function_score_final)<VHLxrLD_gs_path_cutoff))
VHLrLD_fraction_CI_gs_path <- VHLrLD_count_CI_gs_path/VHLrLD_count_CI
VHLrLD_fraction_CI_gs_path

VHLrLD_fraction_VUS_CI_gs_path <- (VHLrLD_count_VUS_gs_path+VHLrLD_count_CI_gs_path)/(VHLrLD_count_VUS+VHLrLD_count_CI)

dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$combined_pop_count > 0 & final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Uncertain significance","Conflicting interpretations of pathogenicity","absent")),])[1]

#total allele count of those lacking clinvar interpretation
sum(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$combined_pop_count > 0 & final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Uncertain significance","Conflicting interpretations of pathogenicity","absent")),]$combined_pop_count)

dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$combined_pop_count > 0 & final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Uncertain significance","Conflicting interpretations of pathogenicity","absent") & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),])[1]

#total allele count of those scoring below path cutoff
sum(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$combined_pop_count > 0 & final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Uncertain significance","Conflicting interpretations of pathogenicity","absent") & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$combined_pop_count)

dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$combined_pop_count > 0 & final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Uncertain significance","Conflicting interpretations of pathogenicity","absent") & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),])[1]

#total allele count of those scoring below above path cutoff
sum(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$combined_pop_count > 0 & final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Uncertain significance","Conflicting interpretations of pathogenicity","absent") & final_filtered_no1c_df_cBPu_RCC$function_score_final > VHLxrLD_gs_neut_cutoff),]$combined_pop_count)

VHLrLD_count_absent <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "absent"),])[1]
#FS absent
VHLrLD_count_absent_gs_path <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "absent"),]$function_score_final)<VHLxrLD_gs_path_cutoff))
#FS absent
VHLrLD_fraction_absent_gs_path <- VHLrLD_count_absent_gs_path/VHLrLD_count_absent

VHLrLD_count_not_absent <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple != "absent"),])[1]
VHLrLD_count_not_absent_gs_path <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple != "absent"),]$function_score_final)<VHLxrLD_gs_path_cutoff))
VHLrLD_fraction_not_absent_gs_path <- VHLrLD_count_not_absent_gs_path/VHLrLD_count_not_absent

ggplot(final_filtered_no1c_df_cBPu_RCC[final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Uncertain significance","Conflicting interpretations of pathogenicity"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus,labels=clinvar_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)

ggplot(final_filtered_no1c_df_cBPu_RCC[final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Pathogenic","Likely pathogenic"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus,labels=clinvar_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)

ggplot(final_filtered_no1c_df_cBPu_RCC[final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Benign","Likely benign"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus,labels=clinvar_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)

ggplot(final_filtered_no1c_df_cBPu_RCC[final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("absent"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple),binwidth=0.1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus,labels=clinvar_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)

#absent variants colored by cBioPortal status
final_filtered_no1c_df_cBPu_RCC[which(is.na(final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC)),]$total_cBP_ccRCC <- 0  

#run next 6 lines together.
final_filtered_no1c_df_cBPu_RCC$clinvar_group <- 'Absent'
final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Benign" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple =="Likely benign"),]$clinvar_group <- "Benign / Likely benign"
final_filtered_no1c_df_cBPu_RCC[final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Pathogenic","Likely pathogenic"),]$clinvar_group <- "Path. / Likely path"
final_filtered_no1c_df_cBPu_RCC[final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Uncertain significance","Conflicting interpretations of pathogenicity"),]$clinvar_group <- "Uncertain / Conflicting"

final_filtered_no1c_df_cBPu_RCC$clinvar_group <- factor(final_filtered_no1c_df_cBPu_RCC$clinvar_group, levels = c("Benign / Likely benign","Path. / Likely path","Uncertain / Conflicting","Absent"))

#FF CANFIG Final Fig. Figure. 4h - VHLrLD_ClinVar_reclass export as 3x6
ggplot(final_filtered_no1c_df_cBPu_RCC[,], aes(x=function_score_final,y=total_cBP_ccRCC))+geom_jitter(data=final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_single == "Absent" ),],alpha=1,height=0.3,aes(color=Cancer_type_single))+geom_jitter(data=final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_single != "Absent"),],alpha=1,height=0.1,aes(color=Cancer_type_single))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab("Function score")+ylab('ccRCC samples')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12))+scale_color_manual(values=cBioPortal_colours_gf)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)+facet_grid(clinvar_group ~ .)

#no sig. difference between clinvar-absent LoF variants and Path / Likely path LoF variants
wilcox.test(x = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_group == "Absent" & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$function_score_final, y = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_group == "Path. / Likely path" & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$function_score_final,paired = FALSE,alternative = "two.sided")

#FS sig. difference between absent LoF variants in ccRCC and absent LoF variants NOT in ccRCC
quantile(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_group == "Absent" & final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC > 0 & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$function_score_final)

quantile(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_group == "Absent" & final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC == 0 & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$function_score_final)

wilcox.test(x = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_group == "Absent" & final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC > 0 & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$function_score_final, y = final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_group == "Absent" & final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC == 0 & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff),]$function_score_final,paired = FALSE,alternative = "two.sided")



final_filtered_no1c_df_cBPu_RCC$OncoKB_simple <- as.factor('')
final_filtered_no1c_df_cBPu_RCC <- final_filtered_no1c_df_cBPu_RCC %>% 
  mutate(
    OncoKB_simple = factor(case_when(
      grepl(x = OncoKB, pattern = "Likely Neutral") ~ "Likely Neutral",
      grepl(x = OncoKB, pattern = "Likely Oncogenic") ~ "Oncogenic / Likely Onco.",
      grepl(x = OncoKB, pattern = "Oncogenic") ~ "Oncogenic / Likely Onco.",
      grepl(x = OncoKB, pattern = "Unknown") ~ "Unknown")))


#ggplot(final_filtered_no1c_df_cBPu_RCC[,], aes(x=function_score_final,y=total_cBP_ccRCC))+geom_jitter(data=final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_single == "Absent" ),],alpha=1,height=0.2,aes(color=Cancer_type_single))+geom_jitter(data=final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$Cancer_type_single != "Absent"),],alpha=1,height=0.2,aes(color=Cancer_type_single))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab("Function score")+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=cBioPortal_colours_gf)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)+facet_grid(OncoKB_simple ~ .)


#FF OncoKB x CancerHotspot - CANFIG Supp. Fig. 7 Figure 7 cBioPortal classifications 3 x 6 Final Fig. S7 
ggplot(final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB)),], aes(x=function_score_final,y=total_cBP_ccRCC))+geom_jitter(alpha=1,height=0.2,aes(color=Cancer_type_single))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.key = element_blank(),legend.position = "None")+xlab("Function score")+ylab('cBioPortal ccRCC samples')+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=cBioPortal_colours_gf)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)+facet_grid(CancerHotspot ~ OncoKB_simple)

#alternative facet only by hotspot status
ggplot(final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB)),], aes(x=function_score_final,y=total_cBP_ccRCC))+geom_jitter(alpha=1,height=0.2,aes(color=Cancer_type_single))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.key = element_blank(),legend.position = "None")+xlab("Function score")+ylab('cBioPortal ccRCC samples')+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=cBioPortal_colours_gf)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)+facet_grid(CancerHotspot ~ .)

final_filtered_no1c_df_cBPu_RCC$Cancer_type_single <- factor(final_filtered_no1c_df_cBPu_RCC$Cancer_type_single, levels = c("Clear Cell Renal Cell Carcinoma","Chromophobe Renal Cell Carcinoma","Papillary Renal Cell Carcinoma","Renal Cell Carcinoma, NOS","Pheochromocytoma","Pancreatic Neuroendocrine","Other"))

#legend only: - export 3 x 6 SFig.6 - last panel legend
ggplot(final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB)),], aes(x=function_score_final,y=total_cBP_ccRCC))+geom_jitter(alpha=1,height=0.2,aes(color=Cancer_type_single))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.key = element_blank())+xlab("Function score")+ylab('cBioPortal ccRCC samples')+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=cBioPortal_colours_gf,order())+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_vline(xintercept=VHLxrLD_gs_path_cutoff,lty=3)+geom_vline(xintercept=VHLxrLD_gs_neut_cutoff,lty=3)+facet_grid(CancerHotspot ~ OncoKB_simple)

#Among ccRCC-like variants (low scoring), no difference in score distribution between "hotspot" and non-hotspot in cBP
quantile(final_filtered_no1c_df_cBPu_RCC[which((!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB))&final_filtered_no1c_df_cBPu_RCC$CancerHotspot == "CancerHotspot: yes"& final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff) ,]$function_score_final)
quantile(final_filtered_no1c_df_cBPu_RCC[which((!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB))&final_filtered_no1c_df_cBPu_RCC$CancerHotspot == "CancerHotspot: no" & final_filtered_no1c_df_cBPu_RCC$function_score_final < VHLxrLD_gs_path_cutoff) ,]$function_score_final)


length(which(final_filtered_no1c_df_cBPu_RCC[which((!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB))&final_filtered_no1c_df_cBPu_RCC$OncoKB_simple == "Oncogenic / Likely Onco.") ,]$function_score_final < 1000))

length(which(final_filtered_no1c_df_cBPu_RCC[which((!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB))&final_filtered_no1c_df_cBPu_RCC$OncoKB_simple == "Oncogenic / Likely Onco.") ,]$function_score_final < VHLxrLD_gs_path_cutoff))

length(which(final_filtered_no1c_df_cBPu_RCC[which((!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB))&final_filtered_no1c_df_cBPu_RCC$OncoKB_simple == "Oncogenic / Likely Onco.") ,]$function_score_final > VHLxrLD_gs_neut_cutoff))

#FS total (Figure 4/ SFig. 7c)
132/150 # "Oncogenic / Likely Onco." variants score below ccRCC threshold
17/150  # "Oncogenic / Likely Onco." variants score neutrally, most of which were not identified in a tumor linked to VHL mutation.

length(which(final_filtered_no1c_df_cBPu_RCC[which((!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB))&final_filtered_no1c_df_cBPu_RCC$OncoKB_simple == "Unknown") ,]$function_score_final < 1000))

length(which(final_filtered_no1c_df_cBPu_RCC[which((!is.na(final_filtered_no1c_df_cBPu_RCC$OncoKB))&final_filtered_no1c_df_cBPu_RCC$OncoKB_simple == "Unknown") ,]$function_score_final < VHLxrLD_gs_path_cutoff))
#FS total (Figure 4/ SFig. 6c)
28/83 # "Unknown variants" score below ccRCC threshold

#FS how many more variants are likely to cause disease?
total_reclassified_path_variants <- VHLrLD_count_absent_gs_path+VHLrLD_count_CI_gs_path+VHLrLD_count_VUS_gs_path

```

Defining variant-VHL Disease type associations based on objective data from VHL-db and cBP

```{r}
VHL_clinvar_path_vus_set <- which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Uncertain significance" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Conflicting interpretations of pathogenicity")
VHL_clinvar_path_set <- which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic" | final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic")

VHL_all_RCC_set <- which(final_filtered_no1c_df_cBPu_RCC$total_cBP_rcc > 0)
VHL_all_ccRCC_set <- which(final_filtered_no1c_df_cBPu_RCC$total_cBP_ccRCC > 0)

VHL_vhldb_set <- which(final_filtered_no1c_df_cBPu_RCC$vhldb_entries > 0)

VHL_all_disease_variants <- union(union(VHL_clinvar_path_vus_set,VHL_all_RCC_set),VHL_vhldb_set)

#defining arbitrary cut-off at -1 to be the lowest LoF set
VHL_lowest_LOF_set <- which(final_filtered_no1c_df_cBPu_RCC$function_score_final < -1)

#FS discussion section - #16/17 are transversion mutations
length(VHL_lowest_LOF_set)
LoF_absent_from_disease_count <- dim(final_filtered_no1c_df_cBPu_RCC[setdiff(VHL_lowest_LOF_set, VHL_all_disease_variants),c("cHGVS","conseq","function_score_final","CADD.phred","clinvar_simple","total_cBP_rcc","total_cBP_entries","vhldb_entries","combined_pop_count")])[1]

#FS
fraction_lowest_LoF_absent_from_disease <- LoF_absent_from_disease_count/length(VHL_lowest_LOF_set)
# 7.1% (17 of the 238 SNVs scoring as < -1, have yet to be reported clinically

```

Defining delta RNA score to be difference between day 20 and day 6 RNA score.
```{r}

final_filtered_no1c_df_cBPu_RCC$delta_rna <- final_filtered_no1c_df_cBPu_RCC$post_rna_score - final_filtered_no1c_df_cBPu_RCC$rna_score
final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$Intron)),]$delta_rna <- NA_character_
final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$Intron)),]$rna_score <- NA_character_
final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$Intron)),]$post_rna_score <- NA_character_
length(final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$rna_score)),]$delta_rna)
length(final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$rna_score)),]$rna_score)
length(final_filtered_no1c_df_cBPu_RCC[which(!is.na(final_filtered_no1c_df_cBPu_RCC$rna_score)),]$post_rna_score)

#visualizing distinction between pheo-predominant and ccRCC-predominant variants from VHLdb annotations
ggplot(final_filtered_no1c_df_cBPu_RCC[final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Pathogenic","Likely pathogenic"),])+geom_jitter(height=0.2,width=0.2,alpha=1,aes(color=conseq,x=vhldb_rcc_count,y=vhldb_pheo_count))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab("vhldb_rcc_count")+ylab('vhldb_pheo_count')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels=conseq_labels)

#creating proxies for VHL Disease types based on ClinVar, cBioPortal, and vhldb entries:
VHL_clinvar_pv_vars <- final_filtered_no1c_df_cBPu_RCC[VHL_clinvar_path_vus_set,c("cHGVS")]
VHL_clinvar_plp_nc_vars <- final_filtered_no1c_df_cBPu_RCC[(final_filtered_no1c_df_cBPu_RCC$clinvar_simple %in% c("Pathogenic","Likely pathogenic"))&(final_filtered_no1c_df_cBPu_RCC$combined_pop_count < 2),c("cHGVS")]
VHL_clinvar_plp_nc_nr_vars <- VHL_clinvar_plp_nc_vars[which(!VHL_clinvar_plp_nc_vars %in% VHL_recessive_variants)]

master_disease_type_df <- final_filtered_no1c_df_cBPu_RCC[(final_filtered_no1c_df_cBPu_RCC$cHGVS %in% VHL_clinvar_plp_nc_nr_vars),]

#at least multiple ccRCC reports AND not seen in a single pheo in vhldb + cBP
rcc_only_df <- master_disease_type_df[which((master_disease_type_df$vhldb_rcc_count > 3 | master_disease_type_df$total_cBP_ccRCC >=2)&(master_disease_type_df$vhldb_pheo_count < 1)&(master_disease_type_df$Pheochromocytoma <1 | is.na(master_disease_type_df$Pheochromocytoma))),]

#more pheo reports than ccRCC reports AND not seen in a single ccRCC in cBP
pheo_predom_df <- master_disease_type_df[which(((master_disease_type_df$vhldb_pheo_count > master_disease_type_df$vhldb_rcc_count) | (master_disease_type_df$Pheochromocytoma >= 1))&(master_disease_type_df$total_cBP_ccRCC<1 | is.na(master_disease_type_df$total_cBP_ccRCC))),]

pheo_mixed_df <- master_disease_type_df[which( (master_disease_type_df$vhldb_pheo_count > 0 | master_disease_type_df$Pheochromocytoma >= 1) & (master_disease_type_df$total_cBP_ccRCC >0 | master_disease_type_df$vhldb_pheo_count <= master_disease_type_df$vhldb_rcc_count) & (master_disease_type_df$Pheochromocytoma == 0 | is.na(master_disease_type_df$Pheochromocytoma))),]

vhl_unspecified_df <-master_disease_type_df[!(master_disease_type_df$cHGVS %in% rcc_only_df$cHGVS | master_disease_type_df$cHGVS %in% pheo_predom_df$cHGVS | master_disease_type_df$cHGVS %in% pheo_mixed_df$cHGVS)&(master_disease_type_df$clinvar_simple %in% c("Pathogenic","Likely pathogenic")),]

#Analysis of tumor-preponderance - define broad categories and test for differences between categories.
rcc_only_df$vhl_disease_type <- "ccRCC only"
pheo_predom_df$vhl_disease_type <- "Pheo predominant"
pheo_mixed_df$vhl_disease_type <- "Pheo mixed"
vhl_unspecified_df$vhl_disease_type <- "Unspecified Pathogenic"

VHL_disease_type_df <- rbind(rcc_only_df,pheo_predom_df,pheo_mixed_df,vhl_unspecified_df)

VHL_disease_colors <- c("ccRCC only" = "#D56209","Pheo predominant" = "#05837A","Pheo mixed" = "#6666C8", "Unspecified"  = "#FFFEFE")

significance_colors3 <- c(named_conseq_colors,c("FDR < 0.01" = "#000000",
  "FDR < 0.05" = "#747474",
  "FDR < 0.2" = "#B6B6B6",
  "FDR < 0.5" = "#B6B6B6",
  "FDR > 0.5" = "#B6B6B6"))

VHL_disease_type_df <- merge(VHL_disease_type_df, rLH_final_filtered_df[,c("rLH_function_score_final","rLH_fdr","cHGVS")], by="cHGVS",all= FALSE,all.x = TRUE, all.y=FALSE)

final_clinical_df <- merge(final_filtered_no1c_df_cBPu_RCC, VHL_disease_type_df[,c("vhl_disease_type","cHGVS")], by="cHGVS",all= FALSE,all.x = TRUE, all.y=FALSE)

final_clinical_df <- merge(final_clinical_df, rLH_final_filtered_df[,c("rLH_function_score_final","rLH_fdr","cHGVS")], by="cHGVS",all= FALSE,all.x = TRUE, all.y=FALSE)


VHL_disease_type_df$vhl_disease_type <- factor(VHL_disease_type_df$vhl_disease_type, levels = c("ccRCC only","Pheo predominant","Pheo mixed","Unspecified Pathogenic"))

VHL_disease_colors2 <- c("ccRCC only" = "#E0004B","Pheo predominant" = "#00BA74","Pheo mixed" = "#FD9100", "Unspecified"  = "#D0D0D0")
VHL_disease_colors3 <- c("ccRCC only" = "#E0004B","Pheo predominant" = "#00BA74","Pheo mixed" = "#FD9100", "Unspecified"  = "#D0D0D0","STOP_GAINED" = "#BD0016","CANONICAL_SPLICE" = "#FFAA38","INTRONIC" = "#88CCEE", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#15327C", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#CB57D7","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#EA93F3", "REF" = "#000000")

#CANFIG - 
ggplot(VHL_disease_type_df, aes(x=vhl_disease_type, y=function_score_final))+geom_boxplot(aes(),outlier.shape=NA)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('Cancer Type')+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values = VHL_disease_colors)+geom_jitter(height=0,width = 0.05,aes(color=conseq))+scale_color_manual(values = named_conseq_colors)+geom_hline(yintercept = VHLxrLD_gs_path_cutoff,lty=3)+geom_hline(yintercept = VHLxrLD_gs_neut_cutoff,lty=3)

VHL_disease_type_df$delta_rna <- as.numeric(as.character(VHL_disease_type_df$delta_rna))
final_clinical_df$delta_rna <- as.numeric(as.character(final_clinical_df$delta_rna))

VHL_disease_colors4 <- c("ccRCC only" = "#E0004B","Pheo predominant" = "#00BA74","Pheo mixed" = "#FD9100", "Unspecified Pathogenic"  = "#D0D0D0","Not pathogenic" = "#858585","delta_rna" = "#B6B6B6","STOP_GAINED" = "#BD0016","CANONICAL_SPLICE" = "#FFAA38","INTRONIC" = "#88CCEE", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#15327C", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#CB57D7","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#EA93F3", "REF" = "#000000")

final_clinical_df$vhl_disease_type <- factor(final_clinical_df$vhl_disease_type, levels = c("ccRCC only","Pheo predominant","Pheo mixed","Unspecified Pathogenic","Not pathogenic"))

#CANFIG
ggplot(final_clinical_df,aes(x=vhl_disease_type, y=function_score_final))+geom_boxplot(aes(color="Not pathogenic"),outlier.shape=NA)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('Cancer association')+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values = VHL_disease_colors4)+geom_jitter(data=final_clinical_df[which(!is.na(final_clinical_df$vhl_disease_type)),],height=0,width = 0.15,aes(color=conseq))+ geom_jitter(data=final_clinical_df[which(is.na(final_clinical_df$vhl_disease_type)),],height=0,width = 0.25,aes(color=conseq,alpha=0.2))+coord_cartesian(ylim=c(-4,1))

VHL_disease_colors <- c("ccRCC only" = "#D56209","Pheo predominant" = "#05837A","Pheo mixed" = "#6666C8", "Unspecified"  = "#A0A0F2","Not assigned" = "#B6B6B6")

```

Mapping indels in cBP to function score data, Figure 5d.

```{r}

#plot to create Figure 5d
VHL_cBioPortal_curated <- read.csv("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/20220728_cBioPortal_curated.csv")

#boundaries of exon: only include indels completely contained within the exon:
min(final_clinical_df[which((final_clinical_df$sge_region == "exon 3a" | final_clinical_df$sge_region == "exon 3b") & is.na(final_clinical_df$Intron)),c("pos")])
#10191471

max(final_clinical_df[which((final_clinical_df$sge_region == "exon 3a" | final_clinical_df$sge_region == "exon 3b") & (final_clinical_df$conseq == "STOP_LOST")),c("pos")])
#10191649

# new cBioPortal df containing only indels entirely within coding sequence,
cBP_x3_rcc_indels_df <- VHL_cBioPortal_curated[VHL_cBioPortal_curated$Variant.Type %in% c("DEL","INS") & (VHL_cBioPortal_curated$Start.Pos >= 10191471 & VHL_cBioPortal_curated$End.Pos < 10191649+10) & VHL_cBioPortal_curated$Cancer.Type.Detailed %in% c("Renal Clear Cell Carcinoma","Renal Clear Cell Carcinoma with Sarcomatoid Features"),]

cBP_x3_rcc_indels_df$del_length <- (-cBP_x3_rcc_indels_df$Start.Pos + cBP_x3_rcc_indels_df$End.Pos)+1
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type %in% c("INS")),]$del_length <- NA

cBP_x3_rcc_indels_df$ins_length <- NA
cBP_x3_rcc_indels_df[cBP_x3_rcc_indels_df$Variant.Type %in% c("INS"),]$ins_length <- nchar(as.character(cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type %in% c("INS")),]$Var))

cBP_x3_rcc_indels_df$frameshift <- 0
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type %in% c("INS")),]$frameshift <- cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type %in% c("INS")),]$ins_length %% 3
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type %in% c("DEL")),]$frameshift <- (cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type %in% c("DEL")),]$del_length %% 3)*-1

cBP_x3_rcc_indels_df$reading_frame <- 0
#reading frame is defined here at being +1 or +2 bases moved forward (e.g. 2 bp deletion leads to the frame being shifted 1 bp forward)
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$frameshift == -2| cBP_x3_rcc_indels_df$frameshift == 1),]$reading_frame <- 1 
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$frameshift == -1| cBP_x3_rcc_indels_df$frameshift == 2),]$reading_frame <- 2 

cBP_x3_rcc_indels_df$height_nudge <- 0

cBP_x3_rcc_indels_df$ins_pos_rank <- NA
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "INS"),]$ins_pos_rank <- rank( cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "INS"),]$Start.Pos+cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "INS"),]$End.Pos/1000,ties.method="random")
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "INS"),]$height_nudge <- cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "INS"),]$ins_pos_rank %%8/24
cBP_x3_rcc_indels_df$del_pos_rank <- NA
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "DEL"),]$del_pos_rank <- rank( cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "DEL"),]$Start.Pos+cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "DEL"),]$End.Pos/1000,ties.method="random")
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "DEL"),]$height_nudge <- cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$Variant.Type == "DEL"),]$del_pos_rank %%20/(20*1.74)

cBP_x3_rcc_indels_df$frame_nudge <- 0

cBP_x3_rcc_indels_df$frame1_pos_rank <- NA
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1),]$frame1_pos_rank <- rank( cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1),]$Start.Pos+cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1),]$End.Pos/1000,ties.method="random")

cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1),]$frame_nudge <- cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1),]$frame1_pos_rank %%6/13

cBP_x3_rcc_indels_df$frame2_pos_rank <- NA
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2),]$frame2_pos_rank <- rank( cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2),]$Start.Pos+cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2),]$End.Pos/1000,ties.method="random")
cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2),]$frame_nudge <- cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2),]$frame2_pos_rank %%6/13

VHL_disease_colors <- c("ccRCC only" = "#D56209","Pheo predominant" = "#05837A","Pheo mixed" = "#6666C8", "Unspecified"  = "#A0A0F2","Not assigned" = "#B6B6B6")
reading_frame_colors = c("1" = "#D56209", "2" = "#05837A","0" = "#999999")
reading_frame_disease_colors <- c("ccRCC only" = "#D56209","Pheo predominant" = "#05837A","Pheo mixed" = "#6666C8", "Unspecified"  = "#A0A0F2","Not assigned" = "#B6B6B6","1" = "#D56209", "2" = "#05837A","0" = "#999999")

named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus_rf = c("STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#979090","Benign" = "#0433FF","Conflicting interpretations of pathogenicity" = "#936104","Likely benign" = "#486BFC","Likely pathogenic" = "#C56B99","Pathogenic" = "#882255","Uncertain significance" = "#DCA237" ,"1" = "#D56209", "2" = "#05837A","0" = "#999999","INS" = "#456CCC","DEL" = "#F3001C")

named_conseq_colors_indels = c("STOP_GAINED" = "#BD0016","CANONICAL_SPLICE" = "#FFAA38","INTRONIC" = "#88CCEE", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#15327C", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#CB57D7","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#EA93F3", "REF" = "#000000","1" = "#D56209", "2" = "#05837A","0" = "#999999","INS" = "#456CCC","DEL" = "#F3001C")
named_conseq_colors_ns_highlight_rf = c("STOP_GAINED" = "#BD0016",
  "CANONICAL_SPLICE" = "#EBEAEA",
  "INTRONIC" = "#EBEAEA",
  "NON_SYNONYMOUS" = "#EBEAEA",
  "SPLICE_SITE" = "#EBEAEA",
  "SYNONYMOUS" = "#EBEAEA",
  "STOP_LOST" = "#CB57D7",
  "REGULATORY" = "#EBEAEA",
  "5PRIME_UTR" = "#EBEAEA",
  "3PRIME_UTR" = "#EBEAEA",
  "REF" = "#EBEAEA","1" = "#D56209", "2" = "#05837A","0" = "#999999","INS" = "#456CCC","DEL" = "#850010")

VHL_disease_sig_indel_colors <- c("ccRCC only" = "#D56209","Pheo predominant" = "#05837A","Pheo mixed" = "#6666C8", "Unspecified"  = "#A0A0F2","FDR < 0.01" = "#000000",
  "FDR < 0.05" = "#B6B6B6",
  "FDR < 0.2" = "#B6B6B6",
  "FDR < 0.5" = "#B6B6B6",
  "FDR > 0.5" = "#B6B6B6","INS" = "#456CCC","DEL" = "#850010","STOP_GAINED" = "#BD0016","CANONICAL_SPLICE" = "#FFAA38","INTRONIC" = "#88CCEE", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#15327C", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#CB57D7","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#EA93F3")

#highlighting nonsense only:
ggplot(final_clinical_df[which((final_clinical_df$sge_region == "exon 3a")|final_clinical_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors_ns_highlight_rf)+coord_cartesian(ylim=c(-3.5,3))+scale_y_continuous(breaks=c(-3,-2,-1,0))+scale_x_continuous(breaks = c(10191507,10191557,10191607,10191646),labels = c('500','550','600','639'))+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1 & cBP_x3_rcc_indels_df$Variant.Type == "DEL"),],aes(y=0.6+frame_nudge,yend=0.6+frame_nudge,x=Start.Pos,xend=Start.Pos+del_length,color=Variant.Type),size=01.0)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1 & cBP_x3_rcc_indels_df$Variant.Type == "INS"),],aes(y=0.6+frame_nudge,yend=0.6+frame_nudge,x=Start.Pos,xend=Start.Pos+ins_length,color=Variant.Type),size=01.0)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2 & cBP_x3_rcc_indels_df$Variant.Type == "DEL"),],aes(y=2+frame_nudge,yend=2+frame_nudge,x=Start.Pos,xend=Start.Pos+del_length,color=Variant.Type),size=01.0)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2 & cBP_x3_rcc_indels_df$Variant.Type == "INS"),],aes(y=2+frame_nudge,yend=2+frame_nudge,x=Start.Pos,xend=Start.Pos+ins_length,color=Variant.Type),size=01.0)+ geom_vline(xintercept=min(final_clinical_df[final_clinical_df$protPos %in% c("201"),]$pos-1),lty=2,color="Lightgray")

#conseq+exon3b_with_indels_from_cbp - CANFIG 
ggplot(final_clinical_df[which((final_clinical_df$sge_region == "exon 3b")|final_clinical_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final,color=conseq))+geom_point(size=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "none")+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=named_conseq_colors_indels)+coord_cartesian(ylim=c(-3.2,2.5),xlim=c(min(final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"),]$pos),max(final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"),]$pos)))+scale_y_continuous(breaks=c(-3,-2,-1,0))+scale_x_continuous(breaks = c(10191557,10191607,10191646),labels = c('550','600','639'))+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1 & cBP_x3_rcc_indels_df$Variant.Type == "DEL"),],aes(y=0.7+frame_nudge,yend=0.7+frame_nudge,x=Start.Pos,xend=Start.Pos+del_length,color=Variant.Type),size=01.0)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1 & cBP_x3_rcc_indels_df$Variant.Type == "INS"),],aes(y=0.7+frame_nudge,yend=0.7+frame_nudge,x=Start.Pos,xend=Start.Pos+ins_length,color=Variant.Type),size=01.0)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2 & cBP_x3_rcc_indels_df$Variant.Type == "DEL"),],aes(y=1.5+frame_nudge,yend=1.5+frame_nudge,x=Start.Pos,xend=Start.Pos+del_length,color=Variant.Type),size=01.0)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2 & cBP_x3_rcc_indels_df$Variant.Type == "INS"),],aes(y=1.5+frame_nudge,yend=1.5+frame_nudge,x=Start.Pos,xend=Start.Pos+ins_length,color=Variant.Type),size=01.0)+
geom_point(data=final_clinical_df[which((final_clinical_df$sge_region == "exon 3b")&(final_clinical_df$total_cBP_ccRCC >= 1)),],pch=21,aes(color="DEL"),size = 3)+
geom_point(data=final_clinical_df[which((final_clinical_df$sge_region == "exon 3b")&(final_clinical_df$vhl_disease_type %in% c("Pheo predominant","Pheo mixed","Unspecified"))),],pch=21,aes(color="INS"),size = 4)

#FF export as 4 x 10 VHLrLD_x3_indels_by_frame Final Fig. 5d, showing all LoF entries in cBP circle:
ggplot(final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final))+geom_point(data=final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"&final_clinical_df$fdr > 0.00),],aes(color="FDR > 0.5"))+geom_point(data=final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"&( final_clinical_df$conseq %in% c("STOP_GAINED","STOP_LOST"))),],aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = 'None')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=VHL_disease_sig_indel_colors)+coord_cartesian(ylim=c(-3.2,2.5),xlim=c(min(final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"),]$pos),max(final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"),]$pos)))+scale_y_continuous(breaks=c(-3,-2,-1,0))+scale_x_continuous(breaks = c(10191557,10191607,10191646),labels = c('550','600','639'))+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1 & cBP_x3_rcc_indels_df$Variant.Type == "DEL"),],aes(y=0.9+frame_nudge,yend=0.9+frame_nudge,x=Start.Pos-0.5,xend=Start.Pos+del_length-0.5,color="STOP_GAINED",alpha=0.5),size=01.25)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1 & cBP_x3_rcc_indels_df$Variant.Type == "INS"),],aes(y=0.9+frame_nudge,yend=0.9+frame_nudge,x=Start.Pos-0.5,xend=Start.Pos+ins_length-0.5,color="SPLICE_SITE"),size=01.25)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2 & cBP_x3_rcc_indels_df$Variant.Type == "DEL"),],aes(y=1.9+frame_nudge,yend=1.9+frame_nudge,x=Start.Pos-0.5,xend=Start.Pos+del_length-0.5,color="STOP_GAINED",alpha=0.5),size=01.25)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2 & cBP_x3_rcc_indels_df$Variant.Type == "INS"),],aes(y=1.9+frame_nudge,yend=1.9+frame_nudge,x=Start.Pos-0.5,xend=Start.Pos+ins_length-0.5,color="SPLICE_SITE"),size=01.25)+annotate("rect", xmin = 10191406.5, xmax = 10191800.5, ymin = 0.75, ymax = 1.65,alpha = .1)+geom_point(data=final_clinical_df[which((final_clinical_df$sge_region == "exon 3b")&(final_clinical_df$total_cBP_ccRCC > 0)&(final_clinical_df$fdr < 0.01)),],pch=21,aes(color="SPLICE_SITE",size = 3,stroke=1.0))+geom_vline(xintercept=10191607,lty=2,alpha=0.5)

#legend only:
ggplot(final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"),], aes(x=as.numeric(as.character(pos)), y=function_score_final))+geom_point(data=final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"&final_clinical_df$fdr > 0.00),],aes(color="FDR > 0.5"))+geom_point(data=final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"&( final_clinical_df$conseq %in% c("STOP_GAINED","STOP_LOST"))),],aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white')) +theme(legend.title=element_blank(),legend.key = element_blank())+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors)+scale_color_manual(values=VHL_disease_sig_indel_colors)+coord_cartesian(ylim=c(-3.2,2.5),xlim=c(min(final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"),]$pos),max(final_clinical_df[which(final_clinical_df$sge_region == "exon 3b"),]$pos)))+scale_y_continuous(breaks=c(-3,-2,-1,0))+scale_x_continuous(breaks = c(10191557,10191607,10191646),labels = c('550','600','639'))+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1 & cBP_x3_rcc_indels_df$Variant.Type == "DEL"),],aes(y=0.9+frame_nudge,yend=0.9+frame_nudge,x=Start.Pos-0.5,xend=Start.Pos+del_length-0.5,color="STOP_GAINED",alpha=0.5),size=01.25)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 1 & cBP_x3_rcc_indels_df$Variant.Type == "INS"),],aes(y=0.9+frame_nudge,yend=0.9+frame_nudge,x=Start.Pos-0.5,xend=Start.Pos+ins_length-0.5,color="SPLICE_SITE"),size=01.25)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2 & cBP_x3_rcc_indels_df$Variant.Type == "DEL"),],aes(y=1.9+frame_nudge,yend=1.9+frame_nudge,x=Start.Pos-0.5,xend=Start.Pos+del_length-0.5,color="STOP_GAINED",alpha=0.5),size=01.25)+ geom_segment(data=cBP_x3_rcc_indels_df[which(cBP_x3_rcc_indels_df$reading_frame == 2 & cBP_x3_rcc_indels_df$Variant.Type == "INS"),],aes(y=1.9+frame_nudge,yend=1.9+frame_nudge,x=Start.Pos-0.5,xend=Start.Pos+ins_length-0.5,color="SPLICE_SITE"),size=01.25)+annotate("rect", xmin = 10191406.5, xmax = 10191800.5, ymin = 0.75, ymax = 1.65,alpha = .1)+geom_point(data=final_clinical_df[which((final_clinical_df$sge_region == "exon 3b")&(final_clinical_df$total_cBP_ccRCC > 0)&(final_clinical_df$fdr < 0.01)),],pch=21,aes(color="SPLICE_SITE",size = 3,stroke=1.0))+geom_vline(xintercept=10191607,lty=2,alpha=0.5)

```

analysing vhldb variant types from parsing (not included due to inconsistency in naming)
```{r}
#Filter all entries on presence in ClinVar, using master_disease_type_df because no recessive, not in 2 or more pop individuals, and must be clinvar path/likely path

length(master_disease_type_df[,c("cHGVS")])

#core set of Type 1 variants by VHL-DB
vhldb_t1_only_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_t1_only == "True"),c("cHGVS")]
vhldb_rcc_only_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_pheo_count == 0 & master_disease_type_df$vhldb_rcc_count > 0),c("cHGVS")]
vhldb_t1_only_or_rcc_only_by_vhldb <- union(vhldb_rcc_only_by_vhldb,vhldb_t1_only_by_vhldb)

#core set of Type 2 variants by VHL-DB (not including type 2B due to high pheo risk)
vhldb_t2c_only_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_t2c_only == "True"),c("cHGVS")]
vhldb_t2a_only_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_t2a_only == "True"),c("cHGVS")]
vhldb_t2b_only_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_t2b_only == "True"),c("cHGVS")]
vhldb_t2_only_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_t2_only == "True"),c("cHGVS")]
vhldb_t2_only_not2b_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_t2_only == "True" & master_disease_type_df$vhldb_t2b_count == 0),c("cHGVS")]
vhldb_pheo_predom_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_pheo_count > master_disease_type_df$vhldb_rcc_count),c("cHGVS")]

vhldb_t2_pheo_predom_by_vhldb <- setdiff(union(vhldb_t2_only_not2b_by_vhldb,vhldb_pheo_predom_by_vhldb),vhldb_t1_only_or_rcc_only_by_vhldb)

the_rest_by_vhldb <- setdiff(master_disease_type_df[,c("cHGVS")],union(vhldb_t2_pheo_predom_by_vhldb,vhldb_t1_only_or_rcc_only_by_vhldb))


#been deemed type 2 at least one
vhldb_t1_any_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_t1_count > 0),c("cHGVS")]
vhldb_t2_any_by_vhldb <- master_disease_type_df[which(master_disease_type_df$vhldb_t2_count > 0),c("cHGVS")]

t2_only_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% c(vhldb_t2_only_by_vhldb),]
t2_only_not2b_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% c(vhldb_t2_only_not2b_by_vhldb),]
t2_pheo_predominant_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% c(vhldb_t2_pheo_predom_by_vhldb),]
t2_any_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% c(vhldb_t2_any_by_vhldb),]

#NOTE:  only 1 variant in the type 2 only set hasn't been seen in a single ccRCC case in cBP, suggesting lack of pathogenic variants that truly cause no increased ccRCC risk.
t2_only_no_ccrcc_vhldb_df <- master_disease_type_df[which(master_disease_type_df$cHGVS %in% c(vhldb_t2_only_by_vhldb) & master_disease_type_df$total_cBP_rcc == 0),]

t2a_only_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% vhldb_t2a_only_by_vhldb,]
t2ac_only_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% union(vhldb_t2a_only_by_vhldb,vhldb_t2c_only_by_vhldb),]
t2b_only_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% vhldb_t2b_only_by_vhldb,]
t1_only_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% vhldb_t1_only_by_vhldb,]
t12b_only_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% union(vhldb_t1_only_by_vhldb,vhldb_t2_only_by_vhldb),]


###

t1_or_rcc_only_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% vhldb_t1_only_or_rcc_only_by_vhldb,]
t2_pheo_predom_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% vhldb_t2_pheo_predom_by_vhldb,]
the_rest_by_vhldb_df <- master_disease_type_df[master_disease_type_df$cHGVS %in% the_rest_by_vhldb,]

t1_or_rcc_only_vhldb_df$vhl_db_annotation <- "Pathogenic, type 1"
t2_pheo_predom_vhldb_df$vhl_db_annotation <- "Pathogenic, pheo-predominant"
the_rest_by_vhldb_df$vhl_db_annotation <- "Pathogenic, no specific type"
#fs on difference in functinon score between t1 and pheo-predom. snvs.
median(t1_or_rcc_only_vhldb_df$function_score_final)
median(t2_pheo_predom_vhldb_df$function_score_final)

VHLrLD_by_vhldb_type_df <- rbind(t1_or_rcc_only_vhldb_df,t2_pheo_predom_vhldb_df,the_rest_by_vhldb_df)

final_clinical_df <- merge(final_clinical_df, VHLrLD_by_vhldb_type_df[,c("vhl_db_annotation","cHGVS")], by="cHGVS",all= FALSE,all.x = TRUE, all.y=FALSE)

final_clinical_df[is.na(final_clinical_df$vhl_db_annotation),]$vhl_db_annotation <- "Not pathogenic"
final_clinical_df$vhl_db_annotation <- factor(final_clinical_df$vhl_db_annotation, levels = c("Pathogenic, type 1","Pathogenic, pheo-predominant","Pathogenic, no specific type","Not pathogenic"))


VHL_disease_colors4 <- c("ccRCC only" = "#E0004B","Pheo predominant" = "#00BA74","Pheo mixed" = "#FD9100", "Unspecified Pathogenic"  = "#D0D0D0","Not pathogenic" = "#858585","delta_rna" = "#B6B6B6","STOP_GAINED" = "#e83677","CANONICAL_SPLICE" = "#f7a83c","INTRONIC" = "#8acdef", "NON_SYNONYMOUS" = "#64bb97", "SPLICE_SITE" = "#253673", "SYNONYMOUS" = "#4e77bb","STOP_LOST" = "#964593","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#d091bf", "REF" = "#000000")


#FF Figure 5c export 3x6 VHLrLD_fs_by_vhldb_disease_type - to include all 
ggplot(final_clinical_df, aes(x=vhl_db_annotation, y=function_score_final))+geom_jitter(width=0.1,height=0,aes(color=conseq))+geom_boxplot(aes(color="Not pathogenic",alpha=0),outlier.shape=NA)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('Cancer Type')+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values = VHL_disease_colors4)+scale_y_continuous(limits=c(-4,1))

#Possible Supplementary Table on clinical classification of VHLdb path/LP SNVs:
#write.table(VHLrLD_by_vhldb_type_df[,c("cHGVS","pHGVS","clinvar_simple","Cancer_type_single","vhl_db_annotation","function_score_final","rna_score","conseq","combined_pop_count","total_cBP_ccRCC","total_cBP_other","vhldb_syndrome_count","vhldb_t1_count","vhldb_t2_count","vhldb_t2a_count","vhldb_t2b_count","vhldb_t2c_count","vhldb_pheo_count","vhldb_rcc_count","vhldb_hb_count","vhldb_polycyt_count")], "/Users/findlag/Downloads/vhldb_classes.txt", sep="\t",row.names=F)

####
vhldb_colors <- c("Type 1, only" = "#D56209","Type 2, pheo-predominant" = "#05837A" ,"Type 2B, only" = "#D56209","Type 2, any" = "#6666C8","Type 2, not Type 1" = "#05837A", "Type 2A or 2C, only"  = "#05837A","Type 2, not Type 1 or 2b" = "#05837A")

VHL_disease_colors5 <- c("ccRCC only" = "#E0004B","Pheo predominant" = "#00BA74","Pheo mixed" = "#FD9100", "Unspecified Pathogenic"  = "#D0D0D0","Not pathogenic" = "#858585","delta_rna" = "#B6B6B6","STOP_GAINED" = "#BD0016","CANONICAL_SPLICE" = "#FFAA38","INTRONIC" = "#88CCEE", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#15327C", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#CB57D7","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#EA93F3", "REF" = "#000000","STOP_GAINED" = "#F3001C","CANONICAL_SPLICE" = "#FD9100","INTRONIC" = "#456CCC", "NON_SYNONYMOUS" = "#00BA74", "SPLICE_SITE" = "#456CCC", "SYNONYMOUS" = "#456CCC","STOP_LOST" = "#99006B","REGULATORY"="#B5A695","5PRIME_UTR"="#B5A695", "3PRIME_UTR"="#3E6BAA", "REF" = "#000000", "absent" = "#979090","Benign" = "#0433FF","Conflicting interpretations of pathogenicity" = "#936104","Likely benign" = "#486BFC","Likely pathogenic" = "#C56B99","Pathogenic" = "#882255","Uncertain significance" = "#DCA237" )

```

Analysis of nonsense variants
```{r}

named_conseq_colors_ns_highlight = c("STOP_GAINED" = "#BD0016", "CANONICAL_SPLICE" = "#EBEAEA", "INTRONIC" = "#EBEAEA","NON_SYNONYMOUS" = "#EBEAEA","SPLICE_SITE" = "#EBEAEA","SYNONYMOUS" = "#EBEAEA","STOP_LOST" = "#CB57D7","REGULATORY" = "#EBEAEA","5PRIME_UTR" = "#EBEAEA","3PRIME_UTR" = "#EBEAEA","REF" = "#EBEAEA")

#SFig. 9 work
#map of nonsense variants across the gene #CANFIG Final Fig. 5
final_clinical_df$CDSpos <- as.numeric(as.character(final_clinical_df$CDSpos))

exon1_min_cds <- min(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 1b")),]$CDSpos)
exon1_min_pos <- min(final_clinical_df[which(final_clinical_df$CDSpos == exon1_min_cds),]$pos)
exon1_max_cds <- max(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 1a")),]$CDSpos)
exon1_max_pos <- max(final_clinical_df[which(final_clinical_df$CDSpos == exon1_max_cds),]$pos)
exon1_CDS_breaks <- append(seq(exon1_min_pos,exon1_max_pos,by=25),exon1_max_pos)
exon1_CDS_labels <- append(seq(exon1_min_cds,exon1_max_cds,by=25),exon1_max_cds)

exon2_min_cds <- min(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 2")),]$CDSpos)
exon2_min_pos <- min(final_clinical_df[which(final_clinical_df$CDSpos == exon2_min_cds),]$pos)
exon2_max_cds <- max(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 2")),]$CDSpos)
exon2_max_pos <- max(final_clinical_df[which(final_clinical_df$CDSpos == exon2_max_cds),]$pos)
exon2_CDS_breaks <- append(seq(exon2_min_pos,exon2_max_pos,by=25),exon2_max_pos)
exon2_CDS_labels <- append(seq(exon2_min_cds,exon2_max_cds,by=25),exon2_max_cds)

exon3_min_cds <- min(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 3a")),]$CDSpos)
exon3_min_pos <- min(final_clinical_df[which(final_clinical_df$CDSpos == exon3_min_cds),]$pos)
exon3_max_cds <- max(final_clinical_df[which(!is.na(final_clinical_df$CDSpos)),]$CDSpos)
exon3_max_pos <- max(final_clinical_df[which(final_clinical_df$CDSpos == exon3_max_cds),]$pos)
exon3_CDS_breaks <- append(seq(exon3_min_pos,exon3_max_pos,by=25),exon3_max_pos)
exon3_CDS_labels <- append(seq(exon3_min_cds,exon3_max_cds,by=25),exon3_max_cds)

VHLrLD_exon1_df <- final_clinical_df[which(final_clinical_df$pos < VHLx1p_start),]
VHLrLD_exon1_ns_map <- ggplot(VHLrLD_exon1_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(data=VHLrLD_exon1_df[which(VHLrLD_exon1_df$conseq != "STOP_GAINED"),],alpha=1)+geom_point(data=VHLrLD_exon1_df[which(VHLrLD_exon1_df$conseq == "STOP_GAINED"),],alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_ns_highlight)+coord_cartesian(ylim=c(-4,1))+ scale_x_continuous(breaks=exon1_CDS_breaks,labels=exon1_CDS_labels)
VHLrLD_exon1_ns_map

VHLrLD_exon2_df <- final_clinical_df[which(final_clinical_df$sge_region == "exon 2"),]
VHLrLD_exon2_ns_map <- ggplot(VHLrLD_exon2_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(data=VHLrLD_exon2_df[which(VHLrLD_exon2_df$conseq != "STOP_GAINED"),],alpha=1)+geom_point(data=VHLrLD_exon2_df[which(VHLrLD_exon2_df$conseq == "STOP_GAINED"),],alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_ns_highlight)+coord_cartesian(ylim=c(-4,1))+ scale_x_continuous(breaks=exon2_CDS_breaks,labels=exon2_CDS_labels)
VHLrLD_exon2_ns_map

VHLrLD_exon3_df <- final_clinical_df[which(final_clinical_df$pos > VHLx3a_start-50),]
VHLrLD_exon3_ns_map <- ggplot(VHLrLD_exon3_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(data=VHLrLD_exon3_df[which(VHLrLD_exon3_df$conseq != "STOP_GAINED"),],alpha=1)+geom_point(data=VHLrLD_exon3_df[which(VHLrLD_exon3_df$conseq == "STOP_GAINED"),],alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_ns_highlight)+coord_cartesian(ylim=c(-4,1))+ scale_x_continuous(breaks=exon3_CDS_breaks,labels=exon3_CDS_labels)
VHLrLD_exon3_ns_map

VHLrLD_ns_only_dot_legend <- ggplot(VHLrLD_exon3_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(data=VHLrLD_exon3_df[which(VHLrLD_exon3_df$conseq != "STOP_GAINED"),],alpha=1)+geom_point(data=VHLrLD_exon3_df[which(VHLrLD_exon3_df$conseq == "STOP_GAINED"),],alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+ylab('Function score')+xlab('CDS position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=10),axis.text.y = element_text(size=10),axis.text.x = element_text(size=10),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_ns_highlight,labels=conseq_labels)+coord_cartesian(ylim=c(-4,1))+ scale_x_continuous(breaks=exon3_CDS_breaks,labels=exon3_CDS_labels)
VHLrLD_ns_only_dot_legend

#FF export VHLrLD_stop_codon_map SFig. 8a #CANFIG Supp. Fig 6 x 12 (shrink in 2)
grid.arrange(VHLrLD_exon1_ns_map,VHLrLD_exon2_ns_map,VHLrLD_exon3_ns_map,nrow=3)

write.table(final_clinical_df[which(final_clinical_df$CDSpos > 160 & final_clinical_df$CDSpos < 570 & final_clinical_df$conseq == "STOP_GAINED"),c("cHGVS","pHGVS","function_score_final","fdr","rna_score","post_rna_score","conseq","total_cBP_ccRCC","COSMIC","clinvar_simple","combined_pop_count","vhldb_germline_count","vhldb_somatic_count","vhldb_syndrome_count","vhldb_rcc_count","vhldb_pheo_count","vhldb_hb_count","vhldb_t1_count","vhldb_t2_count","vhldb_t2a_count","vhldb_t2b_count","vhldb_t2c_count")], "/Users/findlag/Downloads/VHL_stop_codons.txt", sep="\t",row.names=F)

VHL_stops_df <- final_clinical_df[which(final_clinical_df$CDSpos > 160 & final_clinical_df$CDSpos < 570 & final_clinical_df$conseq == "STOP_GAINED"),]

VHL_stops_df$codon_position <- (as.numeric(as.character(VHL_stops_df$CDSpos))+2)%%(as.numeric(as.character(VHL_stops_df$protPos))*3)

VHL_stops_df$codon_ref <- "NNN"
VHL_stops_df$codon_var <- "NNN"

VHL_stops_df$codon_next_base <- "N"

vhl_ref_CDS <- "ATGCCCCGGAGGGCGGAGAACTGGGACGAGGCCGAGGTAGGCGCGGAGGAGGCAGGCGTCGAAGAGTACGGCCCTGAAGAAGACGGCGGGGAGGAGTCGGGCGCCGAGGAGTCCGGCCCGGAAGAGTCCGGCCCGGAGGAACTGGGCGCCGAGGAGGAGATGGAGGCCGGGCGGCCGCGGCCCGTGCTGCGCTCGGTGAACTCGCGCGAGCCCTCCCAGGTCATCTTCTGCAATCGCAGTCCGCGCGTCGTGCTGCCCGTATGGCTCAACTTCGACGGCGAGCCGCAGCCCTACCCAACGCTGCCGCCTGGCACGGGCCGCCGCATCCACAGCTACCGAGGTCACCTTTGGCTCTTCAGAGATGCAGGGACACACGATGGGCTTCTGGTTAACCAAACTGAATTATTTGTGCCATCTCTCAATGTTGACGGACAGCCTATTTTTGCCAATATCACACTGCCAGTGTATACTCTGAAAGAGCGATGCCTCCAGGTTGTCCGGAGCCTAGTCAAGCCTGAGAATTACAGGAGACTGGACATCGTCAGGTCGCTCTACGAAGATCTGGAAGACCACCCAAATGTGCAGAAAGACCTGGAGCGGCTGACACAGGAGCGCATTGCACATCAACGGATGGGAGATTGA"

# TAA, TAG and TGA
VHL_stops_df$vhl_ref_CDS <- "ATGCCCCGGAGGGCGGAGAACTGGGACGAGGCCGAGGTAGGCGCGGAGGAGGCAGGCGTCGAAGAGTACGGCCCTGAAGAAGACGGCGGGGAGGAGTCGGGCGCCGAGGAGTCCGGCCCGGAAGAGTCCGGCCCGGAGGAACTGGGCGCCGAGGAGGAGATGGAGGCCGGGCGGCCGCGGCCCGTGCTGCGCTCGGTGAACTCGCGCGAGCCCTCCCAGGTCATCTTCTGCAATCGCAGTCCGCGCGTCGTGCTGCCCGTATGGCTCAACTTCGACGGCGAGCCGCAGCCCTACCCAACGCTGCCGCCTGGCACGGGCCGCCGCATCCACAGCTACCGAGGTCACCTTTGGCTCTTCAGAGATGCAGGGACACACGATGGGCTTCTGGTTAACCAAACTGAATTATTTGTGCCATCTCTCAATGTTGACGGACAGCCTATTTTTGCCAATATCACACTGCCAGTGTATACTCTGAAAGAGCGATGCCTCCAGGTTGTCCGGAGCCTAGTCAAGCCTGAGAATTACAGGAGACTGGACATCGTCAGGTCGCTCTACGAAGATCTGGAAGACCACCCAAATGTGCAGAAAGACCTGGAGCGGCTGACACAGGAGCGCATTGCACATCAACGGATGGGAGATTGA"

VHL_stops_df <- VHL_stops_df %>%                                                                       
  mutate(codon_ref = case_when(
    (codon_position == 0) ~ substr(vhl_ref_CDS, CDSpos, CDSpos+2),
    (codon_position == 1) ~ substr(vhl_ref_CDS, CDSpos-1, CDSpos+1),
    (codon_position == 2) ~ substr(vhl_ref_CDS, CDSpos-2, CDSpos)
  ))

VHL_stops_df <- VHL_stops_df %>% 
  mutate(codon_var = case_when(
    codon_position == 0 ~ paste(alt,substr(vhl_ref_CDS, CDSpos+1, CDSpos+2),sep = ''),
    codon_position == 1 ~ paste(paste(substr(vhl_ref_CDS,CDSpos-1,CDSpos-1),alt,sep = ''),substr(vhl_ref_CDS,CDSpos+1,CDSpos+1),sep = ''),
    codon_position == 2 ~ paste(substr(vhl_ref_CDS, CDSpos-2, CDSpos-1),alt,sep = '')
  ))

VHL_stops_df <- VHL_stops_df %>% 
  mutate(codon_next_base = case_when(
    codon_position == 0 ~ substr(vhl_ref_CDS, CDSpos+3, CDSpos+3),
    codon_position == 1 ~ substr(vhl_ref_CDS, CDSpos+2, CDSpos+2),
    codon_position == 2 ~ substr(vhl_ref_CDS, CDSpos+1, CDSpos+1)
  ))

VHL_nonsense_mid_fs_by_codon <- ggplot(VHL_stops_df, aes(fill=conseq,x=codon_var, y=as.numeric(as.character(function_score_final))))+geom_boxplot(aes(),outlier.shape=NA)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('Stop codon')+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+geom_jitter(height=0,width = 0.1,size=1)+scale_color_manual(values = significance_colors3)

VHL_nonsense_mid_rna_by_codon <- ggplot(VHL_stops_df, aes(fill=conseq,x=codon_var, y=as.numeric(as.character(rna_score))))+geom_boxplot(aes(),outlier.shape=NA)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('Stop codon')+ylab('RNA score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors,labels=conseq_labels)+geom_jitter(height=0,width = 0.1,size=1)+scale_color_manual(values = significance_colors3)

#FS SFig. 9 
nonsense_codon_aov <- aov(function_score_final ~ codon_var, data = VHL_stops_df)
# Summary of the analysis
summary(nonsense_codon_aov)
TukeyHSD(nonsense_codon_aov)

VHL_stops_df$stop_codon_4bp <- paste0(VHL_stops_df$codon_var,VHL_stops_df$codon_next_base) 

#FF Final Figure SFig. Fig. S9 -- nonsense variants by 4bp context, 4x8
ggplot(VHL_stops_df, aes(x=stop_codon_4bp, y=as.numeric(as.character(function_score_final))))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Stop codon')+ylab('Function score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+geom_point(aes(color=clinvar_simple,size=total_cBP_ccRCC))+scale_color_manual(values = named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+geom_text(data=VHL_stops_df[which(as.numeric(as.character(VHL_stops_df$fdr)) > 0.01),],aes(label=cHGVS,size=2),vjust=1.5)+geom_point(data = VHL_stops_df %>% 
      group_by(stop_codon_4bp) %>% 
      summarise(mean_4bp_score = mean(function_score_final)),
      mapping = aes(y = mean_4bp_score, x = stop_codon_4bp), 
      size = 8,color = 'Black', shape = '-') 

####

```

Other plots
```{r}

### UKB plot (only UKB)
ggplot(clinvar_ord_final_filtered_no1c_df, aes(x=function_score_final,y=log10(ukb_association_ac+1)))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$ukb_association_ac >= 1),],alpha=1,height=0.02,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('Allele Count UKB, Log10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$ukb_association_ac == 0),],alpha=0.75,height=.1,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('Allele Count UKB, Log10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#Bravo plot
ggplot(clinvar_dots_final_filtered_no1c_df, aes(x=function_score_final,y=log2(bravo_het_count+1)))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$bravo_het_count >= 1),],alpha=1,height=0.03,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('Allele Count UKB, Log10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$bravo_het_count == 0),],alpha=0.75,height=.2,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('Bravo allele count, Log10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(clinvar_dots_final_filtered_no1c_df, aes(x=function_score_final,y=log10(bravo_freq)))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$bravo_het_count >= 1),],alpha=1,height=0.00,aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('Bravo AF, Log10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$bravo_het_count == 0),],alpha=0.5,height=1,aes(color=conseq))

#gnomAD plots gv2, gv3
ggplot(clinvar_dots_final_filtered_no1c_df, aes(x=function_score_final,y=log10(gv2_allele_freq)))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$gv2_allele_freq > 0),],alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('gnomADv2 AF, Log10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$gv2_allele_freq == 0),],alpha=0.75,height=.0,aes(color=clinvar_simple))

ggplot(clinvar_dots_final_filtered_no1c_df, aes(x=function_score_final,y=log10(gv3_allele_freq)))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$gv3_allele_freq > 0),],alpha=1,height=0.00,aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('gnomADv3 AF, Log10')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$gv3_allele_freq == 0),],alpha=0.75,height=.0,aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")

#total gnomAD
ggplot(clinvar_dots_final_filtered_no1c_df, aes(x=function_score_final,y=log2(gv2_allele_count+gv3_allele_count)))+geom_jitter(alpha=1,height=0.1,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('gnomAD v2+v3 allele count, Log2')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

clinvar_dots_final_filtered_no1c_df$clinvar_simple <- factor(clinvar_dots_final_filtered_no1c_df$clinvar_simple, levels = c("absent","REF","Uncertain significance","Benign","Likely benign","Pathogenic","Likely pathogenic","Conflicting interpretations of pathogenicity"))

#FF total gnomAD + bravo + UKB #CANFIG export as VHLrLD_combined_pop_vs_fs 3x5
ggplot(clinvar_dots_final_filtered_no1c_df, aes(x=function_score_final,y=combined_pop_count))+geom_jitter(data=clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$clinvar_simple == "absent"& clinvar_dots_final_filtered_no1c_df$combined_pop_count ==0),],alpha=1,width = 0, height=0.20,aes(color=clinvar_simple))+geom_jitter(data=clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$clinvar_simple != "absent" & clinvar_dots_final_filtered_no1c_df$combined_pop_count ==0),],alpha=1,width = 0, height=0.20,aes(color=clinvar_simple))+geom_jitter(data=clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$clinvar_simple == "absent"& clinvar_dots_final_filtered_no1c_df$combined_pop_count !=0),],alpha=1,width = 0, height=0.1,aes(color=clinvar_simple))+geom_jitter(data=clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$clinvar_simple != "absent" & clinvar_dots_final_filtered_no1c_df$combined_pop_count !=0),],alpha=1,width = 0, height=0.1,aes(color=clinvar_simple))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('Allele count')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12))+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+scale_y_continuous(trans=scales::pseudo_log_trans(base = 2),breaks = c(0,1,10,100,1000))

#FS Fig. 4e Combined population allele freq. stats (main text)
length(which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 0))
length(which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 0))

median(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 0),]$function_score_final)
quantile(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 0),]$function_score_final)
mean(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 0),]$function_score_final)

length(which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 1))
min(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 1),]$function_score_final)

length(which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 4))
min(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 4),]$function_score_final)

vhldb_rcc_counts <- ggplot(clinvar_ord_final_filtered_no1c_df, aes(x=function_score_final,y=vhldb_rcc_count))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$vhldb_rcc_count >= 1),],alpha=1,height=0.1,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('VHLdb RCC entries')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$vhldb_rcc_count == 0),],alpha=0.75,height=.3,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

vhldb_hb_counts <- ggplot(clinvar_ord_final_filtered_no1c_df, aes(x=function_score_final,y=vhldb_hb_count))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$vhldb_hb_count >= 1),],alpha=1,height=0.1,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('VHLdb HB entries')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$vhldb_hb_count == 0),],alpha=0.75,height=.3,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

vhldb_pheo_counts <- ggplot(clinvar_ord_final_filtered_no1c_df, aes(x=function_score_final,y=vhldb_pheo_count))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$vhldb_pheo_count >= 1),],alpha=1,height=0.1,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('VHLdb pheo entries')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))+geom_jitter(data = clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$vhldb_pheo_count == 0),],alpha=0.75,height=.3,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#CANFIG
grid.arrange(vhldb_rcc_counts,vhldb_hb_counts,vhldb_pheo_counts,nrow=3)

#### Other computational predictors

ggplot(clinvar_ord_final_filtered_no1c_df, aes(x=function_score_final,y=as.numeric(as.character(polyphen.val))))+geom_jitter(alpha=1,height=0.00,aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('polyphen score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels = conseq_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(clinvar_ord_final_filtered_no1c_df, aes(x=function_score_final,y=as.numeric(as.character(GerpS))))+geom_jitter(alpha=1,height=0.00,aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('GERP')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels = conseq_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#Boost DM 
ggplot(final_filtered_no1c_df, aes(x=function_score_final,y=boostDM_class))+geom_jitter(alpha=1,height=0.25,aes(color=conseq))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('boostDM class')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors,labels = conseq_labels)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#VARITY comparisons - R, ER, others. All perform about the same.
ggplot(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$conseq == "NON_SYNONYMOUS"),], aes(x=function_score_final,y=as.numeric(as.character(VARITY_R))))+geom_jitter(alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('VARITY R score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$conseq == "NON_SYNONYMOUS"),], aes(x=function_score_final,y=as.numeric(as.character(VARITY_ER))))+geom_jitter(alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('VARITY ER score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$conseq == "NON_SYNONYMOUS"),], aes(x=function_score_final,y=as.numeric(as.character(VARITY_R_LOO))))+geom_jitter(alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('VARITY R LOO score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$conseq == "NON_SYNONYMOUS"),], aes(x=function_score_final,y=as.numeric(as.character(VARITY_ER_LOO))))+geom_jitter(alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('VARITY ER LOO score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#REVEL comparison
ggplot(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$conseq == "NON_SYNONYMOUS"),], aes(x=function_score_final,y=as.numeric(as.character(REVEL))))+geom_jitter(alpha=1,height=0.00,aes(color=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+xlab("Function score")+ylab('REVEL score')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))
####

#LoF variants
VHLrLD_count_STOP_GAINED <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "STOP_GAINED"),])[1]
VHLrLD_count_STOP_GAINED_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "STOP_GAINED"),]$fdr) < 0.01))
VHLrLD_fraction_STOP_GAINED_LoF <- VHLrLD_count_STOP_GAINED_LoF/VHLrLD_count_STOP_GAINED

VHLrLD_count_NON_SYNONYMOUS <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "NON_SYNONYMOUS"),])[1]
VHLrLD_count_NON_SYNONYMOUS_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "NON_SYNONYMOUS"),]$fdr) < 0.01))
VHLrLD_fraction_NON_SYNONYMOUS_LoF <- VHLrLD_count_NON_SYNONYMOUS_LoF/VHLrLD_count_NON_SYNONYMOUS

VHLrLD_count_CANONICAL_SPLICE <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "CANONICAL_SPLICE"),])[1]
VHLrLD_count_CANONICAL_SPLICE_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "CANONICAL_SPLICE"),]$fdr) < 0.01))
VHLrLD_fraction_CANONICAL_SPLICE_LoF <- VHLrLD_count_CANONICAL_SPLICE_LoF/VHLrLD_count_CANONICAL_SPLICE

VHLrLD_count_SPLICE_SITE <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "SPLICE_SITE"),])[1]
VHLrLD_count_SPLICE_SITE_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "SPLICE_SITE"),]$fdr) < 0.01))
VHLrLD_fraction_SPLICE_SITE_LoF <- VHLrLD_count_SPLICE_SITE_LoF/VHLrLD_count_SPLICE_SITE

VHLrLD_count_STOP_LOST <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "STOP_LOST"),])[1]
VHLrLD_count_STOP_LOST_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "STOP_LOST"),]$fdr) < 0.01))
VHLrLD_fraction_STOP_LOST_LoF <- VHLrLD_count_STOP_LOST_LoF/VHLrLD_count_STOP_LOST

VHLrLD_count_3PRIME_UTR <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "3PRIME_UTR"),])[1]
VHLrLD_count_3PRIME_UTR_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "3PRIME_UTR"),]$fdr) < 0.01))
VHLrLD_fraction_3PRIME_UTR_LoF <- VHLrLD_count_3PRIME_UTR_LoF/VHLrLD_count_3PRIME_UTR

VHLrLD_count_INTRONIC <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "INTRONIC"),])[1]
VHLrLD_count_INTRONIC_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "INTRONIC"),]$fdr) < 0.01))
VHLrLD_fraction_INTRONIC_LoF <- VHLrLD_count_INTRONIC_LoF/VHLrLD_count_INTRONIC

####ClinVar - plots by individual category

ggplot(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple),bins=15)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('post/pre, log2 mean')+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))


ggplot(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Uncertain significance"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('post/pre, log2 mean')+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Conflicting interpretations of pathogenicity"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('post/pre, log2 mean')+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

ggplot(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "absent"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('post/pre, log2 mean')+ylab('snvs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))

#FF Final Fig.4a CANFIG - export as FF4a_function_scores_by_clinvar 3 x 7
ggplot(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$clinvar_simple != "absent"),], aes(x=function_score_final))+geom_histogram(alpha=1,aes(fill=clinvar_simple))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Function score')+ylab('SNVs')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_fill_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+scale_color_manual(values=named_conseq_colors_clinvar_path_lp_benign_lb_confl_vus)+coord_cartesian(xlim=c(VHL_no1c_lim_lo,VHL_no1c_lim_hi))
#FS
dim(clinvar_ord_final_filtered_no1c_df[which(clinvar_ord_final_filtered_no1c_df$clinvar_simple != "absent"),])

#fractions of clinvar variants scoring below fdr cut-off of 0.01 for functional effect in assay
VHLrLD_count_Likely_benign <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely benign"),])[1]
VHLrLD_count_Likely_benign_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely benign"),]$fdr) < 0.01))
VHLrLD_fraction_Likely_benign_LoF <- VHLrLD_count_Likely_benign_LoF/VHLrLD_count_Likely_benign

VHLrLD_count_VUS <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Uncertain significance"),])[1]
VHLrLD_count_VUS_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Uncertain significance"),]$fdr) < 0.01))
VHLrLD_fraction_VUS_LoF <- VHLrLD_count_VUS_LoF/VHLrLD_count_VUS

VHLrLD_count_absent <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "absent"),])[1]
VHLrLD_count_absent_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "absent"),]$fdr) < 0.01))
VHLrLD_fraction_absent_LoF <- VHLrLD_count_absent_LoF/VHLrLD_count_absent

VHLrLD_count_total <- dim(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple != "hold"),])[1]
VHLrLD_count_total_LoF <- length(which((final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple != "hold"),]$fdr) < 0.01))
VHLrLD_fraction_total_LoF <- VHLrLD_count_total_LoF/VHLrLD_count_total

#Discordant pathogenic, likely pathogenic variants, stop-gained:
View(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Pathogenic" & final_filtered_no1c_df_cBPu_RCC$function_score_final>VHLrLD_min_LB_norm_RNA_score),c("cHGVS","pHGVS","function_score_final","rna_score","conseq","combined_pop_count","total_cBP_ccRCC","vhldb_syndrome_count","vhldb_t1_count","vhldb_t2_count","vhldb_t2c_count","vhldb_polycyt_count")])

View(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$clinvar_simple == "Likely pathogenic" & final_filtered_no1c_df_cBPu_RCC$function_score_final>VHLrLD_min_LB_norm_RNA_score),c("cHGVS","pHGVS","function_score_final","rna_score","conseq","combined_pop_count","total_cBP_ccRCC","vhldb_syndrome_count","vhldb_t1_count","vhldb_t2_count","vhldb_t2c_count","vhldb_polycyt_count")])

View(final_filtered_no1c_df_cBPu_RCC[which(final_filtered_no1c_df_cBPu_RCC$conseq == "STOP_GAINED" & final_filtered_no1c_df_cBPu_RCC$function_score_final>-1 & as.numeric(final_filtered_no1c_df_cBPu_RCC$protPos) > 54 ),c("cHGVS","pHGVS","function_score_final","rna_score","conseq","combined_pop_count","total_cBP_ccRCC","vhldb_syndrome_count","vhldb_t1_count","vhldb_t2_count","vhldb_t2c_count","vhldb_polycyt_count")])

final_filtered_no1c_df_cBPu_RCC$pos_alt <- final_filtered_no1c_df_cBPu_RCC$pos_alt.x
final_clinical_df$pos_alt <- final_clinical_df$pos_alt.x


VHLrLD_editing_rate_df <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/20230408_VHL_all/pipeline_output/editing_data/20230224_VHLrLD_D6_HDR_rates.txt", sep="\t", header=TRUE)

VHLrLD_editing_rate_df$sge_region <- factor(VHLrLD_editing_rate_df$sge_region, levels = c('exon 1c','exon 1b','exon 1a','intron 1','exon 2','exon 3a','exon 3b'))
VHLrLD_editing_rate_df$experiment <- factor(VHLrLD_editing_rate_df$experiment, levels = c("VHLx1c2rL41","VHLx1c2rL42","VHLx1brL43","VHLx1brL44","VHLx1a2rL41","VHLx1a2rL42","VHLx1prL41","VHLx1prL42","VHLx2rL43","VHLx2rL44","VHLx3arL43","VHLx3arL44","VHLx3brL43","VHLx3brL44"))

sge_region_colors <-
  c("exon 1c" = "#D56209",
  "exon 1b" = "#FF882D",
  "exon 1a" = "#FFAC6C",
  "exon 2" = "#05837A",
  "intron 1" = "#4EB9B1",
  "exon 3a" = "#6666C8",
  "exon 3b" = "#1C1C95")

sge_region_labels2 <-
  c("exon 1c" = "Exon 1, 5'",
  "exon 1b" = "Exon 1, mid",
  "exon 1a" = "Exon 1, 3'",
  "intron 1" = "Intron 1",
  "exon 2" = "Exon 2",
  "exon 3a" = "Exon 3, 5'",
  "exon 3b" = "Exon 3, 3'")                  

#FF Final SFig. 4a Supplementary Fig.4a HDR editing rates by experiment on day 6: VHLrLD_d6_HDR_rates 3x4.5
VHLrLD_editing_rate_df %>%  
  group_by(sge_region) %>%  
  ggplot(aes(x=experiment, y=t_hdr_rate, fill=sge_region)) + 
  geom_bar(stat='identity', position= "dodge")+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank())+xlab('Experiment')+ylab('Day 6 HDR rate')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12), axis.text.x=element_blank(), axis.ticks.x=element_blank(), strip.background = element_blank())+scale_fill_manual(values=sge_region_colors,labels = sge_region_labels2)

#Indel analysis from indel data tables, parsed using python script to group indel events

#Import scores of processed indels (singletons, > 0.001 freq, present in both samples, 100 bp of alignment match on both sides of read)
VHLx3b_indel_scores <- read.table("/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/20230324_indels_analysis/VHLx3b_indels_processed.txt", sep="\t", header=TRUE)

VHLx3b_indel_scores_no_wt <- VHLx3b_indel_scores[VHLx3b_indel_scores$frame_effect.1 != "no_indel",]

#change the "levels" e.g. the order by factoring the frame_effect 
VHLx3b_indel_scores_no_wt$frame_effect.1 <- factor(VHLx3b_indel_scores_no_wt$frame_effect.1  , levels = c("in_frame","frame_shift_A","frame_shift_B"))

reading_frame_colors <- c("Not pathogenic" = "#858585","frame_shift_B" = "#BD0016","frame_shift_A" = "#88CCEE","in_frame" = "#15327C")

#FF Final figure 5 export 3x5 VHLrLD_x3_indel_frame_scores
ggplot(VHLx3b_indel_scores_no_wt)+geom_boxplot(aes(x=frame_effect.1,y=normalised_score,color="Not pathogenic"),outlier.shape=NA)+geom_jitter(height=0,width = 0.1,aes(x=frame_effect.1,y=normalised_score,color=frame_effect.1))+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position = "None")+xlab('Reading Frame')+ylab('Indel score (log2)')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values = reading_frame_colors)+scale_x_discrete(labels=c("In frame","-1 bp frame","+1 bp frame"))

VHLx3b_indel_rf_aov <- aov(normalised_score ~ frame_effect.1, data = VHLx3b_indel_scores_no_wt)
#FS Fig. 5h Summary of the analysis
summary(VHLx3b_indel_rf_aov)
TukeyHSD(VHLx3b_indel_rf_aov)

#Spare stats for main text:
dim(final_filtered_df)
#2,655 possible SNVs across all SGE regions
#2,268 SNVs assayed
#351 SNVs uniquely assable in exon 1c, only 68 assayed:
#283 SNVs missing from exon 1c region ()
#2200/(2655-351) 95.5% coverage from exon 1b on.
dim(final_clinical_df)
which(as.numeric(as.character(final_clinical_df$protPos))==54)
length(which(final_clinical_df$pos<10183691))
dim(final_clinical_df[223,c("cHGVS","pos")])

min(final_clinical_df[which(final_clinical_df$pos<10183691),c("fdr")])
mean(final_clinical_df[which(final_clinical_df$pos<10183691),c("function_score_final")])
median(final_clinical_df[which(final_clinical_df$pos<10183691),c("function_score_final")])
min(final_clinical_df[which(as.numeric(as.character(final_clinical_df$protPos))==54),]$pos)
max(final_clinical_df[which(as.numeric(as.character(final_clinical_df$protPos))==199),]$pos)

length(which(final_clinical_df[which(final_clinical_df$pos > 10183691 & final_clinical_df$pos < 10191605 & final_clinical_df$conseq == "STOP_GAINED"),]$fdr < 0.01))
median(final_clinical_df[which(final_clinical_df$pos > 10183691 & final_clinical_df$pos < 10191605 & final_clinical_df$conseq == "STOP_GAINED"),]$function_score_final)

length(which(final_clinical_df[which(final_clinical_df$pos > 10183691 & final_clinical_df$pos < 10191605 & final_clinical_df$conseq == "CANONICAL_SPLICE"),]$fdr < 0.01))
median(final_clinical_df[which(final_clinical_df$pos > 10183691 & final_clinical_df$pos < 10191605 & final_clinical_df$conseq == "CANONICAL_SPLICE"),]$function_score_final)

length(which(final_clinical_df[which(final_clinical_df$conseq == "NON_SYNONYMOUS"),]$fdr < 1000000))
length(which(final_clinical_df[which(final_clinical_df$conseq == "NON_SYNONYMOUS"),]$fdr < 0.01))

dim(VHLrLD_rna_score_df) #1626 SNVs (coding) with RNA scores
length(VHLrLD_rna_score_df[which(VHLrLD_rna_score_df$rna_score < -3),c("function_score_final")])
length(which(VHLrLD_rna_score_df[which(VHLrLD_rna_score_df$rna_score < -3),c("fdr")]<0.01))
#16/17 with RNA score < -3 are LoF by fdr (17th is < 0.05)
min(VHLrLD_rna_score_df[which(VHLrLD_rna_score_df$rna_score < -3),c("function_score_final")])
max(VHLrLD_rna_score_df[which(VHLrLD_rna_score_df$rna_score < -3),c("function_score_final")])

median(VHLrLD_rna_score_df[VHLrLD_rna_score_df$conseq == "STOP_GAINED",]$rna_score)

#sens. spec. calculation for main text:

cuts = seq(-0.5,0,0.001)


specif_func <- function(cut_off) {
  specif = length(which(cv_benign_df$function_score_final>cut_off)) / length(cv_benign_df$function_score_final)
  return(specif)}

specs <- sapply(cuts, specif_func)

sens_func <- function(cut_off) {
  sensi = length(which(cv_firm_path_df$function_score_final < cut_off)) / length(cv_firm_path_df$function_score_final)
  return(sensi)}

sensis <- sapply(cuts, sens_func)

sens_spec_df <- as.data.frame(cbind(cuts,specs,sensis))
sens_spec_df$error <- (2-sens_spec_df$specs-sens_spec_df$sensis)
min(sens_spec_df$error)
sens_spec_df[which(sens_spec_df$error == min(sens_spec_df$error)),]
optimal_cutoff <- -0.40

#Spec.
length(which(cv_benign_df$function_score_final>optimal_cutoff)) / length(cv_benign_df$function_score_final)
#Sens.
length(which(cv_firm_path_df$function_score_final < optimal_cutoff)) / length(cv_firm_path_df$function_score_final)

#fraction of variants seen in any rcc score as LoF?
length(which(final_clinical_df$total_cBP_rcc>0 & final_clinical_df$fdr < 0.01 )) / length(which(final_clinical_df$total_cBP_rcc>0))
length(which(final_clinical_df$total_cBP_rcc>0 & final_clinical_df$total_cBP_ccRCC>0 ))
length(which(final_clinical_df$total_cBP_rcc>0 & final_clinical_df$total_cBP_ccRCC==0 ))
length(which(final_clinical_df$total_cBP_ccRCC & final_clinical_df$fdr < 0.01 )) / length(which(final_clinical_df$total_cBP_ccRCC>0))

length(which(final_clinical_df$Cancer_type_single == "Other" & final_clinical_df$fdr < 0.01 ))

length(which(final_clinical_df$total_cBP_other > 0))
length(which(final_clinical_df$total_cBP_other > 0 & final_clinical_df$total_cBP_rcc == 0))
length(which(final_clinical_df$total_cBP_other > 0 & final_clinical_df$total_cBP_rcc == 0))
length(which(final_clinical_df$total_cBP_other > 0 & final_clinical_df$fdr < 0.01 ))
length(which(final_clinical_df$fdr < 1.01 ))


length(which(final_clinical_df$clinvar_simple == "Uncertain significance"))
length(which(final_clinical_df$clinvar_simple == "Uncertain significance" & final_clinical_df$ccRCC_fs_class == "Pathogenic"))
length(which(final_clinical_df$clinvar_simple == "Uncertain significance" & final_clinical_df$function_score_final < VHLxrLD_gs_path_cutoff))

length(which(final_clinical_df$clinvar_simple == "absent"))
length(which(final_clinical_df$clinvar_simple == "absent" & final_clinical_df$ccRCC_fs_class == "Pathogenic"))
length(which(final_clinical_df$clinvar_simple == "absent" & final_clinical_df$function_score_final < VHLxrLD_gs_path_cutoff))

#middle 90% in population narrowly defined around 0.
quantile(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 1),]$function_score_final,probs = c(0.05,0.5,0.95))
quantile(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 2),]$function_score_final,probs = c(0.05,0.5,0.95))
quantile(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 3),]$function_score_final,probs = c(0.05,0.5,0.95))
quantile(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 0),]$function_score_final,probs = c(0.25,0.5,0.75))
IQR(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 0),]$function_score_final)
mean(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 1),]$function_score_final)
sd(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 1),]$function_score_final)
min(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 4),]$function_score_final)
mean(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 4),]$function_score_final)
sd(clinvar_dots_final_filtered_no1c_df[which(clinvar_dots_final_filtered_no1c_df$combined_pop_count > 4),]$function_score_final)


#write.table(rbind(VHLrLD_cancer_type_df_papillary[,c("cHGVS","pHGVS","function_score_final","rna_score","total_cBP_ccRCC")],VHLrLD_cancer_type_df_chromophobe[,c("cHGVS","pHGVS","function_score_final","rna_score","total_cBP_ccRCC")]),"/Users/findlag/Downloads/VHL_RCC_nonCC.txt",sep="\t",row.names=F)
      
#write.table(VHLrLD_cancer_type_df_ccrcc[which(VHLrLD_cancer_type_df_ccrcc$fdr > 0.01),c("cHGVS","pHGVS","function_score_final","rna_score","total_cBP_ccRCC","multiple_vars","all_vars_present")],"/Users/findlag/Downloads/VHL_ccRCC_non_LoF.txt",sep="\t",row.names=F)

final_clinical_df[which(final_clinical_df$conseq == "STOP_LOST"),c("cHGVS","pHGVS","function_score_final","fdr")]

```

Creating final variant effect map
```{r}

#CANFIG function score by position:

final_clinical_df$CDSpos <- as.numeric(as.character(final_clinical_df$CDSpos))

exon1_min_cds <- min(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 1b")),]$CDSpos)
exon1_min_pos <- min(final_clinical_df[which(final_clinical_df$CDSpos == exon1_min_cds),]$pos)
exon1_max_cds <- max(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 1a")),]$CDSpos)
exon1_max_pos <- max(final_clinical_df[which(final_clinical_df$CDSpos == exon1_max_cds),]$pos)
#editing here
exon1_gDNA_min_pos <-min(final_clinical_df[which(final_clinical_df$sge_region == "exon 1b"),]$pos)
exon1_gDNA_max_pos <-max(final_clinical_df[which(final_clinical_df$sge_region == "exon 1a"),]$pos)

exon1_CDS_breaks <- append(seq(exon1_min_pos,exon1_max_pos,by=20),exon1_max_pos+10)
exon1_CDS_labels <- append(seq(exon1_min_cds,exon1_max_cds,by=20),paste(exon1_max_cds,"+10",sep=''))

intron1_min_pos <-min(final_clinical_df[which(final_clinical_df$sge_region == "exon 1p"),]$pos)
intron1_max_pos <-max(final_clinical_df[which(final_clinical_df$sge_region == "exon 1p"),]$pos)
intron1_CDS_breaks <- seq(intron1_min_pos,intron1_max_pos,by=20)
intron1_CDS_labels <- substr(final_clinical_df[which(final_clinical_df$sge_region == "exon 1p" & final_clinical_df$pos %in% intron1_CDS_breaks & final_clinical_df$pos_nudge == -0.25),]$cHGVS, start = 3, stop=9)
  
exon2_min_cds <- min(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 2")),]$CDSpos)
exon2_min_pos <- min(final_clinical_df[which(final_clinical_df$CDSpos == exon2_min_cds),]$pos)
exon2_max_cds <- max(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 2")),]$CDSpos)
exon2_max_pos <- max(final_clinical_df[which(final_clinical_df$CDSpos == exon2_max_cds),]$pos)
exon2_CDS_breaks <- append(seq(exon2_min_pos,exon2_max_pos,by=20),exon2_min_pos-15)
exon2_CDS_labels <- append(seq(exon2_min_cds,exon2_max_cds,by=20),paste(exon2_min_cds,"-15",sep=''))

exon3_min_cds <- min(final_clinical_df[which((!is.na(final_clinical_df$CDSpos))&(final_clinical_df$sge_region == "exon 3a")),]$CDSpos)
exon3_min_pos <- min(final_clinical_df[which(final_clinical_df$CDSpos == exon3_min_cds),]$pos)
exon3_max_cds <- max(final_clinical_df[which(!is.na(final_clinical_df$CDSpos)),]$CDSpos)
exon3_max_pos <- max(final_clinical_df[which(final_clinical_df$CDSpos == exon3_max_cds),]$pos)
exon3_CDS_breaks <- append(append(append(seq(exon3_min_pos,exon3_max_pos,by=20),exon3_min_pos-20),exon3_max_pos),exon3_max_pos+20)
exon3_CDS_labels <- append(append(append(seq(exon3_min_cds,exon3_max_cds,by=20),paste(exon3_min_cds,"-20",sep='')),'642'),'*20')

VHLrLD_exon1_map.cDNA <- ggplot(VHLrLD_exon1_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))+scale_x_continuous(breaks=exon1_CDS_breaks,labels=exon1_CDS_labels)
VHLrLD_exon1_map.cDNA

VHLrLD_intron1_map.cDNA <- ggplot(VHLrLD_intron1_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))+scale_x_continuous(breaks=intron1_CDS_breaks,labels=intron1_CDS_labels)
VHLrLD_intron1_map.cDNA

VHLrLD_exon2_map.cDNA <- ggplot(VHLrLD_exon2_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))+scale_x_continuous(breaks=exon2_CDS_breaks,labels=exon2_CDS_labels)
VHLrLD_exon2_map.cDNA

VHLrLD_exon3_map.cDNA <- ggplot(VHLrLD_exon3_df, aes(x=pos, y=function_score_final,color=conseq))+geom_point(alpha=1)+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(legend.title=element_blank(),legend.key = element_blank(),legend.position='None')+ylab('Function score')+xlab('Position')+theme(panel.background = element_rect(fill = 'white', colour = 'white'))+theme(text = element_text(size=12),axis.text.y = element_text(size=12),axis.text.x = element_text(size=12),strip.background = element_blank())+scale_color_manual(values=named_conseq_colors)+coord_cartesian(ylim=c(-4,1))+scale_x_continuous(breaks=exon3_CDS_breaks,labels=exon3_CDS_labels)
VHLrLD_exon3_map.cDNA

#FF CANFIG Fig. 2c export 4x12 as VHL_fs_map_cDNA_coords
grid.arrange(VHLrLD_exon1_map.cDNA,VHLrLD_intron1_map.cDNA,VHLrLD_exon2_map.cDNA,VHLrLD_exon3_map.cDNA,nrow=2)

```


Creating final ST1 Supplementary Table 1
```{r}
##
#create data frame combining final_clinical_df and x1c variants only:
rL4_scores_for_merging_df <- rL4_collpased_final_filtered_df[,c("function_score_sns","fdr","pos_alt")]
rL4_scores_for_merging_df$function_score_sns 
rL4_scores_for_merging_df <- rL4_scores_for_merging_df %>% 
  rename(
    function_score_no_dab = function_score_sns,
    fdr_no_dab = fdr
    )

VHL_SGE_final_df <- merge(final_clinical_df, rL4_scores_for_merging_df, by="pos_alt",all = FALSE,all.x = TRUE, all.y=FALSE)


VHL_SGE_ST1_final_df <- VHL_SGE_final_df[,c("hg38_pos","Ref","alt","cHGVS","pHGVS","function_score_final","fdr","rna_score","post_rna_score", "delta_rna", "ccRCC_fs_class", "sge_region" ,"Intron" , "Exon" , "cDNApos" , "CDSpos" , "Chrom" , "conseq" , "oAA" , "nAA" , "CCDS" , "protPos", "variant","cigar","edit_string", "pre" , "post" , "lib" , "neg" , "rna" , "post2" , "rna2" , "tHDR_pre" , "tHDR_post" , "tHDR_lib" , "tHDR_neg" , "tHDR_rna" , "tHDR_post2", "tHDR_rna2", "corr_pre" , "corr_post" , "rLD2_pre" , "rLD2_lib" , "rLD2_post" , "rLD2_rna" , "rLD2_post2" , "rLD2_rna2" , "rLD2_corr_pre" , "rLD2_corr_post" , "rLD2_tHDR_pre" , "rLD2_tHDR_lib" , "rLD2_tHDR_post" , "rLD2_tHDR_rna" , "rLD2_tHDR_post2"  , "rLD2_tHDR_rna2" , "tHDR_post2_pre_ratio_rLD1rLD2_mean_synnorm" , "tHDR_post_lib_ratio_rLD1rLD2_mean_synnorm" , "tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm" , "tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm" , "tHDR_post_post2_ratio_rLD1rLD2_mean_synnorm", "clinvar_simple", "clinvar_group", "vhldb_rcc_count", "vhldb_pheo_count", "vhldb_hb_count", "vhldb_t1_count", "vhldb_t2_count", "vhldb_t2a_count", "vhldb_t2b_count", "vhldb_t2c_count", "vhldb_entries", "vhldb_p_cyst_count" , "vhldb_pgl_count" , "vhldb_ra_count" , "vhldb_polycyt_count", "vhldb_germline_count", "vhldb_somatic_count", "vhldb_homozygous_count" , "vhldb_t1_only" , "vhldb_t1_or_2b_only" , "vhldb_t2_only" , "vhldb_t2a_only" , "vhldb_t2b_only" , "vhldb_t2c_only" , "vhldb_rcc_only" , "vhldb_rcc_no_pheo" , "vhldb_pheo_only" , "vhldb_no_polycythemia" , "vhldb_rcc_any", "vhl_db_annotation", "Allele.Freq..T." , "Diagnosis.Age" , "Metastatic.Site" , "COSMIC", "TMB..nonsynonymous." , "Overall.Survival.Status" , "Progress.Free.Survival..Months.", "Chromophobe.Renal.Cell.Carcinoma.Ç.." , "Renal.Clear.Cell.Carcinoma" , "Renal.Clear.Cell.Carcinoma.with.Sarcomatoid.Features" , "Papillary.Renal.Cell.Carcinoma" , "Unclassified.Renal.Cell.Carcinoma" , "Unclassified.Kidney.Renal.Cell.Carcinoma", "Pheochromocytoma" , "Renal.Cell.Carcinoma" , "Pancreatic.Neuroendocrine.Tumor" , "Kidney.Renal.Cell.Carcinoma.Other" , "total_cBP_entries" , "total_cBP_rcc" , "total_cBP_other" , "total_cBP_ccRCC" , "OncoKB" ,"CancerHotspot" , "independent_samples", "Cancer_type_single", "OncoKB_simple", "ukb_allele_count" , "ukb_allele_number", "ukb_allele_frequency" , "bravo_het_count" , "bravo_freq" , "gv2_allele_count" , "gv2_allele_number" , "gv2_allele_freq" , "gv3_allele_count" , "gv3_allele_number" , "gv3_allele_freq", "combined_pop_count","CADD.phred", "boostDM_score", "boostDM_class", "EVE_scores_ASM" ,  "VARITY_R" , "VARITY_R_LOO" , "VARITY_ER" , "VARITY_ER_LOO" , "REVEL", "rLH_function_score_final" , "rLH_fdr", "function_score_no_dab","fdr_no_dab")]

final_x1c_df <- final_filtered_df[final_filtered_df$sge_region == "exon 1c",]
#columns applicable to x1c analysis
final_x1c_df <- final_x1c_df[,c("hg38_pos","Ref","alt","cHGVS","pHGVS","function_score_final","fdr", "sge_region" ,"Intron" , "Exon" , "cDNApos" , "CDSpos" , "Chrom" , "conseq" , "oAA" , "nAA" , "CCDS" , "protPos", "variant","cigar","edit_string", "pre" , "post" , "lib" , "neg" , "rna" , "post2" , "rna2" , "tHDR_pre" , "tHDR_post" , "tHDR_lib" , "tHDR_neg" , "tHDR_rna" , "tHDR_post2", "tHDR_rna2", "corr_pre" , "corr_post" , "rLD2_pre" , "rLD2_lib" , "rLD2_post" , "rLD2_rna" , "rLD2_post2" , "rLD2_rna2" , "rLD2_corr_pre" , "rLD2_corr_post" , "rLD2_tHDR_pre" , "rLD2_tHDR_lib" , "rLD2_tHDR_post" , "rLD2_tHDR_rna" , "rLD2_tHDR_post2"  , "rLD2_tHDR_rna2" , "tHDR_post2_pre_ratio_rLD1rLD2_mean_synnorm" , "tHDR_post_lib_ratio_rLD1rLD2_mean_synnorm" , "tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm" , "tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm" , "tHDR_post_post2_ratio_rLD1rLD2_mean_synnorm" , "clinvar_simple")]

VHL_SGE_ST1_final_df <- rbind.fill(VHL_SGE_ST1_final_df, final_x1c_df)

#remove RNA data points from experiments without RNA samples
#rna_columns <- c("rna_score","post_rna_score", "delta_rna", "rna" , "rna2" , "tHDR_rna" , "tHDR_rna2", "rLD2_rna" , "rLD2_rna2" , "rLD2_tHDR_rna", "rLD2_tHDR_rna2" , "tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm" , "tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm")

#replace all missing data with NAs where appropriate
VHL_SGE_ST1_final_df$na_replace <- NA

VHL_SGE_ST1_final_df$rna_score[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$post_rna_score[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$delta_rna[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$rna[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$rna2[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$tHDR_rna[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$tHDR_rna2[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$rLD2_rna[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$rLD2_rna2[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$rLD2_tHDR_rna[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$rLD2_tHDR_rna2[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]
VHL_SGE_ST1_final_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm[VHL_SGE_ST1_final_df$sge_region == "exon 1c"] <- VHL_SGE_ST1_final_df$na_replace[VHL_SGE_ST1_final_df$sge_region == "exon 1c"]

#no RNA scores for intronic regions (removing)
VHL_SGE_ST1_final_df$rna_score[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$post_rna_score[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$delta_rna[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$rna[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$rna2[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$tHDR_rna[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$tHDR_rna2[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$rLD2_rna[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$rLD2_rna2[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$rLD2_tHDR_rna[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$rLD2_tHDR_rna2[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$tHDR_rna_pre_ratio_rLD1rLD2_mean_synnorm[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]
VHL_SGE_ST1_final_df$tHDR_rna2_post_ratio_rLD1rLD2_mean_synnorm[is.na(VHL_SGE_ST1_final_df$Exon)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$Exon)]

#no pHGVS entry for noncoding regions (removing)
VHL_SGE_ST1_final_df$pHGVS[is.na(VHL_SGE_ST1_final_df$protPos)] <- VHL_SGE_ST1_final_df$na_replace[is.na(VHL_SGE_ST1_final_df$protPos)]

#remove NA column
VHL_SGE_ST1_final_df <- VHL_SGE_ST1_final_df[,!names(VHL_SGE_ST1_final_df) %in% c("na_replace")]

#rename some columns for interpretability
VHL_SGE_ST1_final_df <- VHL_SGE_ST1_final_df %>% 
  rename(
    Chromophobe.Renal.Cell.Carcinoma = Chromophobe.Renal.Cell.Carcinoma.Ç..,
    ref = Ref,
    q_value = fdr,
    rna_score_day_20 = post_rna_score,
    Allele.Freq.Tumor = Allele.Freq..T.,
    TMB.nonsynonymous = TMB..nonsynonymous.,
    Progress.Free.Survival.Months = Progress.Free.Survival..Months.,
    independent_cbp_samples = independent_samples,
     function_score_HIF1AKO = rLH_function_score_final,
     q_value_HIF1AKO = rLH_fdr,
     q_value_no_dab = fdr_no_dab
    )

#Open in Excel (specifiy intron and exon columns are not to be text cololumns, then paste headers lines for readability)
write.table(VHL_SGE_ST1_final_df,"/Users/findlag/Downloads/VHL_SGE_ST1_20230505.txt",sep="\t",row.names=F)
#write.table(VHL_SGE_ST1_final_df,"/Volumes/lab-findlayg/home/shared/projects/SGE/VHL/VHL_SGE_ST1.txt",sep="\t",row.names=F)


```




